This is the home page of the metagenomics practical by Laura Dijkhuizen.
Click [here](/) to go back to my home page.


## about the practical
Read more about the practical on the [about page](/metagenomicspractical/about/).
This includes a basic description, instalation details, and learning goals.

## Materials
FastQC reports for the paired end reads of one sample are available here online.
* [Forward reads](fastqc_reports/L1.R1.fastqcreport.html)
* [Reverse reads](fastqc_reports/L1.R2.fastqcreport.html)

An example depth matrix is available [here](data/depth_matrix.tab)

The visualisation (excel) exercises in the notebooks are shared [here via google spreadsheats](https://docs.google.com/spreadsheets/d/1Cdkl8dT75CETGUA_l52Gh8g8qtWd9vv4QAGtTe9HpPU/edit#gid=1292451652).

CheckM examples comming up in a bit...

Prokka mapping tables for visualisation in KEGG are stored in the main branch of the GitHub repo [here](github.com/lauralwd/metagenomicspractical/tree/master/data/blastKOala)

## blank example of the practical
Find the GitHub page for the practical [here](https://github.com/lauralwd/metagenomicspractical). 
Blank static examples of the practical pages are available through the links below: 
* [m00 prepare, download and subset reads](practical_pages/m00-prepare_download_and_subset_reads.html)
* [m01 introduction text](practical_pages/m01-introduction.html)
* [m02 juPyter and BASH basics](practical_pages/m02-jupyter_and_bash_basics.html)
* [m03 asses raw data](practical_pages/m03-assess_raw_data.html)
* [m04 plot assembly length and depth](practical_pages/m04-plot_assembly_length.html)
* [m05 backmapping raw reads](practical_pages/m05-backmapping.html)
* [m06 sorting your bam file](practical_pages/m06-sorting_bamfiles.html)
* [m07 binning part 1](practical_pages/m07-binning_part1.html)
* [m08 binning part 2](practical_pages/m08-binning-part2.html)
* [m09 quality control with checkm](practical_pages/m09-QC_checkm.html)
* [m10 annotation of microbial genomes](practical_pages/m10-annotation.html)
* [m11 bonus: bin taxonomy](practical_pages/m11-bonus_exercise_bin_taxonomy.html)


## example code / answers
A branch of the original GitHub repository contains the practical with worked out examples and some intermediate files.
Find [this branch here](https://github.com/lauralwd/metagenomicspractical/tree/example) or browse through them as static html pages.
Static pages with pre-filled code can be found through the links below:
* [m00 prepare, download and subset reads](practical_example/m00-prepare_download_and_subset_reads.html)
* [m01 introduction text](practical_example/m01-introduction.html)
* [m02 juPyter and BASH basics](practical_example/m02-jupyter_and_bash_basics.html)
* [m03 asses raw data](practical_example/m03-assess_raw_data.html)
* [m04 plot assembly length and depth](practical_example/m04-plot_assembly_length.html)
* [m05 backmapping raw reads](practical_example/m05-backmapping.html)
* [m06 sorting your bam file](practical_example/m06-sorting_bamfiles.html)
* [m07 binning part 1](practical_example/m07-binning_part1.html)
* [m08 binning part 2](practical_example/m08-binning-part2.html)
* [m09 quality control with checkm](practical_example/m09-QC_checkm.html)
* [m10 annotation of microbial genomes](practical_example/m10-annotation.html)
* [m11 bonus: bin taxonomy](practical_example/m11-bonus_exercise_bin_taxonomy.html)

