{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# General introduction\n",
    "This practical exercise teaches you to assemble, bin and annotate a metagenome of a plant: *Azolla filiculoides*. \n",
    "It was developed as part of the Introduction To Bioinformatics for Life-Sciences course at Utrecht University.\n",
    "A metagenome is the collection of genomes of micro-organisms associated with a specific organ or organism.\n",
    "Whether you are a plant biologist or a medical doctor, the techniques you learn in this practical apply to metagenomics on all kinds of organisms or organs. \n",
    "In this particular experiment, we focus on bacterial micro-organisms.\n",
    "Intermediate files, answers and extra information can be found on the practical [home page](https://lauradijkhuizen.com/metagenomicspractical/)\n",
    "\n",
    "The fern genus *Azolla* is known for its endophytes: microorganisms that live inside the plant. \n",
    "These microorganisms live specifically inside a leaf cavity in every upper leaf lobe of the plant. \n",
    "One of these endophytes has been known for a long time: *Nostoc azollae*.\n",
    "They are best known for fixing N^2 from the air which they then provide to the plant as a nutrient.\n",
    "This symbiosis makes the plant independent from nitrogen (fertiliser) in its growth medium.\n",
    "The image included below illustrates the physiology of these leaf cavities. \n",
    "\n",
    "<img src=\"http://theazollafoundation.org/wp-content/uploads/2013/05/Azolla-leaf-cavities-with-Anabaena.png\" width=\"800\">\n",
    "*Image taken from the Azolla foundation website: https://theazollafoundation.org/azolla/the-azolla-superorganism/*\n",
    "\n",
    "While the presence of *N. azollae* is well studied, the presence of other microbes in the *Azolla* leaf pocket is not. \n",
    "Here, we will analyse the metagenome of *A. filiculoides* plants, in search for these other \"neglected\" microbes. \n",
    "Hence, the open biological question we are answering today is **\"which species live inside these leaf cavities and what are the metabolic capacities encoded on their genomes?\"** \n",
    "Based on these genomes, you may be able to infer their function and think of hypotheses to test in further experiments. \n",
    "\n",
    "*Tip*: If you stumble upon a step with a long calculation time, first read ahead to see what steps come next in the subsequent notebooks. \n",
    "Alternatively, you can google some information about the plant you are researching. \n",
    "Notice that this [paper](http://onlinelibrary.wiley.com/doi/10.1111/nph.14843/full) specifically describes the data that we will be using today. \n",
    "For a more general introduction into Azolla, perhaps have a look at [this short movie](https://youtu.be/O34gTsxyDq8) (English) or [this short movie](https://youtu.be/OI4VV4M2-f4) (Dutch) recorded for Dutch local television.\n",
    "\n",
    "# Workflow\n",
    "\n",
    "## Acquire Samples\n",
    "A typical metagenomics workflow starts in the environment or the wet lab with DNA extraction from some biological sample. \n",
    "This is also true for this workflow. \n",
    "We have extracted DNA from *Azolla filiculoides* from the environment in two ways. \n",
    "Firstly, the whole Plant (P), secondly from the Leaf cavities (L). \n",
    "Both were collected in three biological replicates. \n",
    "Hence the samples we will work with are coded L1 L2 L3 and P1 P2 P3. \n",
    "The reasoning behind these two sampling types is that true endophytes are expected to be more abundant in leaf samples than in plant samples. \n",
    "So we may later use these different sampling types to infer if a microbial genome that we assembled originated from the leaf cavity, or from somewhere else in or on the plant. \n",
    "\n",
    "Extracted DNA is processed in a sequencing library; it is prepared in a way that an [Illumina sequencing](https://en.wikipedia.org/wiki/DNA_sequencing#Illumina_(Solexa)_sequencing) machine can process it. \n",
    "The sequencing machine then provides us with [FastQ](https://en.wikipedia.org/wiki/FASTQ_format) files, you may have handled these before, perhaps not. \n",
    "We will have a look at FastQ files later in this practical. \n",
    "Note that often, Illumina machines sequence the two outer ends of a single DNA molecule. \n",
    "We call these **paired-end** sequences. \n",
    "For a quick illustration of how Illumina machines work, have a look at this [video](https://www.youtube.com/watch?v=fCd6B5HRaZ8).\n",
    "\n",
    "## Samples from a database\n",
    "For publication, authors often are required to also upload the raw data on which their publication is based. \n",
    "For sequencing data, the NCBI Short Read Archive (SRA) and the EBI European Nucleotide Archive (ENA) serve this purpose. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[Q:] Find the paper linked above and find the sequencing data it refers to in the EBI ENA database: https://www.ebi.ac.uk/ena/browser/home"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[A]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## First Quality Control and processing.\n",
    "FastQ files contain the output of the Illumina sequencing machine: tens of millions of short DNA sequences. \n",
    "In this context, short is anything between 50 and 250 basepairs. \n",
    "These original short DNA sequences are further referred to as DNA reads, or just reads. \n",
    "FastQ files contain the unprocessed output of the machine, every read consists of four lines: \n",
    "1. a header proceeded by the '@' character. \n",
    "2. the DNA sequence\n",
    "3. a comment line proceeded by a '+'\n",
    "4. a quality string.\n",
    "\n",
    "The quality string has the exact same length as the DNA read, every character in the quality string is a measure of quality corresponding to some number between 0 and 40. \n",
    "This measure of quality is called a [phred score](https://en.wikipedia.org/wiki/Phred_quality_score). \n",
    "The sequencing machine assigns this phred score based on \"how sure it is\" that this specific base is correct.\n",
    "\n",
    "These are two reads encoded in a FastQ file.\n",
    "```\n",
    "@NS500813:28:H3M3VAFXX:1:11101:19270:1015 1:N:0:CGATGT\n",
    "GNGGTGAAGAAATCAGCCATTCTAAACCACACTAATTTAGATTGCTCTCCAAGGA\n",
    "+\n",
    "A#AAAEEEEEEEEEEEEEA#EEEEEEEEEEEEEEEEEEEAEEEEE+/EEEEEEEEEEE\n",
    "@NS500813:28:H3M3VAFXX:1:11101:24948:1015 1:N:0:CGATGT\n",
    "ANATTAATTATACAATCACTAATTTACACTAATTTAGGCAGAGCATTTAGAAAGTGC\n",
    "+\n",
    "A#AAAAEEEEE6EEEEEEEEEEAEEEEE+/EEEEEEEAEEA#EEEEEEEEEEEEEEEE\n",
    "```\n",
    "FastQ files often contain some reads of poor quality, we definitively want to filter those out before starting our analyses. \n",
    "There are many ways available, for today's practical, this was done for you already with a tool called [trimmomatic](https://academic.oup.com/bioinformatics/article-abstract/30/15/2114/2390096).\n",
    "\n",
    "Before and after trimming, general Quality Control (QC( of a FastQ file is assessed with a tool called FastQC. \n",
    "The FastQC reports are rendered as html pages so we can view them in our web browsers. \n",
    "You can find the FastQC reports for this practical here:\n",
    " * [Sample L1 forward](https://lauradijkhuizen.com/metagenomicspractical/fastqc_reports/L1.R1.fastqcreport.html)\n",
    " * [Sample L1 reverse](https://lauradijkhuizen.com/metagenomicspractical/fastqc_reports/L1.R2.fastqcreport.html)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**[DO:] open these FastQC reports and check the quality**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**[Q:]** Are there any \"red flags\"? \n",
    "Do you think these are a problem to continue with the experiment?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**[A]**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filtering\n",
    "Finally, we chose here to filter the sequencing data for plant DNA since we are only interested in microbial DNA.\n",
    "The sequencing data was rid of any plant DNA by mapping (aligning) the reads to a reference plant genome, that of the host plant *Azolla filiculoides*.\n",
    "Only the sequencing data that did not map anything was kept for further analysis.\n",
    "\n",
    "## Assembly\n",
    "Assembly is the process of combining these many short DNA reads in longer contiguous strings of DNA: contigs. \n",
    "For the assembly process, we have used both sample types and all biological replicates as if they were one big sample. \n",
    "This yields the best assembly results. \n",
    "A metagenome assembly is different from a regular genome assembly in that it has DNA of multiple species in one sequencing file. \n",
    "Metagenome assembly was achieved with [SPAdes](http://online.liebertpub.com/doi/abs/10.1089/cmb.2012.0021). \n",
    "\n",
    "Some more elaborate information of Assembly:\n",
    "> The process of assembly takes the millions of short reads in a fastq file, and processes these to thousands of longer nucleotide sequences. Typically the output is a fasta file with contiguous sequences called contigs (contigs.fasta) or a fasta file with multiple contigs scaffolded based on their relative position and orientation to each other called scaffolds, in a file scaffolds.fasta. Ideally, these scaffolds represent the genome of a single species in the metagenome. This is, however, extremely unlikely when no long-read sequencing method was used. In reality, the assembly process will often produce hundreds or thousands of longer scaffolds where a single species' genome is still represented by multiple sequences in the assembly fasta file. (Meta)Genome assembly is a complex process. If this has your interest I recommend watching [this video](https://www.youtube.com/watch?v=TNYZZKrjCSk) and frankly the entire series that this video is a part of.\n",
    "\n",
    "In biology, we typically store DNA sequences in fasta files.\n",
    "Like a fastQ file, it is a text file with DNA (or protein) sequences. \n",
    "Also fasta files can contain multiple sequences. \n",
    "Fasta files are simpler than fastq files, they don't contain quality information.\n",
    "That leaves just two types of lines: a sequence name (header) and the actual sequence.\n",
    "The header line is preceded by a `>` and the lines after containing the DNA sequence. The DNA sequence can continue until the next `>` or until the end of the document. We may have a quick look at a fasta file:\n",
    "> \\>Sequenceheader <br>\n",
    ">  ACTCATGAGACTAGACA\n",
    "\n",
    "The assembly was already done for you.\n",
    "With this practical, you'll find a fasta file with the scaffolds we assembled.\n",
    "\n",
    "In case you're interested, the assembly was made with SPAdes using code like this\n",
    "```\n",
    "spades -t 12           \\\n",
    "       -m 60           \\\n",
    "       --pe1-1 data/reads/E1.R1.fastq.gz \\\n",
    "       --pe1-2 data/reads/E1.R1.fastq.gz \\\n",
    "       --pe2-1 data/reads/E2.R1.fastq.gz \\\n",
    "       --pe2-2 data/reads/E2.R1.fastq.gz \\\n",
    "       --pe3-1 data/reads/E3.R1.fastq.gz \\\n",
    "       --pe3-2 data/reads/E3.R1.fastq.gz \\\n",
    "       --pe4-1 data/reads/P1.R1.fastq.gz \\\n",
    "       --pe4-2 data/reads/P1.R1.fastq.gz \\\n",
    "       --pe5-1 data/reads/P2.R1.fastq.gz \\\n",
    "       --pe5-2 data/reads/P2.R1.fastq.gz \\\n",
    "       --pe6-1 data/reads/P3.R1.fastq.gz \\\n",
    "       --pe6-2 data/reads/P3.R1.fastq.gz \\\n",
    "       -o data/assembly/\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Here you will take over the workflow\n",
    "From here on, you will take over the workflow. \n",
    "Each major step is listed below and shortly explained. \n",
    "All these steps are executed in separate notebooks. \n",
    "Often the last step has a computing time of a few minutes. \n",
    "During this time you may proceed to the next notebook to prepare the subsequent steps.\n",
    "To keep the practical feasible, we have provided you with a subset of the assembly and a subset of the reads. \n",
    "The complete workflow you will follow is seen on this sketch below. \n",
    "Whenever you are lost in the command-line, error messages and output tables, have a look again at this flowchart.\n",
    "In these bioinformatics workflows, it is essential to keep a feeling for where you are, where you are going and where you come from.  \n",
    "\n",
    "The programmes mentioned in the flowchart are preinstalled on this virtual machine you are currently working on:\n",
    "bwa: http://bio-bwa.sourceforge.net/ <br>\n",
    "samtools: http://samtools.sourceforge.net/ <br>\n",
    "binningmetabat: https://bitbucket.org/berkeleylab/metabat/overview <br>\n",
    "checkm: https://github.com/Ecogenomics/CheckM/wiki/Installation#how-to-install-checkm <br>\n",
    "prokka: http://www.vicbioinformatics.com/software.prokka.shtml <br>\n",
    "\n",
    "Each of these steps is contained in a separate notebook:\n",
    "\n",
    "## Jupyter and Bash Basics `m02`\n",
    "Before we will get our hands dirty with the 'real data', we will first do some exercises to learn how to work in a notebook like this then we will continue the metagenomics workflow. \n",
    "To keep all information central, I will continue here with the explanation of the workflow. \n",
    "The exercises will be available in another notebook after you have read the full introduction. \n",
    "(In the Introduction to Bioinformatics course, you have already done this exercise on day 1).\n",
    "\n",
    "## Assess raw data `m03` & `m04`\n",
    "After we have learned about bash and Jupyter notebooks, we will practice our bash skills by having a quick look at the data files and assessing the quality.\n",
    "Secondly, we'll use a bit of simple python code to plot the length distribution and abundances of the scaffolds in the original biological samples.\n",
    "\n",
    "## Back mapping `m05`\n",
    "After assembly of the scaffolds, we want to know how abundant each scaffold is in the different metagenomic samples. \n",
    "To do this, we align the Illumina reads (FastQ files) to the scaffolds (fasta) of the assembly in a step called **back mapping** with a tool called BWA (Burrows-wheeler aligner). \n",
    "BWA will provide us then with a table of which reads mapped on the metagenome assembly and their specific coördinates: a `.bam` file.\n",
    "\n",
    "### Sorting bam files `m06`\n",
    "These tables in the `.bam` files must be sorted by coordinate on the genome assembly.\n",
    "This we achieve with a tool called `samtools`. \n",
    "Strictly speaking, this is still part of the back mapping procedure, but here it is a separate notebook.\n",
    "\n",
    "## Binning `m07` & `m08`\n",
    "A single metagenome assembly contains scaffolds representing DNA of many species in the original sample, but without any information on which scaffold belongs to which species. \n",
    "To tackle this issue, we will categorise each individual scaffold in a 'bin' which then ideally comprises one species in the metagenome. \n",
    "In recent literature, these bins are often called Metagenome Assembled Genomes or MAGs.\n",
    "Despite the name change, the procedure is still often called binning. \n",
    "The binning procedure needs the bam files we will make in the 'back mapping' step. \n",
    "The tool we use for binning is called `metabat2`. \n",
    "Metabat2 uses the sorted bamfiles to subdivide the scaffolds into bins.\n",
    "This will produce six bins or MAGs in this particular experiment.\n",
    "\n",
    "The binning process is again subdivided over two notebooks.\n",
    "In part 1 (`m07`), we take bam files, then calculate the depth of coverage for each biological sample on each scaffold in the assembly.\n",
    "In part 2, we take the original scaffolds in fasta format and the depths calculated in part one to create the bins or MAGs.\n",
    "At this stage, you have succesfully moved from millions of very short sequences (reads) with very little biological information to thousands of longer scaffolds with more biological information, to dozens of long sequences with a lot of biological information. \n",
    "\n",
    "## QC with CheckM `m09`\n",
    "One bin ideally represents one species. \n",
    "To check whether this is the case, we can use the tool `checkm`. \n",
    "CheckM assesses the completeness and contamination of bins based on single-copy-marker-genes (collocated sets of genes that are ubiquitous and single-copy within a phylogenetic lineage). \n",
    "\n",
    "## Annotation `m10`\n",
    "Annotation is the process of adding functional information to parts of a DNA sequence, i.e. making a table of genes and locations of these genes. \n",
    "These are often stored in a GFF file, the genomic feature format. \n",
    "Once you have this gff file, and a fasta for your specific bin/microbial species, we will visualise it using KEGG.\n",
    "Again, for more details please google kegg. \n",
    "Here we will use it to load our gff file and then explore the metabolic pathways present. \n",
    "\n",
    "## Bin taxonomy with BAT  `m11`\n",
    "This step is very resource intensive, hence I only show examples here.\n",
    "\n",
    "## Phylogeny (Bonus) `m12`\n",
    "As a bonus exercise, you can make a simple species phylogeny of the cyanobacteria you retrieved during this practical."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
