{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mapping reads to scaffolds with BWA (Burrows-Wheeler Aligner) and sorting with Samtools\n",
    "We now have seen the reads and the assembly with our own eyes. \n",
    "In any bio-informatic step, learn yourself always to have a peek into the actual data you are producing. \n",
    "Since you now have seen that the files look OK, we can start by mapping (aligning) the original reads in `.fastq.gz` files back to the scaffolds created with the assembly; the `scaffolds.fasta` file. \n",
    "Doing this will allow us to calculate the **depth on the scaffolds**; a prerequisite for the binning procedure.\n",
    "\n",
    "Algning is achieved with bwa.\n",
    "Per sample, we map the reads against the scaffolds and save the mapping as a `.bam` file. \n",
    "1. To run `bwa`, we first need to make an index file of the scaffolds. \n",
    "2. Then we can run `bwa` and immediately pipe the output of `bwa` through `samtools view` which will output a `.bam` file .\n",
    "3. Finally, since we have quite a few `.bam` files to align to the assembly, we will make a `for` loop that will iterate over the different samples we have. \n",
    "\n",
    "Rember that the reads are paired-end Illumina sequences, which means that for each DNA fragment, we have sequence data from both ends. \n",
    "Therefore, the sequences are stored in two separate files (one for the data from each end). \n",
    "We'll use `bwa` with default settings to align the reads back to the scaffolds, and then `samtools` allows us to work with the bwa results.\n",
    "\n",
    "**[Q:] Why this called back-mapping and not just mapping?**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**[A:]**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**[Q:] Why do we not use the depth in the scaffold names?**\n",
    "\n",
    "<details>\n",
    "    <summary>&#10551; Click here for a hint</summary> \n",
    "    Remember that in the back-mapping step, we map, store and process all samples individually.\n",
    "</details>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**[A:]**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The next assignment is a complicated one.\n",
    "Make sure to read the instructions carefully and remember what you learned about bash loops earlier.\n",
    "\n",
    "When in doubt, don't forget to read the 'help page' of a command like so:\n",
    "> ls --help\n",
    "\n",
    "or sometimes by executing the command without any arguments like so:\n",
    "> bwa\n",
    "\n",
    "> bwa index\n",
    "\n",
    "For more elaborate information, read the manual with the `man` command.\n",
    "Unfortunately, this doesn't work well inside Jupyter notebooks.\n",
    "> man bwa\n",
    "\n",
    "> man samtools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1 Index the assembly with bwa index\n",
    "**[DO:] Create an index of the scaffolds with** `bwa index`**\n",
    "Start by reading the `bwa index` instructions. \n",
    "1. Remember to look for the 'usage' line first. \n",
    "2. You won't need any options here.\n",
    "3. Remember where the assembly was? We just gUnzipped it in the last notebook. \n",
    "4. use `ls` and auto completion to find your way to the right file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Usage:   bwa index [options] <in.fasta>\n",
      "\n",
      "Options: -a STR    BWT construction algorithm: bwtsw, is or rb2 [auto]\n",
      "         -p STR    prefix of the index [same as fasta name]\n",
      "         -b INT    block size for the bwtsw algorithm (effective with -a bwtsw) [10000000]\n",
      "         -6        index files named as <in.fasta>.64.* instead of <in.fasta>.* \n",
      "\n",
      "Warning: `-a bwtsw' does not work for short genomes, while `-a is' and\n",
      "         `-a div' do not work not for long genomes.\n",
      "\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bwa index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scaffolds.fasta  \u001b[0m\u001b[01;31mscaffolds.fasta.gz\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls data/assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 0.61 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=210601834, availableWord=26818708\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 44238506 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 81726650 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 115042122 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 144649002 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 170959642 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 194340570 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 68 iterations.\n",
      "[bwa_index] 61.56 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.51 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.40 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 31.49 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index ./data/assembly/scaffolds.fasta\n",
      "[main] Real time: 96.467 sec; CPU: 94.577 sec\n"
     ]
    }
   ],
   "source": [
    "bwa index ./data/assembly/scaffolds.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "While this runs, think about what is happening. \n",
    "Mapping always takes a query sequence and a reference sequence. \n",
    "The reference is our assembly, specifically the `scaffolds.fasta` file. \n",
    "What `bwa index` does, is take this reference and 'index' it. \n",
    "This means as much as it converts this regular text fasta file, into a binary format that computers can efficiently search."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check the output of bwa\n",
    "After creating the index, there should be multiple files in the assembly folder. \n",
    "Think about what command you could use to see if that is the case. \n",
    "Use the empty cell below.\n",
    "\n",
    "**[DO:] Check if the index files were created:** "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scaffolds.fasta      scaffolds.fasta.bwt  scaffolds.fasta.sa\n",
      "scaffolds.fasta.amb  \u001b[0m\u001b[01;31mscaffolds.fasta.gz\u001b[0m\n",
      "scaffolds.fasta.ann  scaffolds.fasta.pac\n"
     ]
    }
   ],
   "source": [
    "ls data/assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Besides the original scaffolds file, we now expect multiple other files. \n",
    "All of these together comprise the `index` we just made.\n",
    "A bwa index of a fasta file can be seen as a binary representation of the fasta file that bwa can search efficiently. \n",
    "It is not meant for human eyes, but purely for the computer algorithms to search through."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run bwa mem for backmapping\n",
    "For this part, we will combine your skills on **bash loops** and **variables** to run bwa mem and samtools view on all the reads in data/reads and create BAM output files in a newly made directory.\n",
    "\n",
    "Then we use a bash for loop to run samtools sort on all the files created by samtools view\n",
    "\n",
    "Step by step instructions:\n",
    "1. make a new directory for the samtools view results, give this a useful name\n",
    "    1. I suggest ./data/mapped\n",
    "    2.  to make a directory, use the command `mkdir`\n",
    "2. See how the loop I pre-made for you works, play a bit to see if you understand correctly\n",
    "3. Don't forget to read the manual page of both! You can make extra cells to print these manual pages if you want to.\n",
    "4. finish the for loop and run both these commands in a pipe\n",
    "    1. `bwa mem` \n",
    "    2. `samtools view`\n",
    "5. bwa mem does not require any options, \n",
    "6. For `samtools view`, look for the options to convert to a BAM file and write the output to a file.\n",
    "   \n",
    "It should look somewhat like this `bwa mem argument argument.fastq  | samtools view argument argument`\n",
    "    \n",
    "If you forgot how a loop works, check the notebook of this morning' m1-jupyter_and_bash_basics.ipynb' .\n",
    "\n",
    "Mapping each separate sample will take about 5 to 6 minutes.\n",
    "If you think your loop works (read: it doesn't crash immediately), then check in the next notebook if the files are created and if they increase in size. Use `ls -sh`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**[DO:] 1 First make your new directory here:**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir ./data/mapped"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**[DO:] 2 now see how this loop works:**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L1\n",
      "L2\n",
      "L3\n",
      "P1\n",
      "P2\n",
      "P3\n"
     ]
    }
   ],
   "source": [
    "# first I define the samples in a variable called 'samples'\n",
    "samples=( L1 L2 L3 P1 P2 P3 )\n",
    "# next I use this variable in my loop\n",
    "for i in ${samples[@]}\n",
    "    do echo $i  \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**[DO:] 3 Read the manual pages of bwa and samtools.** \n",
    "Remember to find the **usage** lines."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Usage: bwa mem [options] <idxbase> <in1.fq> [in2.fq]\n",
      "\n",
      "Algorithm options:\n",
      "\n",
      "       -t INT        number of threads [1]\n",
      "       -k INT        minimum seed length [19]\n",
      "       -w INT        band width for banded alignment [100]\n",
      "       -d INT        off-diagonal X-dropoff [100]\n",
      "       -r FLOAT      look for internal seeds inside a seed longer than {-k} * FLOAT [1.5]\n",
      "       -y INT        seed occurrence for the 3rd round seeding [20]\n",
      "       -c INT        skip seeds with more than INT occurrences [500]\n",
      "       -D FLOAT      drop chains shorter than FLOAT fraction of the longest overlapping chain [0.50]\n",
      "       -W INT        discard a chain if seeded bases shorter than INT [0]\n",
      "       -m INT        perform at most INT rounds of mate rescues for each read [50]\n",
      "       -S            skip mate rescue\n",
      "       -P            skip pairing; mate rescue performed unless -S also in use\n",
      "\n",
      "Scoring options:\n",
      "\n",
      "       -A INT        score for a sequence match, which scales options -TdBOELU unless overridden [1]\n",
      "       -B INT        penalty for a mismatch [4]\n",
      "       -O INT[,INT]  gap open penalties for deletions and insertions [6,6]\n",
      "       -E INT[,INT]  gap extension penalty; a gap of size k cost '{-O} + {-E}*k' [1,1]\n",
      "       -L INT[,INT]  penalty for 5'- and 3'-end clipping [5,5]\n",
      "       -U INT        penalty for an unpaired read pair [17]\n",
      "\n",
      "       -x STR        read type. Setting -x changes multiple parameters unless overridden [null]\n",
      "                     pacbio: -k17 -W40 -r10 -A1 -B1 -O1 -E1 -L0  (PacBio reads to ref)\n",
      "                     ont2d: -k14 -W20 -r10 -A1 -B1 -O1 -E1 -L0  (Oxford Nanopore 2D-reads to ref)\n",
      "                     intractg: -B9 -O16 -L5  (intra-species contigs to ref)\n",
      "\n",
      "Input/output options:\n",
      "\n",
      "       -p            smart pairing (ignoring in2.fq)\n",
      "       -R STR        read group header line such as '@RG\\tID:foo\\tSM:bar' [null]\n",
      "       -H STR/FILE   insert STR to header if it starts with @; or insert lines in FILE [null]\n",
      "       -o FILE       sam file to output results to [stdout]\n",
      "       -j            treat ALT contigs as part of the primary assembly (i.e. ignore <idxbase>.alt file)\n",
      "       -5            for split alignment, take the alignment with the smallest coordinate as primary\n",
      "       -q            don't modify mapQ of supplementary alignments\n",
      "       -K INT        process INT input bases in each batch regardless of nThreads (for reproducibility) []\n",
      "\n",
      "       -v INT        verbosity level: 1=error, 2=warning, 3=message, 4+=debugging [3]\n",
      "       -T INT        minimum score to output [30]\n",
      "       -h INT[,INT]  if there are <INT hits with score >80% of the max score, output all in XA [5,200]\n",
      "       -a            output all alignments for SE or unpaired PE\n",
      "       -C            append FASTA/FASTQ comment to SAM output\n",
      "       -V            output the reference FASTA header in the XR tag\n",
      "       -Y            use soft clipping for supplementary alignments\n",
      "       -M            mark shorter split hits as secondary\n",
      "\n",
      "       -I FLOAT[,FLOAT[,INT[,INT]]]\n",
      "                     specify the mean, standard deviation (10% of the mean if absent), max\n",
      "                     (4 sigma from the mean if absent) and min of the insert size distribution.\n",
      "                     FR orientation only. [inferred]\n",
      "\n",
      "Note: Please read the man page for detailed description of the command line and options.\n",
      "\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bwa mem"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Usage: samtools view [options] <in.bam>|<in.sam>|<in.cram> [region ...]\n",
      "\n",
      "Options:\n",
      "  -b       output BAM\n",
      "  -C       output CRAM (requires -T)\n",
      "  -1       use fast BAM compression (implies -b)\n",
      "  -u       uncompressed BAM output (implies -b)\n",
      "  -h       include header in SAM output\n",
      "  -H       print SAM header only (no alignments)\n",
      "  -c       print only the count of matching records\n",
      "  -o FILE  output file name [stdout]\n",
      "  -U FILE  output reads not selected by filters to FILE [null]\n",
      "  -t FILE  FILE listing reference names and lengths (see long help) [null]\n",
      "  -X       include customized index file\n",
      "  -L FILE  only include reads overlapping this BED FILE [null]\n",
      "  -r STR   only include reads in read group STR [null]\n",
      "  -R FILE  only include reads with read group listed in FILE [null]\n",
      "  -d STR:STR\n",
      "           only include reads with tag STR and associated value STR [null]\n",
      "  -D STR:FILE\n",
      "           only include reads with tag STR and associated values listed in\n",
      "           FILE [null]\n",
      "  -q INT   only include reads with mapping quality >= INT [0]\n",
      "  -l STR   only include reads in library STR [null]\n",
      "  -m INT   only include reads with number of CIGAR operations consuming\n",
      "           query sequence >= INT [0]\n",
      "  -f INT   only include reads with all  of the FLAGs in INT present [0]\n",
      "  -F INT   only include reads with none of the FLAGS in INT present [0]\n",
      "  -G INT   only EXCLUDE reads with all  of the FLAGs in INT present [0]\n",
      "  -s FLOAT subsample reads (given INT.FRAC option value, 0.FRAC is the\n",
      "           fraction of templates/read pairs to keep; INT part sets seed)\n",
      "  -M       use the multi-region iterator (increases the speed, removes\n",
      "           duplicates and outputs the reads as they are ordered in the file)\n",
      "  -x STR   read tag to strip (repeatable) [null]\n",
      "  -B       collapse the backward CIGAR operation\n",
      "  -?       print long help, including note about region specification\n",
      "  -S       ignored (input format is auto-detected)\n",
      "  --no-PG  do not add a PG line\n",
      "      --input-fmt-option OPT[=VAL]\n",
      "               Specify a single input file format option in the form\n",
      "               of OPTION or OPTION=VALUE\n",
      "  -O, --output-fmt FORMAT[,OPT[=VAL]]...\n",
      "               Specify output format (SAM, BAM, CRAM)\n",
      "      --output-fmt-option OPT[=VAL]\n",
      "               Specify a single output file format option in the form\n",
      "               of OPTION or OPTION=VALUE\n",
      "  -T, --reference FILE\n",
      "               Reference sequence FASTA FILE [null]\n",
      "  -@, --threads INT\n",
      "               Number of additional threads to use [0]\n",
      "      --write-index\n",
      "               Automatically index the output files [off]\n",
      "      --verbosity INT\n",
      "               Set level of verbosity\n",
      "\n"
     ]
    }
   ],
   "source": [
    "samtools view"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now it's up to you! Here you have another variant of the loop I made above. \n",
    "Substitute the `ls` command for a `bwa` command. \n",
    "Also, note that you can use the variable inside a path!\n",
    "\n",
    "**[DO:] Map the paired-end reads of each sample to the index scaffolds and save the output as a** `.bam` **file.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 66522 sequences (10000240 bp)...\n",
      "[M::process] read 66530 sequences (10000146 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 24302, 3, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (465, 542, 621)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (153, 933)\n",
      "[M::mem_pestat] mean and std.dev: (540.79, 119.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1089)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66522 reads in 9.961 CPU sec, 9.837 real sec\n",
      "[M::process] read 66546 sequences (10000170 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 24410, 2, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (465, 541, 618)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (159, 924)\n",
      "[M::mem_pestat] mean and std.dev: (539.39, 117.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (6, 1077)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66530 reads in 9.434 CPU sec, 9.273 real sec\n",
      "[M::process] read 66548 sequences (10000211 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 24254, 0, 9)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (463, 540, 617)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (155, 925)\n",
      "[M::mem_pestat] mean and std.dev: (537.71, 117.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1079)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66546 reads in 11.190 CPU sec, 11.034 real sec\n",
      "[M::process] read 66566 sequences (10000064 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 24211, 0, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (462, 540, 615)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (156, 921)\n",
      "[M::mem_pestat] mean and std.dev: (536.74, 117.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (3, 1074)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66548 reads in 15.693 CPU sec, 15.469 real sec\n",
      "[M::process] read 66554 sequences (10000211 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 24254, 4, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (466, 542, 617)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (164, 919)\n",
      "[M::mem_pestat] mean and std.dev: (538.74, 118.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (13, 1070)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66566 reads in 13.012 CPU sec, 12.862 real sec\n",
      "[M::process] read 66556 sequences (10000016 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 24127, 4, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (463, 540, 617)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (155, 925)\n",
      "[M::mem_pestat] mean and std.dev: (538.18, 118.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1079)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66554 reads in 10.621 CPU sec, 10.454 real sec\n",
      "[M::process] read 66568 sequences (10000247 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 24245, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (465, 543, 620)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (155, 930)\n",
      "[M::mem_pestat] mean and std.dev: (540.17, 118.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1085)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66556 reads in 11.501 CPU sec, 11.257 real sec\n",
      "[M::process] read 66568 sequences (10000190 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 24225, 1, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (465, 541, 617)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (161, 921)\n",
      "[M::mem_pestat] mean and std.dev: (538.81, 117.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (9, 1073)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66568 reads in 14.206 CPU sec, 14.056 real sec\n",
      "[M::process] read 66578 sequences (10000282 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 24110, 4, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (464, 540, 617)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (158, 923)\n",
      "[M::mem_pestat] mean and std.dev: (538.23, 119.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (5, 1076)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66568 reads in 12.631 CPU sec, 12.469 real sec\n",
      "[M::process] read 66576 sequences (10000052 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 24187, 3, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (462, 540, 615)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (156, 921)\n",
      "[M::mem_pestat] mean and std.dev: (536.15, 118.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (3, 1074)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66578 reads in 12.391 CPU sec, 12.303 real sec\n",
      "[M::process] read 66574 sequences (10000021 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 24059, 2, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (464, 540, 615)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (162, 917)\n",
      "[M::mem_pestat] mean and std.dev: (537.47, 116.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (11, 1068)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66576 reads in 15.195 CPU sec, 15.028 real sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::process] read 66574 sequences (10000292 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 24095, 1, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (462, 539, 615)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (156, 921)\n",
      "[M::mem_pestat] mean and std.dev: (536.18, 117.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (3, 1074)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66574 reads in 10.721 CPU sec, 10.550 real sec\n",
      "[M::process] read 66570 sequences (10000186 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 24231, 3, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (463, 540, 615)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (159, 919)\n",
      "[M::mem_pestat] mean and std.dev: (536.93, 117.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (7, 1071)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66574 reads in 9.426 CPU sec, 9.268 real sec\n",
      "[M::process] read 66576 sequences (10000022 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 24143, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (462, 540, 614)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (158, 918)\n",
      "[M::mem_pestat] mean and std.dev: (536.58, 118.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (6, 1070)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66570 reads in 9.090 CPU sec, 8.934 real sec\n",
      "[M::process] read 66560 sequences (10000009 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 24215, 5, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (465, 542, 617)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (161, 921)\n",
      "[M::mem_pestat] mean and std.dev: (538.89, 117.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (9, 1073)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66576 reads in 9.136 CPU sec, 8.973 real sec\n",
      "[M::process] read 66566 sequences (10000173 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 24143, 4, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (463, 540, 617)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (155, 925)\n",
      "[M::mem_pestat] mean and std.dev: (537.97, 118.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1079)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66560 reads in 9.186 CPU sec, 9.028 real sec\n",
      "[M::process] read 66578 sequences (10000159 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 24116, 4, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (464, 539, 617)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (158, 923)\n",
      "[M::mem_pestat] mean and std.dev: (537.23, 117.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (5, 1076)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66566 reads in 9.083 CPU sec, 8.925 real sec\n",
      "[M::process] read 66580 sequences (10000120 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 24224, 2, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (465, 540, 618)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (159, 924)\n",
      "[M::mem_pestat] mean and std.dev: (539.29, 117.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (6, 1077)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66578 reads in 9.347 CPU sec, 9.193 real sec\n",
      "[M::process] read 66568 sequences (10000035 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 24228, 5, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (464, 541, 617)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (158, 923)\n",
      "[M::mem_pestat] mean and std.dev: (538.51, 118.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (5, 1076)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66580 reads in 9.108 CPU sec, 8.951 real sec\n",
      "[M::process] read 66582 sequences (10000285 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 24140, 4, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (464, 541, 618)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (156, 926)\n",
      "[M::mem_pestat] mean and std.dev: (539.12, 118.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (2, 1080)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66568 reads in 9.016 CPU sec, 8.854 real sec\n",
      "[M::process] read 66572 sequences (10000301 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 24223, 3, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (465, 540, 618)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (159, 924)\n",
      "[M::mem_pestat] mean and std.dev: (538.64, 117.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (6, 1077)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66582 reads in 9.113 CPU sec, 8.961 real sec\n",
      "[M::process] read 66578 sequences (10000202 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 24208, 1, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (466, 540, 616)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (166, 916)\n",
      "[M::mem_pestat] mean and std.dev: (537.89, 117.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (16, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66572 reads in 9.132 CPU sec, 8.967 real sec\n",
      "[M::process] read 66568 sequences (10000166 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 24273, 3, 5)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (463, 539, 616)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (157, 922)\n",
      "[M::mem_pestat] mean and std.dev: (537.01, 117.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (4, 1075)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66578 reads in 9.169 CPU sec, 9.012 real sec\n",
      "[M::process] read 66558 sequences (10000010 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 24145, 1, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (463, 541, 618)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (153, 928)\n",
      "[M::mem_pestat] mean and std.dev: (538.07, 119.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1083)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66568 reads in 9.078 CPU sec, 8.922 real sec\n",
      "[M::process] read 66556 sequences (10000037 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 24243, 3, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (464, 541, 619)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (154, 929)\n",
      "[M::mem_pestat] mean and std.dev: (538.87, 118.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1084)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66558 reads in 9.111 CPU sec, 8.953 real sec\n",
      "[M::process] read 66564 sequences (10000119 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 24110, 2, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (463, 540, 617)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (155, 925)\n",
      "[M::mem_pestat] mean and std.dev: (537.46, 118.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1079)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66556 reads in 9.222 CPU sec, 9.065 real sec\n",
      "[M::process] read 66574 sequences (10000151 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 24300, 4, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (461, 538, 613)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (157, 917)\n",
      "[M::mem_pestat] mean and std.dev: (535.11, 116.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (5, 1069)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66564 reads in 9.098 CPU sec, 8.940 real sec\n",
      "[M::process] read 66548 sequences (10000159 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 24198, 3, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (464, 539, 616)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (160, 920)\n",
      "[M::mem_pestat] mean and std.dev: (537.55, 116.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (8, 1072)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66574 reads in 9.240 CPU sec, 9.084 real sec\n",
      "[M::process] read 66564 sequences (10000269 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 24232, 4, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (462, 538, 614)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (158, 918)\n",
      "[M::mem_pestat] mean and std.dev: (535.04, 117.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (6, 1070)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66548 reads in 9.281 CPU sec, 9.121 real sec\n",
      "[M::process] read 3078 sequences (462865 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 24247, 2, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (463, 539, 613)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (163, 913)\n",
      "[M::mem_pestat] mean and std.dev: (534.96, 115.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (13, 1063)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66564 reads in 9.233 CPU sec, 9.182 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 1059, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (460, 528, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (182, 877)\n",
      "[M::mem_pestat] mean and std.dev: (526.32, 109.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (43, 1016)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 3078 reads in 0.519 CPU sec, 0.470 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem data/assembly/scaffolds.fasta data/reads/L1.R1.fastq.gz data/reads/L1.R2.fastq.gz\n",
      "[main] Real time: 307.658 sec; CPU: 312.386 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 66560 sequences (10000114 bp)...\n",
      "[M::process] read 66558 sequences (10000059 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 24462, 0, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 214, 277)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 587)\n",
      "[M::mem_pestat] mean and std.dev: (217.50, 121.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 742)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 514, 591)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (141, 891)\n",
      "[M::mem_pestat] mean and std.dev: (515.67, 115.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1041)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66560 reads in 9.059 CPU sec, 8.941 real sec\n",
      "[M::process] read 66584 sequences (10000065 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (20, 24501, 3, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 205, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1110)\n",
      "[M::mem_pestat] mean and std.dev: (260.58, 172.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1442)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (442, 514, 591)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 889)\n",
      "[M::mem_pestat] mean and std.dev: (515.43, 115.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1038)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66558 reads in 9.119 CPU sec, 8.961 real sec\n",
      "[M::process] read 66574 sequences (10000045 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 24510, 4, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (95, 174, 320)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 770)\n",
      "[M::mem_pestat] mean and std.dev: (184.83, 112.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 995)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 514, 588)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (147, 882)\n",
      "[M::mem_pestat] mean and std.dev: (513.69, 113.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1029)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66584 reads in 9.133 CPU sec, 8.975 real sec\n",
      "[M::process] read 66598 sequences (10000214 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 24415, 0, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (144, 217, 316)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 660)\n",
      "[M::mem_pestat] mean and std.dev: (221.17, 114.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 832)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 513, 591)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (141, 891)\n",
      "[M::mem_pestat] mean and std.dev: (514.13, 116.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1041)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66574 reads in 9.109 CPU sec, 8.947 real sec\n",
      "[M::process] read 66576 sequences (10000250 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (16, 24314, 1, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 277, 363)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 683)\n",
      "[M::mem_pestat] mean and std.dev: (261.12, 152.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 872)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 512, 590)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 888)\n",
      "[M::mem_pestat] mean and std.dev: (513.73, 114.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1037)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66598 reads in 9.151 CPU sec, 8.992 real sec\n",
      "[M::process] read 66602 sequences (10000209 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 24399, 1, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (66, 276, 393)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1047)\n",
      "[M::mem_pestat] mean and std.dev: (235.69, 162.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1374)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 513, 591)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (141, 891)\n",
      "[M::mem_pestat] mean and std.dev: (514.18, 116.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1041)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66576 reads in 9.006 CPU sec, 8.856 real sec\n",
      "[M::process] read 66612 sequences (10000195 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 24241, 1, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (148, 197, 393)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 883)\n",
      "[M::mem_pestat] mean and std.dev: (233.67, 125.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1128)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 514, 590)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 888)\n",
      "[M::mem_pestat] mean and std.dev: (515.16, 115.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1037)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66602 reads in 9.198 CPU sec, 9.035 real sec\n",
      "[M::process] read 66606 sequences (10000115 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 24324, 1, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (95, 203, 397)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1001)\n",
      "[M::mem_pestat] mean and std.dev: (241.41, 157.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1303)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 513, 590)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 888)\n",
      "[M::mem_pestat] mean and std.dev: (514.44, 116.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1037)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66612 reads in 9.202 CPU sec, 9.058 real sec\n",
      "[M::process] read 66604 sequences (10000129 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 24337, 3, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 212, 392)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 944)\n",
      "[M::mem_pestat] mean and std.dev: (219.54, 120.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1220)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (440, 512, 589)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (142, 887)\n",
      "[M::mem_pestat] mean and std.dev: (512.83, 115.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1036)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66606 reads in 9.143 CPU sec, 8.981 real sec\n",
      "[M::process] read 66584 sequences (10000215 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 24318, 4, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (182, 213, 349)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 683)\n",
      "[M::mem_pestat] mean and std.dev: (231.10, 125.19)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 850)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 514, 590)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 888)\n",
      "[M::mem_pestat] mean and std.dev: (514.61, 114.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1037)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66604 reads in 9.075 CPU sec, 8.916 real sec\n",
      "[M::process] read 66590 sequences (10000210 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 24394, 2, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (440, 513, 590)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (140, 890)\n",
      "[M::mem_pestat] mean and std.dev: (514.05, 115.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1040)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66584 reads in 9.235 CPU sec, 9.070 real sec\n",
      "[M::process] read 66590 sequences (10000172 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 24352, 3, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 197, 389)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 935)\n",
      "[M::mem_pestat] mean and std.dev: (202.75, 106.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1208)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 513, 589)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (145, 885)\n",
      "[M::mem_pestat] mean and std.dev: (514.23, 114.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1033)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66590 reads in 9.269 CPU sec, 9.107 real sec\n",
      "[M::process] read 66606 sequences (10000245 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 24276, 1, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (105, 142, 216)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 438)\n",
      "[M::mem_pestat] mean and std.dev: (168.21, 84.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 549)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (442, 514, 590)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 886)\n",
      "[M::mem_pestat] mean and std.dev: (515.55, 114.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1034)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66590 reads in 9.093 CPU sec, 8.932 real sec\n",
      "[M::process] read 66612 sequences (10000152 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 24350, 3, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (132, 305, 376)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 864)\n",
      "[M::mem_pestat] mean and std.dev: (234.33, 146.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1108)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (440, 514, 589)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (142, 887)\n",
      "[M::mem_pestat] mean and std.dev: (513.86, 115.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1036)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66606 reads in 9.555 CPU sec, 9.394 real sec\n",
      "[M::process] read 66628 sequences (10000043 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (20, 24323, 1, 9)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (142, 281, 430)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1006)\n",
      "[M::mem_pestat] mean and std.dev: (264.68, 172.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1294)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (442, 515, 590)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 886)\n",
      "[M::mem_pestat] mean and std.dev: (515.60, 114.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1034)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66612 reads in 9.255 CPU sec, 9.096 real sec\n",
      "[M::process] read 66634 sequences (10000283 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 24350, 5, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 350, 595)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1383)\n",
      "[M::mem_pestat] mean and std.dev: (285.00, 175.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1777)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (442, 515, 591)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 889)\n",
      "[M::mem_pestat] mean and std.dev: (515.75, 115.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1038)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66628 reads in 9.186 CPU sec, 9.028 real sec\n",
      "[M::process] read 66592 sequences (10000092 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 24237, 4, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (175, 270, 450)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1000)\n",
      "[M::mem_pestat] mean and std.dev: (265.40, 127.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1275)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 515, 591)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (141, 891)\n",
      "[M::mem_pestat] mean and std.dev: (515.74, 114.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1041)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66634 reads in 12.772 CPU sec, 12.613 real sec\n",
      "[M::process] read 66600 sequences (10000135 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 24271, 3, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (153, 189, 293)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 573)\n",
      "[M::mem_pestat] mean and std.dev: (232.82, 128.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 748)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (442, 514, 590)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 886)\n",
      "[M::mem_pestat] mean and std.dev: (515.62, 114.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1034)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66592 reads in 8.561 CPU sec, 8.404 real sec\n",
      "[M::process] read 66610 sequences (10000108 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 24430, 4, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (155, 257, 383)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 839)\n",
      "[M::mem_pestat] mean and std.dev: (252.44, 115.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1067)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 513, 590)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 888)\n",
      "[M::mem_pestat] mean and std.dev: (514.49, 115.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1037)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66600 reads in 8.623 CPU sec, 8.468 real sec\n",
      "[M::process] read 66598 sequences (10000055 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 24278, 5, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (151, 236, 402)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 904)\n",
      "[M::mem_pestat] mean and std.dev: (266.25, 129.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1155)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (440, 514, 590)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (140, 890)\n",
      "[M::mem_pestat] mean and std.dev: (514.18, 116.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1040)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66610 reads in 8.626 CPU sec, 8.471 real sec\n",
      "[M::process] read 66594 sequences (10000130 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 24281, 3, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (217, 330, 415)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 811)\n",
      "[M::mem_pestat] mean and std.dev: (279.36, 154.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1009)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (440, 511, 588)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 884)\n",
      "[M::mem_pestat] mean and std.dev: (512.60, 114.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1032)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66598 reads in 8.777 CPU sec, 8.625 real sec\n",
      "[M::process] read 66592 sequences (10000262 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (20, 24315, 6, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (190, 271, 340)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 640)\n",
      "[M::mem_pestat] mean and std.dev: (253.30, 122.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 790)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (442, 514, 590)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 886)\n",
      "[M::mem_pestat] mean and std.dev: (514.44, 114.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1034)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66594 reads in 8.661 CPU sec, 8.506 real sec\n",
      "[M::process] read 66612 sequences (10000298 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 24351, 2, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (107, 205, 454)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1148)\n",
      "[M::mem_pestat] mean and std.dev: (251.81, 156.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1495)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (440, 513, 589)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (142, 887)\n",
      "[M::mem_pestat] mean and std.dev: (513.13, 115.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1036)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66592 reads in 8.707 CPU sec, 8.552 real sec\n",
      "[M::process] read 66600 sequences (10000011 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 24329, 4, 7)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (135, 377, 494)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1212)\n",
      "[M::mem_pestat] mean and std.dev: (319.31, 189.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1571)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (442, 513, 589)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 883)\n",
      "[M::mem_pestat] mean and std.dev: (514.29, 114.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1030)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66612 reads in 8.770 CPU sec, 8.611 real sec\n",
      "[M::process] read 66598 sequences (10000166 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 24343, 5, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (89, 112, 499)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1319)\n",
      "[M::mem_pestat] mean and std.dev: (188.27, 167.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1729)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (441, 513, 587)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (149, 879)\n",
      "[M::mem_pestat] mean and std.dev: (512.66, 113.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (3, 1025)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66600 reads in 8.745 CPU sec, 8.586 real sec\n",
      "[M::process] read 66594 sequences (10000087 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 24135, 2, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (75, 236, 332)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 846)\n",
      "[M::mem_pestat] mean and std.dev: (227.29, 142.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1103)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] (25, 50, 75) percentile: (442, 513, 588)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 880)\n",
      "[M::mem_pestat] mean and std.dev: (513.58, 112.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (4, 1026)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66598 reads in 8.841 CPU sec, 8.686 real sec\n",
      "[M::process] read 66604 sequences (10000077 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 24354, 1, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (165, 238, 407)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 891)\n",
      "[M::mem_pestat] mean and std.dev: (263.30, 112.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1133)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (440, 511, 587)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 881)\n",
      "[M::mem_pestat] mean and std.dev: (511.50, 113.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1028)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66594 reads in 8.711 CPU sec, 8.552 real sec\n",
      "[M::process] read 66568 sequences (10000257 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 24217, 2, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (162, 296, 375)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 801)\n",
      "[M::mem_pestat] mean and std.dev: (273.80, 158.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1014)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (440, 511, 585)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 875)\n",
      "[M::mem_pestat] mean and std.dev: (512.19, 111.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (5, 1020)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66604 reads in 8.861 CPU sec, 8.709 real sec\n",
      "[M::process] read 66590 sequences (10000001 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 24438, 1, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (94, 155, 266)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 610)\n",
      "[M::mem_pestat] mean and std.dev: (181.23, 112.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 782)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (442, 512, 587)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (152, 877)\n",
      "[M::mem_pestat] mean and std.dev: (513.93, 113.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (7, 1022)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66568 reads in 8.777 CPU sec, 8.619 real sec\n",
      "[M::process] read 2130 sequences (319671 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 24420, 0, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (135, 230, 393)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 909)\n",
      "[M::mem_pestat] mean and std.dev: (207.20, 118.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1167)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (442, 513, 588)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 880)\n",
      "[M::mem_pestat] mean and std.dev: (513.73, 112.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (4, 1026)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 66590 reads in 8.721 CPU sec, 8.673 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 788, 1, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (436, 510, 593)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (122, 907)\n",
      "[M::mem_pestat] mean and std.dev: (512.71, 114.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1064)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 2130 reads in 0.324 CPU sec, 0.292 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem data/assembly/scaffolds.fasta data/reads/L2.R1.fastq.gz data/reads/L2.R2.fastq.gz\n",
      "[main] Real time: 269.004 sec; CPU: 273.515 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 66528 sequences (10000159 bp)...\n",
      "[M::process] read 66552 sequences (10000238 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 24043, 1, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (450, 527, 606)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (138, 918)\n",
      "[M::mem_pestat] mean and std.dev: (525.19, 120.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1074)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66528 reads in 8.913 CPU sec, 8.790 real sec\n",
      "[M::process] read 66540 sequences (10000056 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 24019, 2, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (449, 525, 604)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (139, 914)\n",
      "[M::mem_pestat] mean and std.dev: (523.96, 120.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1069)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66552 reads in 8.845 CPU sec, 8.690 real sec\n",
      "[M::process] read 66558 sequences (10000287 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 24023, 4, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (448, 524, 600)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 904)\n",
      "[M::mem_pestat] mean and std.dev: (522.30, 117.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1056)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66540 reads in 8.857 CPU sec, 8.698 real sec\n",
      "[M::process] read 66556 sequences (10000112 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 23785, 5, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (450, 523, 601)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 903)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] mean and std.dev: (523.30, 117.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1054)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66558 reads in 8.991 CPU sec, 8.844 real sec\n",
      "[M::process] read 66570 sequences (10000269 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 23804, 4, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 522, 598)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (145, 900)\n",
      "[M::mem_pestat] mean and std.dev: (520.98, 117.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1051)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66556 reads in 9.097 CPU sec, 8.941 real sec\n",
      "[M::process] read 66564 sequences (10000270 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 23876, 4, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (444, 521, 598)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (136, 906)\n",
      "[M::mem_pestat] mean and std.dev: (519.03, 119.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1060)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66570 reads in 8.900 CPU sec, 8.746 real sec\n",
      "[M::process] read 66596 sequences (10000235 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 23903, 5, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 522, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 903)\n",
      "[M::mem_pestat] mean and std.dev: (520.32, 117.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1055)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66564 reads in 8.834 CPU sec, 8.679 real sec\n",
      "[M::process] read 66568 sequences (10000250 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 23881, 1, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (446, 521, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (140, 905)\n",
      "[M::mem_pestat] mean and std.dev: (520.10, 118.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1058)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66596 reads in 8.996 CPU sec, 8.840 real sec\n",
      "[M::process] read 66584 sequences (10000044 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 23934, 1, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (448, 522, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 901)\n",
      "[M::mem_pestat] mean and std.dev: (520.74, 117.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1052)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66568 reads in 9.034 CPU sec, 8.878 real sec\n",
      "[M::process] read 66594 sequences (10000278 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 23900, 2, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (446, 522, 600)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (138, 908)\n",
      "[M::mem_pestat] mean and std.dev: (519.96, 118.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1062)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66584 reads in 8.946 CPU sec, 8.789 real sec\n",
      "[M::process] read 66596 sequences (10000197 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 23708, 6, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (446, 522, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (140, 905)\n",
      "[M::mem_pestat] mean and std.dev: (520.47, 118.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1058)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66594 reads in 9.100 CPU sec, 8.949 real sec\n",
      "[M::process] read 66572 sequences (10000202 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 23797, 2, 9)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (448, 524, 600)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 904)\n",
      "[M::mem_pestat] mean and std.dev: (521.91, 118.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1056)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66596 reads in 8.958 CPU sec, 8.802 real sec\n",
      "[M::process] read 66578 sequences (10000257 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 23780, 2, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 522, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 903)\n",
      "[M::mem_pestat] mean and std.dev: (521.12, 118.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1055)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66572 reads in 8.958 CPU sec, 8.803 real sec\n",
      "[M::process] read 66586 sequences (10000232 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 23865, 2, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (448, 525, 602)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (140, 910)\n",
      "[M::mem_pestat] mean and std.dev: (523.02, 118.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1064)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66578 reads in 9.066 CPU sec, 8.910 real sec\n",
      "[M::process] read 66602 sequences (10000001 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 23841, 3, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 524, 600)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (141, 906)\n",
      "[M::mem_pestat] mean and std.dev: (521.64, 119.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1059)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66586 reads in 8.984 CPU sec, 8.828 real sec\n",
      "[M::process] read 66606 sequences (10000143 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 23722, 3, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 522, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 903)\n",
      "[M::mem_pestat] mean and std.dev: (520.82, 118.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1055)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66602 reads in 9.177 CPU sec, 9.024 real sec\n",
      "[M::process] read 66594 sequences (10000045 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 24024, 2, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (446, 521, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (140, 905)\n",
      "[M::mem_pestat] mean and std.dev: (520.21, 118.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1058)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66606 reads in 9.372 CPU sec, 9.217 real sec\n",
      "[M::process] read 66588 sequences (10000040 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 23699, 0, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 522, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 903)\n",
      "[M::mem_pestat] mean and std.dev: (520.69, 118.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1055)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66594 reads in 9.517 CPU sec, 9.363 real sec\n",
      "[M::process] read 66596 sequences (10000161 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 23914, 3, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 523, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 903)\n",
      "[M::mem_pestat] mean and std.dev: (520.21, 118.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1055)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66588 reads in 8.957 CPU sec, 8.802 real sec\n",
      "[M::process] read 66578 sequences (10000006 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 23716, 1, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 523, 601)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (139, 909)\n",
      "[M::mem_pestat] mean and std.dev: (521.74, 118.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1063)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66596 reads in 9.017 CPU sec, 8.861 real sec\n",
      "[M::process] read 66606 sequences (10000203 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 23779, 3, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (446, 521, 600)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (138, 908)\n",
      "[M::mem_pestat] mean and std.dev: (520.66, 118.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1062)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66578 reads in 8.928 CPU sec, 8.774 real sec\n",
      "[M::process] read 66592 sequences (10000181 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 23888, 1, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (449, 524, 600)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (147, 902)\n",
      "[M::mem_pestat] mean and std.dev: (522.70, 116.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1053)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66606 reads in 8.973 CPU sec, 8.817 real sec\n",
      "[M::process] read 66578 sequences (10000047 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 23750, 3, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (449, 524, 600)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (147, 902)\n",
      "[M::mem_pestat] mean and std.dev: (522.45, 117.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1053)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66592 reads in 9.010 CPU sec, 8.854 real sec\n",
      "[M::process] read 66576 sequences (10000284 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 23900, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (448, 522, 600)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 904)\n",
      "[M::mem_pestat] mean and std.dev: (521.53, 117.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1056)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66578 reads in 8.954 CPU sec, 8.798 real sec\n",
      "[M::process] read 66604 sequences (10000263 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 24079, 4, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (448, 522, 598)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 898)\n",
      "[M::mem_pestat] mean and std.dev: (520.36, 117.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1048)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66576 reads in 8.990 CPU sec, 8.846 real sec\n",
      "[M::process] read 66586 sequences (10000027 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 23938, 5, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (445, 521, 599)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (137, 907)\n",
      "[M::mem_pestat] mean and std.dev: (519.64, 118.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1061)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] (25, 50, 75) percentile: (4, 6, 11)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 25)\n",
      "[M::mem_pestat] mean and std.dev: (5.38, 2.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 32)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66604 reads in 8.926 CPU sec, 8.769 real sec\n",
      "[M::process] read 66584 sequences (10000136 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 23800, 2, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (446, 521, 597)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 899)\n",
      "[M::mem_pestat] mean and std.dev: (518.50, 117.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1050)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66586 reads in 9.010 CPU sec, 8.851 real sec\n",
      "[M::process] read 66576 sequences (10000063 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 23772, 6, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 521, 596)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (149, 894)\n",
      "[M::mem_pestat] mean and std.dev: (518.68, 116.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1043)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66584 reads in 9.314 CPU sec, 9.155 real sec\n",
      "[M::process] read 66588 sequences (10000101 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 23833, 0, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (445, 519, 594)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (147, 892)\n",
      "[M::mem_pestat] mean and std.dev: (517.07, 116.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1041)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66576 reads in 9.347 CPU sec, 9.186 real sec\n",
      "[M::process] read 2604 sequences (390762 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 23810, 3, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 521, 596)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (149, 894)\n",
      "[M::mem_pestat] mean and std.dev: (518.78, 115.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1043)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66588 reads in 9.150 CPU sec, 9.099 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 960, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (440, 522, 596)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (128, 908)\n",
      "[M::mem_pestat] mean and std.dev: (515.49, 116.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1064)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 2604 reads in 0.404 CPU sec, 0.362 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem data/assembly/scaffolds.fasta data/reads/L3.R1.fastq.gz data/reads/L3.R2.fastq.gz\n",
      "[main] Real time: 267.236 sec; CPU: 271.757 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 66556 sequences (10000032 bp)...\n",
      "[M::process] read 66566 sequences (10000081 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 16862, 7, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (456, 530, 611)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 921)\n",
      "[M::mem_pestat] mean and std.dev: (531.82, 119.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1076)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66556 reads in 10.234 CPU sec, 10.120 real sec\n",
      "[M::process] read 66618 sequences (10000274 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 16672, 4, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 530, 611)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (140, 925)\n",
      "[M::mem_pestat] mean and std.dev: (531.18, 119.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1082)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66566 reads in 10.270 CPU sec, 10.112 real sec\n",
      "[M::process] read 66614 sequences (10000288 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16630, 4, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 529, 608)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 916)\n",
      "[M::mem_pestat] mean and std.dev: (530.70, 118.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1070)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66618 reads in 10.196 CPU sec, 10.039 real sec\n",
      "[M::process] read 66660 sequences (10000127 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 16585, 6, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 528, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 913)\n",
      "[M::mem_pestat] mean and std.dev: (528.45, 117.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5, 6, 10)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 20)\n",
      "[M::mem_pestat] mean and std.dev: (5.91, 2.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 25)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66614 reads in 10.257 CPU sec, 10.100 real sec\n",
      "[M::process] read 66662 sequences (10000215 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16717, 6, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 530, 608)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 918)\n",
      "[M::mem_pestat] mean and std.dev: (529.20, 118.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1073)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66660 reads in 10.291 CPU sec, 10.133 real sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::process] read 66688 sequences (10000147 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 16365, 6, 13)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 528, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (145, 915)\n",
      "[M::mem_pestat] mean and std.dev: (528.76, 118.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1069)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5, 6, 331)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 983)\n",
      "[M::mem_pestat] mean and std.dev: (35.09, 93.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1309)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66662 reads in 10.375 CPU sec, 10.219 real sec\n",
      "[M::process] read 66660 sequences (10000202 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 16484, 8, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 529, 608)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 916)\n",
      "[M::mem_pestat] mean and std.dev: (529.08, 118.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1070)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66688 reads in 10.367 CPU sec, 10.222 real sec\n",
      "[M::process] read 66676 sequences (10000252 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16462, 7, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 529, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (141, 921)\n",
      "[M::mem_pestat] mean and std.dev: (529.46, 119.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1077)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66660 reads in 10.380 CPU sec, 10.223 real sec\n",
      "[M::process] read 66696 sequences (10000199 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 16575, 13, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (455, 529, 606)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (153, 908)\n",
      "[M::mem_pestat] mean and std.dev: (529.52, 116.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (2, 1059)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (2141, 4238, 6139)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14135)\n",
      "[M::mem_pestat] mean and std.dev: (3585.15, 2123.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18133)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (3, 5, 7)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15)\n",
      "[M::mem_pestat] mean and std.dev: (4.44, 2.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 19)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66676 reads in 10.325 CPU sec, 10.168 real sec\n",
      "[M::process] read 66706 sequences (10000232 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 16436, 15, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 529, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (141, 921)\n",
      "[M::mem_pestat] mean and std.dev: (529.14, 119.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1077)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (174, 1773, 3950)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11502)\n",
      "[M::mem_pestat] mean and std.dev: (2062.87, 2040.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 15278)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66696 reads in 10.318 CPU sec, 10.164 real sec\n",
      "[M::process] read 66718 sequences (10000136 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16604, 9, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (456, 532, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 915)\n",
      "[M::mem_pestat] mean and std.dev: (531.05, 116.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1068)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66706 reads in 10.237 CPU sec, 10.082 real sec\n",
      "[M::process] read 66672 sequences (10000267 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 16434, 8, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (450, 526, 604)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (142, 912)\n",
      "[M::mem_pestat] mean and std.dev: (525.82, 118.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66718 reads in 10.346 CPU sec, 10.191 real sec\n",
      "[M::process] read 66716 sequences (10000204 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 16490, 3, 9)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 528, 608)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 918)\n",
      "[M::mem_pestat] mean and std.dev: (528.47, 119.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1073)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66672 reads in 10.385 CPU sec, 10.229 real sec\n",
      "[M::process] read 66726 sequences (10000274 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 16512, 14, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 528, 606)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 910)\n",
      "[M::mem_pestat] mean and std.dev: (528.54, 117.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1062)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (292, 3950, 4408)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12640)\n",
      "[M::mem_pestat] mean and std.dev: (2702.71, 2460.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16756)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66716 reads in 10.340 CPU sec, 10.183 real sec\n",
      "[M::process] read 66704 sequences (10000122 bp)...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 16635, 5, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (451, 527, 605)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 913)\n",
      "[M::mem_pestat] mean and std.dev: (527.63, 118.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1067)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66726 reads in 10.269 CPU sec, 10.114 real sec\n",
      "[M::process] read 66704 sequences (10000251 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 16390, 5, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (456, 528, 606)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (156, 906)\n",
      "[M::mem_pestat] mean and std.dev: (529.16, 115.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (6, 1056)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66704 reads in 10.348 CPU sec, 10.193 real sec\n",
      "[M::process] read 66704 sequences (10000161 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16593, 7, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (455, 531, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (151, 911)\n",
      "[M::mem_pestat] mean and std.dev: (530.05, 117.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1063)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66704 reads in 10.402 CPU sec, 10.259 real sec\n",
      "[M::process] read 66702 sequences (10000279 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16690, 5, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (452, 528, 606)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 914)\n",
      "[M::mem_pestat] mean and std.dev: (528.41, 117.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1068)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66704 reads in 10.351 CPU sec, 10.195 real sec\n",
      "[M::process] read 66708 sequences (10000184 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 16404, 6, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 528, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (145, 915)\n",
      "[M::mem_pestat] mean and std.dev: (528.75, 117.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1069)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66702 reads in 10.483 CPU sec, 10.331 real sec\n",
      "[M::process] read 66728 sequences (10000282 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 16590, 5, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 527, 606)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 910)\n",
      "[M::mem_pestat] mean and std.dev: (528.60, 117.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1062)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66708 reads in 10.392 CPU sec, 10.238 real sec\n",
      "[M::process] read 66688 sequences (10000273 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16770, 11, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (452, 528, 606)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 914)\n",
      "[M::mem_pestat] mean and std.dev: (528.29, 117.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1068)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1762, 2203, 4700)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10576)\n",
      "[M::mem_pestat] mean and std.dev: (2889.73, 2149.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13514)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66728 reads in 10.332 CPU sec, 10.179 real sec\n",
      "[M::process] read 66692 sequences (10000209 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 16512, 7, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 528, 605)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (149, 909)\n",
      "[M::mem_pestat] mean and std.dev: (527.72, 117.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1061)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66688 reads in 10.389 CPU sec, 10.234 real sec\n",
      "[M::process] read 66700 sequences (10000065 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 16638, 5, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 528, 606)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (147, 912)\n",
      "[M::mem_pestat] mean and std.dev: (527.98, 117.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1065)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66692 reads in 10.313 CPU sec, 10.159 real sec\n",
      "[M::process] read 66686 sequences (10000257 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16687, 2, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (452, 528, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (142, 917)\n",
      "[M::mem_pestat] mean and std.dev: (528.34, 118.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1072)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66700 reads in 10.349 CPU sec, 10.194 real sec\n",
      "[M::process] read 66682 sequences (10000226 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 16628, 6, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 528, 606)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (147, 912)\n",
      "[M::mem_pestat] mean and std.dev: (528.41, 117.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1065)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66686 reads in 10.329 CPU sec, 10.176 real sec\n",
      "[M::process] read 66680 sequences (10000183 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 16534, 8, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (450, 526, 604)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (142, 912)\n",
      "[M::mem_pestat] mean and std.dev: (526.57, 117.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66682 reads in 10.481 CPU sec, 10.325 real sec\n",
      "[M::process] read 66682 sequences (10000021 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 16727, 3, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (451, 526, 604)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (145, 910)\n",
      "[M::mem_pestat] mean and std.dev: (526.27, 117.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1063)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66680 reads in 11.102 CPU sec, 10.939 real sec\n",
      "[M::process] read 66686 sequences (10000104 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16610, 12, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (451, 525, 602)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (149, 904)\n",
      "[M::mem_pestat] mean and std.dev: (525.03, 115.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1055)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (206, 4481, 6235)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 18293)\n",
      "[M::mem_pestat] mean and std.dev: (3432.50, 2567.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 24322)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66682 reads in 11.120 CPU sec, 10.980 real sec\n",
      "[M::process] read 66320 sequences (9949919 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16397, 10, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (452, 526, 602)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (152, 902)\n",
      "[M::mem_pestat] mean and std.dev: (525.95, 114.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (2, 1052)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (256, 4232, 4562)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13174)\n",
      "[M::mem_pestat] mean and std.dev: (2432.20, 2222.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 17480)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66686 reads in 11.019 CPU sec, 10.864 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 16462, 5, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (452, 525, 604)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 908)\n",
      "[M::mem_pestat] mean and std.dev: (525.89, 115.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1060)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66320 reads in 10.407 CPU sec, 10.368 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem data/assembly/scaffolds.fasta data/reads/P1.R1.fastq.gz data/reads/P1.R2.fastq.gz\n",
      "[main] Real time: 308.526 sec; CPU: 312.671 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 66554 sequences (10000007 bp)...\n",
      "[M::process] read 66592 sequences (10000066 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 18261, 6, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 532, 612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (138, 928)\n",
      "[M::mem_pestat] mean and std.dev: (530.77, 121.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1086)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66554 reads in 9.599 CPU sec, 9.476 real sec\n",
      "[M::process] read 66618 sequences (10000153 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 18207, 6, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 531, 611)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (140, 925)\n",
      "[M::mem_pestat] mean and std.dev: (529.76, 121.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1082)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66592 reads in 9.512 CPU sec, 9.356 real sec\n",
      "[M::process] read 66614 sequences (10000052 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 18408, 10, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (457, 534, 610)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (151, 916)\n",
      "[M::mem_pestat] mean and std.dev: (531.90, 119.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1069)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (93, 1233, 1973)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5733)\n",
      "[M::mem_pestat] mean and std.dev: (1271.70, 1404.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7613)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66618 reads in 9.514 CPU sec, 9.358 real sec\n",
      "[M::process] read 66644 sequences (10000248 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 18325, 13, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (456, 533, 610)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 918)\n",
      "[M::mem_pestat] mean and std.dev: (530.22, 119.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1072)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (149, 637, 4168)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12206)\n",
      "[M::mem_pestat] mean and std.dev: (1803.31, 2277.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16225)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66614 reads in 9.565 CPU sec, 9.409 real sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::process] read 66686 sequences (10000281 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 18206, 15, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (456, 532, 608)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (152, 912)\n",
      "[M::mem_pestat] mean and std.dev: (530.61, 118.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1064)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 445, 3103)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8895)\n",
      "[M::mem_pestat] mean and std.dev: (1311.60, 1533.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11791)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66644 reads in 9.608 CPU sec, 9.452 real sec\n",
      "[M::process] read 66686 sequences (10000241 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 18237, 10, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (456, 533, 611)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 921)\n",
      "[M::mem_pestat] mean and std.dev: (531.00, 121.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1076)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (103, 3459, 3979)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11731)\n",
      "[M::mem_pestat] mean and std.dev: (2432.40, 2090.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 15607)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66686 reads in 9.521 CPU sec, 9.368 real sec\n",
      "[M::process] read 66712 sequences (10000266 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 18300, 4, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (458, 533, 612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 920)\n",
      "[M::mem_pestat] mean and std.dev: (533.16, 118.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1074)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66686 reads in 9.618 CPU sec, 9.464 real sec\n",
      "[M::process] read 66702 sequences (10000078 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 18130, 9, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 533, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 919)\n",
      "[M::mem_pestat] mean and std.dev: (530.11, 120.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1074)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66712 reads in 9.579 CPU sec, 9.426 real sec\n",
      "[M::process] read 66730 sequences (10000112 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 18274, 8, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (455, 532, 610)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (145, 920)\n",
      "[M::mem_pestat] mean and std.dev: (530.59, 119.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1075)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66702 reads in 9.584 CPU sec, 9.431 real sec\n",
      "[M::process] read 66686 sequences (10000095 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 18286, 6, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 531, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 913)\n",
      "[M::mem_pestat] mean and std.dev: (528.24, 118.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66730 reads in 9.537 CPU sec, 9.381 real sec\n",
      "[M::process] read 66732 sequences (10000248 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 18172, 8, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 530, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 913)\n",
      "[M::mem_pestat] mean and std.dev: (528.24, 119.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66686 reads in 9.574 CPU sec, 9.423 real sec\n",
      "[M::process] read 66726 sequences (10000094 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 18258, 8, 11)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 531, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 913)\n",
      "[M::mem_pestat] mean and std.dev: (529.07, 119.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (8, 8, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 452)\n",
      "[M::mem_pestat] mean and std.dev: (52.50, 99.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 600)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66732 reads in 9.573 CPU sec, 9.418 real sec\n",
      "[M::process] read 66702 sequences (10000251 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 18169, 9, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 531, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (141, 921)\n",
      "[M::mem_pestat] mean and std.dev: (528.68, 121.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1077)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66726 reads in 9.765 CPU sec, 9.618 real sec\n",
      "[M::process] read 66684 sequences (10000111 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 18303, 9, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (456, 532, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 915)\n",
      "[M::mem_pestat] mean and std.dev: (530.38, 119.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1068)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 66702 reads in 9.568 CPU sec, 9.411 real sec\n",
      "[M::process] read 66696 sequences (10000093 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 18305, 7, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 530, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 913)\n",
      "[M::mem_pestat] mean and std.dev: (528.11, 119.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66684 reads in 9.591 CPU sec, 9.438 real sec\n",
      "[M::process] read 66692 sequences (10000092 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 18194, 7, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (455, 532, 605)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (155, 905)\n",
      "[M::mem_pestat] mean and std.dev: (527.92, 117.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (5, 1055)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66696 reads in 9.650 CPU sec, 9.495 real sec\n",
      "[M::process] read 66708 sequences (10000173 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 18353, 9, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 530, 608)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 916)\n",
      "[M::mem_pestat] mean and std.dev: (528.25, 120.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1070)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66692 reads in 9.545 CPU sec, 9.390 real sec\n",
      "[M::process] read 66718 sequences (10000155 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 18314, 8, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (457, 534, 612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (147, 922)\n",
      "[M::mem_pestat] mean and std.dev: (532.64, 119.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1077)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66708 reads in 9.640 CPU sec, 9.482 real sec\n",
      "[M::process] read 66756 sequences (10000113 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 18228, 9, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (457, 534, 610)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (151, 916)\n",
      "[M::mem_pestat] mean and std.dev: (531.23, 119.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1069)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66718 reads in 9.541 CPU sec, 9.389 real sec\n",
      "[M::process] read 66726 sequences (10000225 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 18186, 5, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (455, 530, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (151, 911)\n",
      "[M::mem_pestat] mean and std.dev: (529.95, 116.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1063)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66756 reads in 9.611 CPU sec, 9.455 real sec\n",
      "[M::process] read 66696 sequences (10000157 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 18295, 6, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 532, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (145, 915)\n",
      "[M::mem_pestat] mean and std.dev: (528.26, 120.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1069)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5, 7, 597)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1781)\n",
      "[M::mem_pestat] mean and std.dev: (79.62, 195.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2373)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66726 reads in 9.545 CPU sec, 9.388 real sec\n",
      "[M::process] read 66744 sequences (10000200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 18407, 5, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 530, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (145, 915)\n",
      "[M::mem_pestat] mean and std.dev: (527.85, 119.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1069)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66696 reads in 9.515 CPU sec, 9.360 real sec\n",
      "[M::process] read 66706 sequences (10000269 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 18441, 7, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (456, 532, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 915)\n",
      "[M::mem_pestat] mean and std.dev: (529.83, 119.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1068)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66744 reads in 9.568 CPU sec, 9.411 real sec\n",
      "[M::process] read 66716 sequences (10000200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 18200, 8, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (455, 531, 608)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (149, 914)\n",
      "[M::mem_pestat] mean and std.dev: (529.85, 118.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1067)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66706 reads in 9.685 CPU sec, 9.529 real sec\n",
      "[M::process] read 66718 sequences (10000016 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 18319, 10, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 532, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 913)\n",
      "[M::mem_pestat] mean and std.dev: (529.17, 119.59)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (93, 889, 4609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13641)\n",
      "[M::mem_pestat] mean and std.dev: (1918.40, 2283.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18157)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66716 reads in 9.542 CPU sec, 9.387 real sec\n",
      "[M::process] read 66690 sequences (10000153 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 18286, 6, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (453, 530, 606)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (147, 912)\n",
      "[M::mem_pestat] mean and std.dev: (527.61, 117.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1065)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66718 reads in 9.568 CPU sec, 9.418 real sec\n",
      "[M::process] read 66702 sequences (10000222 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 18243, 6, 11)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (457, 531, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (157, 907)\n",
      "[M::mem_pestat] mean and std.dev: (530.01, 117.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (7, 1057)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (100, 243, 586)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1558)\n",
      "[M::mem_pestat] mean and std.dev: (220.44, 198.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2044)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66690 reads in 9.595 CPU sec, 9.441 real sec\n",
      "[M::process] read 66718 sequences (10000205 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 18247, 3, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 529, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 913)\n",
      "[M::mem_pestat] mean and std.dev: (527.77, 118.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5, 8, 13)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 29)\n",
      "[M::mem_pestat] mean and std.dev: (7.50, 3.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 37)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66702 reads in 9.664 CPU sec, 9.515 real sec\n",
      "[M::process] read 65946 sequences (9883488 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 18118, 6, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 530, 607)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 913)\n",
      "[M::mem_pestat] mean and std.dev: (527.62, 118.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66718 reads in 9.861 CPU sec, 9.704 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 18152, 7, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (451, 528, 605)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 913)\n",
      "[M::mem_pestat] mean and std.dev: (525.56, 117.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1067)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 65946 reads in 9.468 CPU sec, 9.423 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem data/assembly/scaffolds.fasta data/reads/P2.R1.fastq.gz data/reads/P2.R2.fastq.gz\n",
      "[main] Real time: 283.812 sec; CPU: 287.979 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 66632 sequences (10000229 bp)...\n",
      "[M::process] read 66630 sequences (10000134 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 17697, 6, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (459, 537, 612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (153, 918)\n",
      "[M::mem_pestat] mean and std.dev: (533.25, 118.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1071)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66632 reads in 9.939 CPU sec, 9.815 real sec\n",
      "[M::process] read 66668 sequences (10000273 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 17616, 7, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (460, 537, 611)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (158, 913)\n",
      "[M::mem_pestat] mean and std.dev: (532.93, 117.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (7, 1064)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66630 reads in 9.974 CPU sec, 9.820 real sec\n",
      "[M::process] read 66674 sequences (10000163 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 17561, 4, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (458, 536, 611)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (152, 917)\n",
      "[M::mem_pestat] mean and std.dev: (532.08, 119.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1070)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66668 reads in 9.989 CPU sec, 9.832 real sec\n",
      "[M::process] read 66722 sequences (10000027 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 17326, 5, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (460, 537, 613)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (154, 919)\n",
      "[M::mem_pestat] mean and std.dev: (533.12, 119.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1072)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66674 reads in 10.150 CPU sec, 9.990 real sec\n",
      "[M::process] read 66738 sequences (10000104 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 17199, 8, 9)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (459, 537, 613)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (151, 921)\n",
      "[M::mem_pestat] mean and std.dev: (533.26, 119.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1075)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66722 reads in 9.986 CPU sec, 9.826 real sec\n",
      "[M::process] read 66732 sequences (10000110 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 17366, 6, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (457, 534, 612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (147, 922)\n",
      "[M::mem_pestat] mean and std.dev: (530.64, 119.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1077)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66738 reads in 9.949 CPU sec, 9.793 real sec\n",
      "[M::process] read 66758 sequences (10000284 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 17290, 6, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (458, 538, 614)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (146, 926)\n",
      "[M::mem_pestat] mean and std.dev: (532.54, 120.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1082)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66732 reads in 10.060 CPU sec, 9.906 real sec\n",
      "[M::process] read 66734 sequences (10000041 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 17335, 8, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (458, 537, 612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 920)\n",
      "[M::mem_pestat] mean and std.dev: (533.05, 119.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1074)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66758 reads in 10.057 CPU sec, 9.901 real sec\n",
      "[M::process] read 66736 sequences (10000003 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 17482, 9, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (459, 538, 612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (153, 918)\n",
      "[M::mem_pestat] mean and std.dev: (532.81, 117.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1071)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66734 reads in 9.989 CPU sec, 9.834 real sec\n",
      "[M::process] read 66736 sequences (10000084 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 17351, 6, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (459, 536, 610)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (157, 912)\n",
      "[M::mem_pestat] mean and std.dev: (531.81, 118.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (6, 1063)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5, 6, 456)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1358)\n",
      "[M::mem_pestat] mean and std.dev: (78.10, 150.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1809)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66736 reads in 9.952 CPU sec, 9.796 real sec\n",
      "[M::process] read 66714 sequences (10000178 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 17585, 3, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (460, 538, 614)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (152, 922)\n",
      "[M::mem_pestat] mean and std.dev: (534.08, 119.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1076)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66736 reads in 10.042 CPU sec, 9.884 real sec\n",
      "[M::process] read 66746 sequences (10000238 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 17449, 4, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (460, 538, 614)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (152, 922)\n",
      "[M::mem_pestat] mean and std.dev: (534.11, 118.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1076)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66714 reads in 10.013 CPU sec, 9.854 real sec\n",
      "[M::process] read 66734 sequences (10000025 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 17633, 9, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (461, 538, 611)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (161, 911)\n",
      "[M::mem_pestat] mean and std.dev: (534.33, 117.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (11, 1061)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (4, 5, 7)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13)\n",
      "[M::mem_pestat] mean and std.dev: (4.62, 1.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66746 reads in 10.024 CPU sec, 9.870 real sec\n",
      "[M::process] read 66746 sequences (10000121 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 17598, 7, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (462, 540, 616)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (154, 924)\n",
      "[M::mem_pestat] mean and std.dev: (536.60, 119.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1078)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (6, 9, 3813)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11427)\n",
      "[M::mem_pestat] mean and std.dev: (1246.33, 2119.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 15234)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66734 reads in 10.030 CPU sec, 9.880 real sec\n",
      "[M::process] read 66728 sequences (10000019 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 17576, 11, 6)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (460, 537, 614)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (152, 922)\n",
      "[M::mem_pestat] mean and std.dev: (534.72, 119.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1076)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (3160, 4106, 5773)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10999)\n",
      "[M::mem_pestat] mean and std.dev: (3669.27, 1987.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13612)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66746 reads in 10.148 CPU sec, 9.993 real sec\n",
      "[M::process] read 66762 sequences (10000278 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 17403, 12, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (458, 535, 612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 920)\n",
      "[M::mem_pestat] mean and std.dev: (532.04, 118.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1074)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (240, 2188, 5096)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14808)\n",
      "[M::mem_pestat] mean and std.dev: (2312.92, 2079.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 19664)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66728 reads in 10.129 CPU sec, 9.974 real sec\n",
      "[M::process] read 66764 sequences (10000154 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 17606, 7, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (458, 536, 613)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (148, 923)\n",
      "[M::mem_pestat] mean and std.dev: (532.58, 119.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1078)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66762 reads in 9.993 CPU sec, 9.838 real sec\n",
      "[M::process] read 66720 sequences (10000212 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 17683, 5, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (461, 538, 614)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (155, 920)\n",
      "[M::mem_pestat] mean and std.dev: (535.18, 118.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (2, 1073)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66764 reads in 10.029 CPU sec, 9.874 real sec\n",
      "[M::process] read 66724 sequences (10000158 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 17622, 8, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (459, 536, 613)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (151, 921)\n",
      "[M::mem_pestat] mean and std.dev: (532.95, 120.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1075)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1, 6, 9)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 25)\n",
      "[M::mem_pestat] mean and std.dev: (4.38, 3.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 33)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 66720 reads in 10.052 CPU sec, 9.896 real sec\n",
      "[M::process] read 66738 sequences (10000088 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 17451, 4, 9)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (460, 536, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (162, 907)\n",
      "[M::mem_pestat] mean and std.dev: (532.46, 116.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (13, 1056)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66724 reads in 10.147 CPU sec, 9.996 real sec\n",
      "[M::process] read 66714 sequences (10000015 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 17442, 6, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (461, 538, 613)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (157, 917)\n",
      "[M::mem_pestat] mean and std.dev: (533.91, 117.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (5, 1069)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66738 reads in 10.087 CPU sec, 9.930 real sec\n",
      "[M::process] read 66702 sequences (10000031 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 17414, 7, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (458, 535, 612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (150, 920)\n",
      "[M::mem_pestat] mean and std.dev: (531.93, 118.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1074)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66714 reads in 10.024 CPU sec, 9.867 real sec\n",
      "[M::process] read 66714 sequences (10000167 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 17414, 11, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (457, 535, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (153, 913)\n",
      "[M::mem_pestat] mean and std.dev: (530.87, 117.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1065)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (365, 1149, 4541)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12893)\n",
      "[M::mem_pestat] mean and std.dev: (2249.45, 2330.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 17069)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66702 reads in 10.357 CPU sec, 10.199 real sec\n",
      "[M::process] read 66718 sequences (10000262 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 17389, 6, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (457, 534, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (153, 913)\n",
      "[M::mem_pestat] mean and std.dev: (529.89, 117.24)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1065)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66714 reads in 10.222 CPU sec, 10.066 real sec\n",
      "[M::process] read 66710 sequences (10000189 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 17398, 6, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (459, 535, 609)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (159, 909)\n",
      "[M::mem_pestat] mean and std.dev: (531.18, 116.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (9, 1059)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66718 reads in 10.133 CPU sec, 9.973 real sec\n",
      "[M::process] read 66724 sequences (10000239 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 17641, 6, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (458, 535, 610)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (154, 914)\n",
      "[M::mem_pestat] mean and std.dev: (532.07, 116.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (2, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66710 reads in 10.133 CPU sec, 9.973 real sec\n",
      "[M::process] read 66722 sequences (10000093 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 17495, 2, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (459, 536, 611)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (155, 915)\n",
      "[M::mem_pestat] mean and std.dev: (532.84, 116.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (3, 1067)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66724 reads in 10.198 CPU sec, 10.040 real sec\n",
      "[M::process] read 66708 sequences (10000136 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 17517, 10, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (454, 531, 603)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (156, 901)\n",
      "[M::mem_pestat] mean and std.dev: (525.70, 114.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (7, 1050)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (88, 363, 3138)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9238)\n",
      "[M::mem_pestat] mean and std.dev: (1709.40, 2197.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12288)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 66722 reads in 10.142 CPU sec, 9.982 real sec\n",
      "[M::process] read 65152 sequences (9763689 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 17427, 3, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (455, 532, 604)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (157, 902)\n",
      "[M::mem_pestat] mean and std.dev: (526.37, 114.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (8, 1051)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 66708 reads in 10.190 CPU sec, 10.043 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 17109, 9, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (455, 532, 602)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (161, 896)\n",
      "[M::mem_pestat] mean and std.dev: (526.37, 112.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (14, 1043)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 65152 reads in 9.829 CPU sec, 9.783 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem data/assembly/scaffolds.fasta data/reads/P3.R1.fastq.gz data/reads/P3.R2.fastq.gz\n",
      "[main] Real time: 298.032 sec; CPU: 302.234 sec\n"
     ]
    }
   ],
   "source": [
    "# first I define the samples in a variable called 'samples'\n",
    "samples=( L1 L2 L3 P1 P2 P3 )\n",
    "# next I use this variable in my loop\n",
    "for i in ${samples[@]}\n",
    "    do bwa mem data/assembly/scaffolds.fasta data/reads/$i.R1.fastq.gz data/reads/$i.R2.fastq.gz | samtools view -b -o data/mapped/$i.bam\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TIP: If you are working on your own computer and not some shared server, you can speed up the process substantially by using more threads (CPUs/cores) to do the mapping. \n",
    "Find the amount of cores available on your computer with `nproc` and read the bwa manual to use more threads.\n",
    "\n",
    "**[DO:] If the loop is running, then proceed to the next notebook.**\n",
    "Check with `ls` if your files are being created and perhaps if they increase in size over time.\n",
    "Then start preparing the next loop: sorting of the bam files."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Some background: \n",
    "> By default, mapping algorithms like bwa spit out a .sam file. The sam format is widely used and accepted by many different downstream programmes. Basically, it's just a big table with a standardised format. Although a sam file is human-readable, it is also rather bulky. The file size of a single SAM file will quickly exceed the disk space you have on this virtual machine. Therefore, we convert it to a BAM file, a Binary sAM file. There is no loss of information; it is just saved much more efficiently. Naturally, a binary file is not human readable. Always store your files as bam files. If you need to have a look in the bam file, you can view them using `samtools view` or any of the many downstream programmes available online."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
