{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genome annotation with Prokka\n",
    "\n",
    "Now that we some extra information about our bins, we can continue to analyze the high quality bins. The final CheckM results will give you a good overview of the bins with low contamination and high completeness and also shows the lowest taxonomic rank of the bin. Pick a bin that you think is interesting to further study. Alternativelly, you may also make a loop to annotate multiple bins.\n",
    "\n",
    "With this selected bin(s) we are going to do some genome annotation. Whole genome annotation is the process of identifying features of interest in a set of genomic DNA sequences, and labelling them with useful information. Prokka is a software tool to annotate bacterial, archaeal and viral genomes quickly and produce standards-compliant output files.\n",
    "\n",
    "<b>Assignment:</b><br>\n",
    "Have a look at the prokka help pages below and come up with the right command to run prokka. (HINT: look at the Usage tag to run prokka on default mode, think of threads, use `--centre X --compliant` to stop prokkas complaints about ugly contig names, direct the output towars data/prokka)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a directory for the prokka output:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "prokka help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!prokka -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run prokka:  This may take a while."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[15:17:46] This is prokka 1.12\n",
      "[15:17:46] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[15:17:46] Homepage is https://github.com/tseemann/prokka\n",
      "[15:17:46] Local time is Fri Jun  1 15:17:46 2018\n",
      "[15:17:46] You are laura\n",
      "[15:17:46] Operating system is linux\n",
      "[15:17:46] You have BioPerl 1.007002\n",
      "[15:17:46] System has 12 cores.\n",
      "[15:17:46] Will use maximum of 1 cores.\n",
      "[15:17:46] Annotating as >>> Bacteria <<<\n",
      "[15:17:46] Generating locus_tag from './data/bins/testlaura.8.fa' contents.\n",
      "[15:17:46] Setting --locustag FLIOPFNA from MD5 f5289f7ac62fb4311d084dce9eeb871c\n",
      "[15:17:46] Enabling options to ensure Genbank/ENA/DDJB submission compliance.\n",
      "[15:17:46] Creating new output folder: ./data/annotation/8/\n",
      "[15:17:46] Running: mkdir -p \\.\\/data\\/annotation\\/8\\/\n",
      "[15:17:46] Using filename prefix: PROKKA_06012018.XXX\n",
      "[15:17:46] Setting HMMER_NCPU=1\n",
      "[15:17:46] Writing log to: ./data/annotation/8//PROKKA_06012018.log\n",
      "[15:17:46] Command: /opt/prokka/bin/prokka ./data/bins/testlaura.8.fa --cpus 1 --outdir ./data/annotation/8/ --centre X --compliant\n",
      "[15:17:46] Appending to PATH: /opt/prokka/bin/../binaries/linux\n",
      "[15:17:46] Appending to PATH: /opt/prokka/bin/../binaries/linux/../common\n",
      "[15:17:46] Appending to PATH: /opt/prokka/bin\n",
      "[15:17:46] Looking for 'aragorn' - found /opt/prokka/bin/../binaries/linux/aragorn\n",
      "[15:17:46] Determined aragorn version is 1.2\n",
      "[15:17:46] Looking for 'blastp' - found /usr/bin/blastp\n",
      "[15:17:46] Determined blastp version is 2.6\n",
      "[15:17:46] Looking for 'cmpress' - found /usr/bin/cmpress\n",
      "[15:17:46] Determined cmpress version is 1.1\n",
      "[15:17:46] Looking for 'cmscan' - found /usr/bin/cmscan\n",
      "[15:17:46] Determined cmscan version is 1.1\n",
      "[15:17:46] Looking for 'egrep' - found /bin/egrep\n",
      "[15:17:46] Looking for 'find' - found /usr/bin/find\n",
      "[15:17:46] Looking for 'grep' - found /bin/grep\n",
      "[15:17:46] Looking for 'hmmpress' - found /usr/bin/hmmpress\n",
      "[15:17:46] Determined hmmpress version is 3.1\n",
      "[15:17:46] Looking for 'hmmscan' - found /usr/bin/hmmscan\n",
      "[15:17:46] Determined hmmscan version is 3.1\n",
      "[15:17:46] Looking for 'java' - found /usr/bin/java\n",
      "[15:17:46] Looking for 'less' - found /usr/bin/less\n",
      "[15:17:46] Looking for 'makeblastdb' - found /usr/bin/makeblastdb\n",
      "[15:17:46] Determined makeblastdb version is 2.6\n",
      "[15:17:46] Looking for 'minced' - found /opt/prokka/bin/../binaries/linux/../common/minced\n",
      "[15:17:47] Determined minced version is 2.0\n",
      "[15:17:47] Looking for 'parallel' - found /usr/bin/parallel\n",
      "[15:17:47] Determined parallel version is 20161222\n",
      "[15:17:47] Looking for 'prodigal' - found /usr/bin/prodigal\n",
      "[15:17:47] Determined prodigal version is 2.6\n",
      "[15:17:47] Looking for 'prokka-genbank_to_fasta_db' - found /opt/prokka/bin/prokka-genbank_to_fasta_db\n",
      "[15:17:47] Looking for 'rnammer' - found /usr/local/bin/rnammer\n",
      "[15:17:47] Determined rnammer version is 1.2\n",
      "[15:17:47] Looking for 'sed' - found /bin/sed\n",
      "[15:17:47] Looking for 'tbl2asn' - found /usr/bin/tbl2asn\n",
      "[15:17:47] Determined tbl2asn version is 25.3\n",
      "[15:17:47] Using genetic code table 11.\n",
      "[15:17:47] Loading and checking input file: ./data/bins/testlaura.8.fa\n",
      "[15:17:48] Wrote 35 contigs totalling 3710623 bp.\n",
      "[15:17:48] Predicting tRNAs and tmRNAs\n",
      "[15:17:48] Running: aragorn -l -gc11  -w \\.\\/data\\/annotation\\/8\\/\\/PROKKA_06012018\\.fna\n",
      "[15:17:51] 1 tRNA-Trp [240027,240102] 34 (cca)\n",
      "[15:17:51] 2 tRNA-Phe [257827,257902] 34 (gaa)\n",
      "[15:17:51] 3 tRNA-Met c[430606,430682] 35 (cat)\n",
      "[15:17:51] 1 tRNA-Arg c[235465,235539] 34 (cct)\n",
      "[15:17:51] 1 tRNA-Val [59825,59899] 33 (cac)\n",
      "[15:17:51] 1 tRNA-Thr [187169,187244] 34 (cgt)\n",
      "[15:17:51] 1 tRNA-Leu [3702,3786] 35 (gag)\n",
      "[15:17:51] 2 tRNA-Lys c[38705,38780] 34 (ctt)\n",
      "[15:17:51] 1 tRNA-Tyr [138392,138477] 35 (gta)\n",
      "[15:17:51] 2 tRNA-Gly [138543,138616] 33 (tcc)\n",
      "[15:17:51] 3 tRNA-Thr [138648,138722] 33 (ggt)\n",
      "[15:17:51] 4 tRNA-Arg [196637,196712] 34 (ccg)\n",
      "[15:17:51] 1 tRNA-Thr [63380,63455] 34 (tgt)\n",
      "[15:17:51] 2 tRNA-Pro c[74586,74662] 35 (cgg)\n",
      "[15:17:51] 3 tRNA-Ser c[183447,183536] 35 (cga)\n",
      "[15:17:51] 1 tRNA-Gly [11083,11156] 33 (ccc)\n",
      "[15:17:51] 2 tmRNA c[100473,100849] 91,123 ANDERFALAA*\n",
      "[15:17:51] 3 tRNA-SeC c[149732,149823] 35 (tca)\n",
      "[15:17:51] 1 tRNA-Glu [89,164] 35 (ttc)\n",
      "[15:17:51] 2 tRNA-Asp [257,333] 35 (gtc)\n",
      "[15:17:51] 1 tRNA-Ser [17475,17567] 35 (gct)\n",
      "[15:17:51] 2 tRNA-Leu [39867,39953] 35 (tag)\n",
      "[15:17:51] 3 tRNA-Gly [46152,46227] 34 (gcc)\n",
      "[15:17:51] 4 tRNA-Leu [46335,46428] 35 (taa)\n",
      "[15:17:51] 5 tRNA-Val [91216,91291] 34 (tac)\n",
      "[15:17:51] 6 tRNA-His [93344,93419] 34 (gtg)\n",
      "[15:17:51] 1 tRNA-Leu [55147,55231] 35 (caa)\n",
      "[15:17:51] 1 tRNA-Ser c[39410,39499] 35 (gga)\n",
      "[15:17:51] 2 tRNA-Asn [105762,105837] 34 (gtt)\n",
      "[15:17:51] 1 tRNA-Pro c[9703,9779] 35 (tgg)\n",
      "[15:17:51] 1 tRNA-Met c[25664,25740] 35 (cat)\n",
      "[15:17:51] 1 tRNA-Val [44167,44243] 35 (gac)\n",
      "[15:17:51] 2 tRNA-Leu c[49368,49452] 35 (cag)\n",
      "[15:17:51] 1 tRNA-Lys [55010,55085] 34 (ttt)\n",
      "[15:17:51] 1 tRNA-Arg [6177,6253] 35 (acg)\n",
      "[15:17:51] 2 tRNA-Ser c[31563,31650] 35 (tga)\n",
      "[15:17:51] 1 tRNA-Gln c[7441,7517] 35 (ttg)\n",
      "[15:17:51] 2 tRNA-Met [19301,19377] 35 (cat)\n",
      "[15:17:51] 1 tRNA-Pro c[417,493] 35 (ggg)\n",
      "[15:17:51] Found 39 tRNAs\n",
      "[15:17:51] Predicting Ribosomal RNAs\n",
      "[15:17:51] Running RNAmmer\n",
      "[15:17:51] Running: rnammer -S bac  -xml \\.\\/data\\/annotation\\/8\\/\\/rnammer\\.xml \\.\\/data\\/annotation\\/8\\/\\/PROKKA_06012018\\.fna\n",
      "[15:18:46] 1\t5S ribosomal RNA\tgnl|X|FLIOPFNA_24\t29065\t29176\t-\n",
      "[15:18:46] Deleting unwanted file: ./data/annotation/8//rnammer.xml\n",
      "[15:18:46] Found 1 rRNAs\n",
      "[15:18:46] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[15:18:46] Total of 39 tRNA + rRNA features\n",
      "[15:18:46] Searching for CRISPR repeats\n",
      "[15:18:48] Found 0 CRISPRs\n",
      "[15:18:48] Predicting coding sequences\n",
      "[15:18:48] Contigs total 3710623 bp, so using single mode\n",
      "[15:18:48] Running: prodigal -i \\.\\/data\\/annotation\\/8\\/\\/PROKKA_06012018\\.fna -c -m -g 11 -p single -f sco -q\n",
      "[15:18:59] Found 3369 CDS\n",
      "[15:18:59] Connecting features back to sequences\n",
      "[15:18:59] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[15:18:59] Annotating CDS, please be patient.\n",
      "[15:18:59] Will use 1 CPUs for similarity searching.\n",
      "[15:19:03] There are still 3369 unannotated CDS left (started with 3369)\n",
      "[15:19:03] Will use blast to search against /opt/prokka/bin/../db/kingdom/Bacteria/sprot with 1 CPUs\n",
      "[15:19:03] Running: cat \\.\\/data\\/annotation\\/8\\/\\/sprot\\.faa | parallel --gnu --plain -j 1 --block 564072 --recstart '>' --pipe blastp -query - -db /opt/prokka/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\.\\/data\\/annotation\\/8\\/\\/sprot\\.blast 2> /dev/null\n",
      "[15:28:23] Modify product: Uncharacterized oxidoreductase SA2266 => putative oxidoreductase\n",
      "[15:28:23] Modify product: Putative quercetin 2,3-dioxygenase Rv0181c => Putative quercetin 2,3-dioxygenase\n",
      "[15:28:23] Modify product: Probable chromosome-partitioning protein ParB => putative chromosome-partitioning protein ParB\n",
      "[15:28:23] Modify product: Probable TonB-dependent receptor BfrD => putative TonB-dependent receptor BfrD\n",
      "[15:28:23] Modify product: Uncharacterized FAD-linked oxidoreductase Rv2280 => putative FAD-linked oxidoreductase\n",
      "[15:28:23] Modify product: Probable type I restriction enzyme BthVORF4518P M protein => putative type I restriction enzymeP M protein\n",
      "[15:28:23] Modify product: Probable amino-acid metabolite efflux pump => putative amino-acid metabolite efflux pump\n",
      "[15:28:23] Modify product: Probable dual-specificity RNA methyltransferase RlmN => putative dual-specificity RNA methyltransferase RlmN\n",
      "[15:28:23] Modify product: Uncharacterized FAD-linked oxidoreductase Rv2280 => putative FAD-linked oxidoreductase\n",
      "[15:28:23] Modify product: Probable phospholipid ABC transporter permease protein MlaE => putative phospholipid ABC transporter permease protein MlaE\n",
      "[15:28:23] Modify product: Probable ribonucleotide transport ATP-binding protein mkl => putative ribonucleotide transport ATP-binding protein mkl\n",
      "[15:28:23] Modify product: Uncharacterized oxidoreductase Rv0769 => putative oxidoreductase\n",
      "[15:28:24] Modify product: Uncharacterized oxidoreductase YhhX => putative oxidoreductase YhhX\n",
      "[15:28:24] Modify product: Probable GTP-binding protein EngB => putative GTP-binding protein EngB\n",
      "[15:28:24] Modify product: Probable enoyl-CoA hydratase echA8 => putative enoyl-CoA hydratase echA8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[15:28:24] Modify product: Probable peptidoglycan glycosyltransferase FtsW => putative peptidoglycan glycosyltransferase FtsW\n",
      "[15:28:24] Modify product: Probable peptidoglycan D,D-transpeptidase PenA => putative peptidoglycan D,D-transpeptidase PenA\n",
      "[15:28:24] Modify product: Uncharacterized GTP-binding protein YjiA => putative GTP-binding protein YjiA\n",
      "[15:28:24] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:24] Modify product: Probable glutamine ABC transporter permease protein GlnM => putative glutamine ABC transporter permease protein GlnM\n",
      "[15:28:24] Modify product: Uncharacterized acyl-CoA thioester hydrolase HI_0827 => putative acyl-CoA thioester hydrolase\n",
      "[15:28:24] Modify product: Probable 3-hydroxyacyl-CoA dehydrogenase => putative 3-hydroxyacyl-CoA dehydrogenase\n",
      "[15:28:24] Modify product: Probable type I restriction enzyme BthVORF4518P M protein => putative type I restriction enzymeP M protein\n",
      "[15:28:24] Modify product: Uncharacterized HTH-type transcriptional regulator Rv1395 => putative HTH-type transcriptional regulator\n",
      "[15:28:25] Modify product: Probable thiol:disulfide interchange protein DsbC => putative thiol:disulfide interchange protein DsbC\n",
      "[15:28:25] Modify product: Probable tRNA-dihydrouridine synthase => putative tRNA-dihydrouridine synthase\n",
      "[15:28:25] Modify product: Uncharacterized HTH-type transcriptional regulator TtgW => putative HTH-type transcriptional regulator TtgW\n",
      "[15:28:25] Modify product: Probable cadmium-transporting ATPase => putative cadmium-transporting ATPase\n",
      "[15:28:25] Modify product: Solute-binding protein Veis_3954 => Solute-binding protein\n",
      "[15:28:25] Modify product: Solute-binding protein NAS141_03721 => Solute-binding protein\n",
      "[15:28:25] Modify product: Probable malonic semialdehyde reductase RutE => putative malonic semialdehyde reductase RutE\n",
      "[15:28:25] Modify product: Probable enoyl-CoA hydratase 1 => putative enoyl-CoA hydratase 1\n",
      "[15:28:25] Modify product: SCO1 protein homolog => SCO1 protein \n",
      "[15:28:25] Modify product: Cytochrome b561 homolog 2 => Cytochrome b561 \n",
      "[15:28:25] Modify product: S-adenosyl-L-methionine-binding protein RPA0152 => S-adenosyl-L-methionine-binding protein\n",
      "[15:28:25] Modify product: Uncharacterized zinc protease Rv2782c => putative zinc protease\n",
      "[15:28:25] Modify product: Uncharacterized zinc protease Rv2782c => putative zinc protease\n",
      "[15:28:26] Modify product: Probable ECF RNA polymerase sigma factor SigI => putative ECF RNA polymerase sigma factor SigI\n",
      "[15:28:26] Modify product: DNA-binding protein VF_0530 => DNA-binding protein\n",
      "[15:28:26] Modify product: Probable copper-transporting ATPase PacS => putative copper-transporting ATPase PacS\n",
      "[15:28:26] Modify product: Probable GTPase Rv1496 => putative GTPase\n",
      "[15:28:26] Modify product: Co-chaperone protein HscB homolog => Co-chaperone protein HscB \n",
      "[15:28:26] Modify product: Bifunctional protein FolD => Bifunctional protein FolD protein\n",
      "[15:28:26] Modify product: Probable sulfoacetate--CoA ligase => putative sulfoacetate--CoA ligase\n",
      "[15:28:26] Modify product: Probable ATP-dependent helicase DinG homolog => putative ATP-dependent helicase DinG \n",
      "[15:28:26] Modify product: Segregation and condensation protein B homolog => Segregation and condensation protein B \n",
      "[15:28:26] Modify product: HTH-type transcriptional regulator Rv2034 => HTH-type transcriptional regulator\n",
      "[15:28:26] Modify product: Probable diguanylate cyclase YdaM => putative diguanylate cyclase YdaM\n",
      "[15:28:26] Modify product: Probable chromate transport protein => putative chromate transport protein\n",
      "[15:28:26] Modify product: Uncharacterized ABC transporter ATP-binding protein YheS => putative ABC transporter ATP-binding protein YheS\n",
      "[15:28:26] Modify product: Uncharacterized protease YdcP => putative protease YdcP\n",
      "[15:28:26] Modify product: Probable transcriptional regulatory protein pdtaR => putative transcriptional regulatory protein pdtaR\n",
      "[15:28:26] Modify product: Probable 5-dehydro-4-deoxyglucarate dehydratase => putative 5-dehydro-4-deoxyglucarate dehydratase\n",
      "[15:28:27] Modify product: Probable amino-acid metabolite efflux pump => putative amino-acid metabolite efflux pump\n",
      "[15:28:27] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:27] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:27] Modify product: Uncharacterized ABC transporter ATP-binding protein YknY => putative ABC transporter ATP-binding protein YknY\n",
      "[15:28:27] Modify product: Probable glycerol-3-phosphate acyltransferase => putative glycerol-3-phosphate acyltransferase\n",
      "[15:28:27] Modify product: Uncharacterized oxidoreductase SA2266 => putative oxidoreductase\n",
      "[15:28:27] Modify product: Probable crotonobetaine/carnitine-CoA ligase => putative crotonobetaine/carnitine-CoA ligase\n",
      "[15:28:27] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:27] Modify product: Probable chromate transport protein => putative chromate transport protein\n",
      "[15:28:27] Modify product: Putative multidrug export ATP-binding/permease protein SA1683 => Putative multidrug export ATP-binding/permease protein\n",
      "[15:28:27] Modify product: Uncharacterized protease YhbU => putative protease YhbU\n",
      "[15:28:27] Modify product: Uncharacterized ABC transporter ATP-binding protein Rv0986 => putative ABC transporter ATP-binding protein\n",
      "[15:28:27] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:27] Modify product: Uncharacterized membrane protein Rv0364 => putative membrane protein\n",
      "[15:28:27] Modify product: Uncharacterized MFS-type transporter YhjX => putative MFS-type transporter YhjX\n",
      "[15:28:28] Modify product: Probable phosphatase Rv3661 => putative phosphatase\n",
      "[15:28:28] Modify product: Leu/Ile/Val-binding protein homolog 3 => Leu/Ile/Val-binding protein \n",
      "[15:28:28] Modify product: Ribonuclease TTHA0252 => Ribonuclease\n",
      "[15:28:28] Modify product: Uncharacterized oxidoreductase Rv1144 => putative oxidoreductase\n",
      "[15:28:28] Modify product: Uncharacterized HTH-type transcriptional regulator YybR => putative HTH-type transcriptional regulator YybR\n",
      "[15:28:28] Modify product: Probable type I restriction enzyme BthVORF4518P M protein => putative type I restriction enzymeP M protein\n",
      "[15:28:28] Modify product: Leu/Ile/Val-binding protein homolog 3 => Leu/Ile/Val-binding protein \n",
      "[15:28:28] Modify product: Probable diguanylate cyclase AdrA => putative diguanylate cyclase AdrA\n",
      "[15:28:28] Modify product: Uncharacterized MFS-type transporter Rv2994 => putative MFS-type transporter\n",
      "[15:28:28] Modify product: Probable cystine transporter YijE => putative cystine transporter YijE\n",
      "[15:28:28] Modify product: Uncharacterized acyl-CoA thioester hydrolase HI_0827 => putative acyl-CoA thioester hydrolase\n",
      "[15:28:28] Modify product: Probable glycerophosphodiester phosphodiesterase 1 => putative glycerophosphodiester phosphodiesterase 1\n",
      "[15:28:28] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:28] Modify product: Putative peroxiredoxin sll1621 => Putative peroxiredoxin\n",
      "[15:28:28] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:29] Modify product: Putative phosphoribosyl transferase MT0597 => Putative phosphoribosyl transferase\n",
      "[15:28:29] Modify product: Uncharacterized FAD-linked oxidoreductase Rv2280 => putative FAD-linked oxidoreductase\n",
      "[15:28:29] Modify product: Probable phospholipid import ATP-binding protein MlaF => putative phospholipid import ATP-binding protein MlaF\n",
      "[15:28:29] Modify product: Probable phospholipid ABC transporter permease protein MlaE => putative phospholipid ABC transporter permease protein MlaE\n",
      "[15:28:29] Modify product: Probable phospholipid ABC transporter-binding protein MlaD => putative phospholipid ABC transporter-binding protein MlaD\n",
      "[15:28:29] Modify product: Probable phospholipid-binding lipoprotein MlaA => putative phospholipid-binding lipoprotein MlaA\n",
      "[15:28:29] Modify product: Probable phospholipid-binding protein MlaC => putative phospholipid-binding protein MlaC\n",
      "[15:28:29] Modify product: GTP cyclohydrolase 1 type 2 homolog => GTP cyclohydrolase 1 type 2 \n",
      "[15:28:29] Modify product: Stringent starvation protein B homolog => Stringent starvation protein B \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[15:28:29] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:29] Modify product: Uncharacterized FAD-linked oxidoreductase Rv2280 => putative FAD-linked oxidoreductase\n",
      "[15:28:29] Modify product: Probable protein kinase UbiB => putative protein kinase UbiB\n",
      "[15:28:29] Modify product: Probable CtpA-like serine protease => putative CtpA-like serine protease\n",
      "[15:28:29] Modify product: Putative multidrug export ATP-binding/permease protein SA1683 => Putative multidrug export ATP-binding/permease protein\n",
      "[15:28:29] Modify product: Probable DNA endonuclease SmrA => putative DNA endonuclease SmrA\n",
      "[15:28:29] Modify product: Uncharacterized metal-dependent hydrolase YjjV => putative metal-dependent hydrolase YjjV\n",
      "[15:28:29] Modify product: Solute-binding protein Veis_3954 => Solute-binding protein\n",
      "[15:28:29] Modify product: Probable succinyl-CoA:3-ketoacid coenzyme A transferase subunit B => putative succinyl-CoA:3-ketoacid coenzyme A transferase subunit B\n",
      "[15:28:29] Modify product: Probable succinyl-CoA:3-ketoacid coenzyme A transferase subunit A => putative succinyl-CoA:3-ketoacid coenzyme A transferase subunit A\n",
      "[15:28:30] Modify product: Uncharacterized protease YdcP => putative protease YdcP\n",
      "[15:28:30] Modify product: Probable 3-phenylpropionic acid transporter => putative 3-phenylpropionic acid transporter\n",
      "[15:28:30] Modify product: Uncharacterized FAD-linked oxidoreductase Rv2280 => putative FAD-linked oxidoreductase\n",
      "[15:28:30] Modify product: Probable cystine transporter YijE => putative cystine transporter YijE\n",
      "[15:28:30] Modify product: Probable enoyl-CoA hydratase echA12 => putative enoyl-CoA hydratase echA12\n",
      "[15:28:30] Modify product: Solute-binding protein Rfer_1840 => Solute-binding protein\n",
      "[15:28:30] Modify product: Putative formate dehydrogenase SA2102 => Putative formate dehydrogenase\n",
      "[15:28:30] Modify product: Probable acyltransferase YihG => putative acyltransferase YihG\n",
      "[15:28:30] Modify product: Probable TonB-dependent receptor NMB0964 => putative TonB-dependent receptor\n",
      "[15:28:30] Modify product: Probable xanthine dehydrogenase subunit A => putative xanthine dehydrogenase subunit A\n",
      "[15:28:30] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:30] Modify product: Uncharacterized aminopeptidase Rv1333 => putative aminopeptidase\n",
      "[15:28:30] Modify product: Cyclic nucleotide-gated potassium channel mll3241 => Cyclic nucleotide-gated potassium channel\n",
      "[15:28:30] Modify product: Putative universal stress protein SA1532 => Putative universal stress protein\n",
      "[15:28:30] Modify product: Putative peptidyl-prolyl cis-trans isomerase HP_0175 => Putative peptidyl-prolyl cis-trans isomerase\n",
      "[15:28:30] Modify product: Chemotaxis protein CheY homolog => Chemotaxis protein CheY \n",
      "[15:28:31] Modify product: Cyclic diguanosine monophosphate-binding protein PA4608 => Cyclic diguanosine monophosphate-binding protein\n",
      "[15:28:31] Modify product: Uncharacterized HTH-type transcriptional regulator Rv1395 => putative HTH-type transcriptional regulator\n",
      "[15:28:31] Modify product: Cyclic diguanosine monophosphate-binding protein PA4608 => Cyclic diguanosine monophosphate-binding protein\n",
      "[15:28:31] Modify product: Probable type I restriction enzyme BthVORF4518P M protein => putative type I restriction enzymeP M protein\n",
      "[15:28:31] Modify product: Probable fatty acid methyltransferase Rv3720 => putative fatty acid methyltransferase\n",
      "[15:28:31] Modify product: Leu/Ile/Val-binding protein homolog 3 => Leu/Ile/Val-binding protein \n",
      "[15:28:31] Modify product: Uncharacterized NTE family protein Rv1063c => putative NTE family protein\n",
      "[15:28:31] Modify product: PKHD-type hydroxylase Sbal_3634 => PKHD-type hydroxylase\n",
      "[15:28:31] Modify product: Probable TonB-dependent receptor BfrD => putative TonB-dependent receptor BfrD\n",
      "[15:28:31] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:31] Modify product: Probable sulfate transporter Rv1739c => putative sulfate transporter\n",
      "[15:28:31] Modify product: Leu/Ile/Val-binding protein homolog 3 => Leu/Ile/Val-binding protein \n",
      "[15:28:32] Modify product: Uncharacterized HTH-type transcriptional regulator Rv1395 => putative HTH-type transcriptional regulator\n",
      "[15:28:32] Modify product: Probable Fe(2+)-trafficking protein => putative Fe(2+)-trafficking protein\n",
      "[15:28:32] Modify product: Phosphate-specific transport system accessory protein PhoU homolog => Phosphate-specific transport system accessory protein PhoU \n",
      "[15:28:32] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:32] Modify product: Monocarboxylate 2-oxoacid-binding periplasmic protein all3028 => Monocarboxylate 2-oxoacid-binding periplasmic protein\n",
      "[15:28:32] Modify product: Uncharacterized hydrolase YxeP => putative hydrolase YxeP\n",
      "[15:28:32] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:33] Modify product: Probable enoyl-CoA hydratase echA12 => putative enoyl-CoA hydratase echA12\n",
      "[15:28:33] Modify product: Uncharacterized GST-like protein YibF => putative GST-like protein YibF\n",
      "[15:28:33] Modify product: Putative quercetin 2,3-dioxygenase Rv0181c => Putative quercetin 2,3-dioxygenase\n",
      "[15:28:33] Modify product: Uncharacterized oxidoreductase Rv1144 => putative oxidoreductase\n",
      "[15:28:33] Modify product: RNA-binding protein HI_1333 => RNA-binding protein\n",
      "[15:28:33] Modify product: Probable diguanylate cyclase AdrA => putative diguanylate cyclase AdrA\n",
      "[15:28:33] Modify product: Putative short-chain type dehydrogenase/reductase Rv0148 => Putative short-chain type dehydrogenase/reductase\n",
      "[15:28:33] Modify product: Uncharacterized oxidoreductase SA2266 => putative oxidoreductase\n",
      "[15:28:33] Modify product: Uncharacterized symporter YjmB => putative symporter YjmB\n",
      "[15:28:33] Modify product: Probable transcriptional regulatory protein CBU_1566 => putative transcriptional regulatory protein\n",
      "[15:28:33] Modify product: Probable nicotinate-nucleotide adenylyltransferase => putative nicotinate-nucleotide adenylyltransferase\n",
      "[15:28:34] Modify product: Uncharacterized signaling protein CC_0091 => putative signaling protein\n",
      "[15:28:34] Modify product: Probable transcriptional regulatory protein aq_1575 => putative transcriptional regulatory protein\n",
      "[15:28:34] Modify product: Probable aminotransferase Rv3329 => putative aminotransferase\n",
      "[15:28:34] Modify product: Phosphatase Rv3376 => Phosphatase\n",
      "[15:28:34] Modify product: Probable parvulin-type peptidyl-prolyl cis-trans isomerase => putative parvulin-type peptidyl-prolyl cis-trans isomerase\n",
      "[15:28:34] Modify product: Probable intracellular septation protein A => putative intracellular septation protein A\n",
      "[15:28:34] Modify product: Acetyltransferase SACOL1063 => Acetyltransferase\n",
      "[15:28:34] Modify product: Uncharacterized metal-dependent hydrolase YcfH => putative metal-dependent hydrolase YcfH\n",
      "[15:28:34] Modify product: Probable diguanylate cyclase YdaM => putative diguanylate cyclase YdaM\n",
      "[15:28:34] Modify product: Solute-binding protein Dde_0634 => Solute-binding protein\n",
      "[15:28:35] Modify product: Probable chromosome-partitioning protein ParB => putative chromosome-partitioning protein ParB\n",
      "[15:28:35] Modify product: Probable cytochrome c oxidase subunit 1 => putative cytochrome c oxidase subunit 1\n",
      "[15:28:35] Cleaned 160 /product names\n",
      "[15:28:35] Deleting unwanted file: ./data/annotation/8//sprot.faa\n",
      "[15:28:35] Deleting unwanted file: ./data/annotation/8//sprot.blast\n",
      "[15:28:36] There are still 1169 unannotated CDS left (started with 3369)\n",
      "[15:28:36] Will use hmmer3 to search against /opt/prokka/bin/../db/hmm/HAMAP.hmm with 1 CPUs\n",
      "[15:28:36] Running: cat \\.\\/data\\/annotation\\/8\\/\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 1 --block 151432 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /opt/prokka/bin/../db/hmm/HAMAP.hmm /dev/stdin > \\.\\/data\\/annotation\\/8\\/\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[15:30:09] Modify product: Uncharacterized Nudix hydrolase NudL => putative Nudix hydrolase NudL\n",
      "[15:30:10] Cleaned 1 /product names\n",
      "[15:30:10] Deleting unwanted file: ./data/annotation/8//HAMAP.hmm.faa\n",
      "[15:30:10] Deleting unwanted file: ./data/annotation/8//HAMAP.hmm.hmmer3\n",
      "[15:30:10] Labelling remaining 1054 proteins as 'hypothetical protein'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[15:30:10] Possible /pseudo 'Homoprotocatechuate catabolism bifunctional isomerase/decarboxylase' at gnl|X|FLIOPFNA_1 position 186127\n",
      "[15:30:10] Possible /pseudo 'Lipid A biosynthesis lauroyltransferase' at gnl|X|FLIOPFNA_1 position 368563\n",
      "[15:30:10] Possible /pseudo 'Methyl-accepting chemotaxis protein II' at gnl|X|FLIOPFNA_1 position 509422\n",
      "[15:30:10] Possible /pseudo 'Methyl-accepting chemotaxis protein II' at gnl|X|FLIOPFNA_1 position 511129\n",
      "[15:30:10] Possible /pseudo 'Transcriptional regulatory protein WalR' at gnl|X|FLIOPFNA_2 position 74598\n",
      "[15:30:10] Possible /pseudo 'Protein YceI' at gnl|X|FLIOPFNA_2 position 280799\n",
      "[15:30:10] Possible /pseudo 'putative zinc protease' at gnl|X|FLIOPFNA_2 position 295398\n",
      "[15:30:10] Possible /pseudo 'Low-molecular weight cobalt-containing nitrile hydratase subunit alpha' at gnl|X|FLIOPFNA_3 position 76286\n",
      "[15:30:10] Possible /pseudo 'Low-molecular weight cobalt-containing nitrile hydratase subunit alpha' at gnl|X|FLIOPFNA_3 position 76766\n",
      "[15:30:10] Possible /pseudo 'Low-molecular weight cobalt-containing nitrile hydratase subunit alpha' at gnl|X|FLIOPFNA_3 position 77339\n",
      "[15:30:10] Possible /pseudo 'putative signaling protein' at gnl|X|FLIOPFNA_4 position 65327\n",
      "[15:30:10] Possible /pseudo 'Twitching mobility protein' at gnl|X|FLIOPFNA_6 position 201883\n",
      "[15:30:10] Possible /pseudo 'Cyanophycin synthetase' at gnl|X|FLIOPFNA_7 position 129146\n",
      "[15:30:10] Possible /pseudo 'Formate dehydrogenase-O major subunit' at gnl|X|FLIOPFNA_8 position 159762\n",
      "[15:30:10] Possible /pseudo 'Alpha-keto acid-binding periplasmic protein TakP' at gnl|X|FLIOPFNA_8 position 171779\n",
      "[15:30:10] Possible /pseudo 'Dibenzothiophene desulfurization enzyme C' at gnl|X|FLIOPFNA_9 position 24241\n",
      "[15:30:10] Possible /pseudo 'Nitrite reductase' at gnl|X|FLIOPFNA_9 position 39654\n",
      "[15:30:10] Possible /pseudo 'A-type flagellin' at gnl|X|FLIOPFNA_10 position 26508\n",
      "[15:30:10] Possible /pseudo 'Peptidyl-prolyl cis-trans isomerase cyp18' at gnl|X|FLIOPFNA_11 position 8834\n",
      "[15:30:10] Possible /pseudo 'Putative aliphatic sulfonates-binding protein' at gnl|X|FLIOPFNA_13 position 87418\n",
      "[15:30:10] Possible /pseudo 'Formyl-CoA:oxalate CoA-transferase' at gnl|X|FLIOPFNA_19 position 5632\n",
      "[15:30:10] Found 1680 unique /gene codes.\n",
      "[15:30:10] Fixed 4 duplicate /gene - pcpR_1 pcpR_2 pcpR_3 pcpR_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - pbuE_1 pbuE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - eamA_1 eamA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - glnS_1 glnS_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - loiP_1 loiP_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rsxB_1 rsxB_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - kynB_1 kynB_2 kynB_3 kynB_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - rpsA_1 rpsA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ligB_1 ligB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - gno_1 gno_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - hemA_1 hemA_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - fhuA_1 fhuA_2 fhuA_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - tolC_1 tolC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - glnL_1 glnL_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - cysG_1 cysG_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - infA_1 infA_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - glnQ_1 glnQ_2 glnQ_3 glnQ_4\n",
      "[15:30:10] Fixed 4 duplicate /gene - envZ_1 envZ_2 envZ_3 envZ_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - nirQ_1 nirQ_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - aroG_1 aroG_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - mopA_1 mopA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - tsaM1_1 tsaM1_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - smc_1 smc_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - slyD_1 slyD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - kefC_1 kefC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - menI_1 menI_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rlmN_1 rlmN_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yecS_1 yecS_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - cc4_1 cc4_2 cc4_3\n",
      "[15:30:10] Fixed 3 duplicate /gene - gltC_1 gltC_2 gltC_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - mepM_1 mepM_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ubiG_1 ubiG_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - gmhA_1 gmhA_2 gmhA_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - cheA_1 cheA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rsmI_1 rsmI_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - czcB_1 czcB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ais_1 ais_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - gatA_1 gatA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - fccB_1 fccB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nqo2_1 nqo2_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ctaD_1 ctaD_2\n",
      "[15:30:10] Fixed 5 duplicate /gene - czcA_1 czcA_2 czcA_3 czcA_4 czcA_5\n",
      "[15:30:10] Fixed 2 duplicate /gene - scpA_1 scpA_2\n",
      "[15:30:10] Fixed 7 duplicate /gene - pleD_1 pleD_2 pleD_3 pleD_4 pleD_5 pleD_6 pleD_7\n",
      "[15:30:10] Fixed 4 duplicate /gene - degU_1 degU_2 degU_3 degU_4\n",
      "[15:30:10] Fixed 3 duplicate /gene - soj_1 soj_2 soj_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - dpnA_1 dpnA_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - pdeB_1 pdeB_2 pdeB_3 pdeB_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - hldD_1 hldD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - metE_1 metE_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - ligA_1 ligA_2 ligA_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - uvrY_1 uvrY_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - apxIB_1 apxIB_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - cphA_1 cphA_2 cphA_3 cphA_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - yfdE_1 yfdE_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - trxA_1 trxA_2 trxA_3\n",
      "[15:30:10] Fixed 3 duplicate /gene - trg_1 trg_2 trg_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - iscR_1 iscR_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - phoR_1 phoR_2 phoR_3 phoR_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - cysB_1 cysB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - fmt_1 fmt_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - pchR_1 pchR_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - crtB_1 crtB_2 crtB_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - mdeA_1 mdeA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - amtB_1 amtB_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - nicS_1 nicS_2 nicS_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - glnB_1 glnB_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - mnmC_1 mnmC_2 mnmC_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - rfaE_1 rfaE_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - gcvA_1 gcvA_2 gcvA_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - mrcA_1 mrcA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - sumT_1 sumT_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - metH_1 metH_2\n",
      "[15:30:10] Fixed 7 duplicate /gene - dctM_1 dctM_2 dctM_3 dctM_4 dctM_5 dctM_6 dctM_7\n",
      "[15:30:10] Fixed 2 duplicate /gene - ygiD_1 ygiD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - oprF_1 oprF_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - dapE_1 dapE_2 dapE_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - sbp_1 sbp_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - icd_1 icd_2\n",
      "[15:30:10] Fixed 10 duplicate /gene - lptB_1 lptB_2 lptB_3 lptB_4 lptB_5 lptB_6 lptB_7 lptB_8 lptB_9 lptB_10\n",
      "[15:30:10] Fixed 4 duplicate /gene - guaB_1 guaB_2 guaB_3 guaB_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - hpcE_1 hpcE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - acdA_1 acdA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rluF_1 rluF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - gbpR_1 gbpR_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - cobN_1 cobN_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - uvrA_1 uvrA_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - resA_1 resA_2 resA_3 resA_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - srrA_1 srrA_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - ssuC_1 ssuC_2 ssuC_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - frc_1 frc_2\n",
      "[15:30:10] Fixed 14 duplicate /gene - livF_1 livF_2 livF_3 livF_4 livF_5 livF_6 livF_7 livF_8 livF_9 livF_10 livF_11 livF_12 livF_13 livF_14\n",
      "[15:30:10] Fixed 2 duplicate /gene - dacC_1 dacC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rsmB_1 rsmB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rpfG_1 rpfG_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - walR_1 walR_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - czcC_1 czcC_2 czcC_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - ytrE_1 ytrE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ilvD_1 ilvD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - trxC_1 trxC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - dxs_1 dxs_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - petA_1 petA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yiaJ_1 yiaJ_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - Hgd_1 Hgd_2 Hgd_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - corA_1 corA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - cydA_1 cydA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - csoR_1 csoR_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - fliC_1 fliC_2 fliC_3 fliC_4\n",
      "[15:30:10] Fixed 5 duplicate /gene - menH_1 menH_2 menH_3 menH_4 menH_5\n",
      "[15:30:10] Fixed 2 duplicate /gene - garD_1 garD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - fdxA_1 fdxA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ppsC_1 ppsC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - fdoI_1 fdoI_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rfaF_1 rfaF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - hosA_1 hosA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ftsN_1 ftsN_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yiaM_1 yiaM_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - cpdA_1 cpdA_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - pfeA_1 pfeA_2 pfeA_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoF_1 nuoF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rmlC_1 rmlC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yceI_1 yceI_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - argH_1 argH_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - fdhD_1 fdhD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - flgH_1 flgH_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yehU_1 yehU_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - pyrC_1 pyrC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - pstS_1 pstS_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - nirS_1 nirS_2 nirS_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - ssuB_1 ssuB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - pcaF_1 pcaF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - tuf1_1 tuf1_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - stp_1 stp_2 stp_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - fadA_1 fadA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - bepE_1 bepE_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - cycA_1 cycA_2 cycA_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - rimI_1 rimI_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rpoE_1 rpoE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - fdoG_1 fdoG_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - fdhF_1 fdhF_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - accA1_1 accA1_2 accA1_3\n",
      "[15:30:10] Fixed 3 duplicate /gene - gmr_1 gmr_2 gmr_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - lutR_1 lutR_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - lpxL_1 lpxL_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - plsC_1 plsC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - clcD_1 clcD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yjiE_1 yjiE_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - mdtA_1 mdtA_2 mdtA_3 mdtA_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - ydcP_1 ydcP_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - takP_1 takP_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - macA_1 macA_2 macA_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - lplT_1 lplT_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - barA_1 barA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yajR_1 yajR_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - copA_1 copA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nagG_1 nagG_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - cysL_1 cysL_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - srpC_1 srpC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - lon1_1 lon1_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - zraR_1 zraR_2 zraR_3 zraR_4\n",
      "[15:30:10] Fixed 3 duplicate /gene - oxyR_1 oxyR_2 oxyR_3\n",
      "[15:30:10] Fixed 3 duplicate /gene - novR_1 novR_2 novR_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - bfrD_1 bfrD_2\n",
      "[15:30:10] Fixed 8 duplicate /gene - dmlR_1 dmlR_2 dmlR_3 dmlR_4 dmlR_5 dmlR_6 dmlR_7 dmlR_8\n",
      "[15:30:10] Fixed 4 duplicate /gene - oprM_1 oprM_2 oprM_3 oprM_4\n",
      "[15:30:10] Fixed 4 duplicate /gene - exbB_1 exbB_2 exbB_3 exbB_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - cysW_1 cysW_2\n",
      "[15:30:10] Fixed 5 duplicate /gene - xerC_1 xerC_2 xerC_3 xerC_4 xerC_5\n",
      "[15:30:10] Fixed 2 duplicate /gene - pcm_1 pcm_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - glxR_1 glxR_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yfcG_1 yfcG_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - hsdR_1 hsdR_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - gsiA_1 gsiA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - dnaK_1 dnaK_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ubiE_1 ubiE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nudC_1 nudC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - aer_1 aer_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - slt_1 slt_2 slt_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - bclA_1 bclA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - mmgC_1 mmgC_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - crt_1 crt_2 crt_3 crt_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - corC_1 corC_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - acsA_1 acsA_2 acsA_3 acsA_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - araC_1 araC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nlhH_1 nlhH_2\n",
      "[15:30:10] Fixed 5 duplicate /gene - cysA_1 cysA_2 cysA_3 cysA_4 cysA_5\n",
      "[15:30:10] Fixed 2 duplicate /gene - rimO_1 rimO_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - rhlE_1 rhlE_2 rhlE_3 rhlE_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - ywlC_1 ywlC_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - amiC_1 amiC_2 amiC_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - macB_1 macB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ugpC_1 ugpC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - gph_1 gph_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - fixL_1 fixL_2 fixL_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - qseC_1 qseC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - hemN_1 hemN_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - dsbD_1 dsbD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - cmpD_1 cmpD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - adrA_1 adrA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - uvrD_1 uvrD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - paaZ_1 paaZ_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ywrD_1 ywrD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - pyrD_1 pyrD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - acrR_1 acrR_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - legF_1 legF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - gcvH_1 gcvH_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - bcp_1 bcp_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - argD_1 argD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ydaM_1 ydaM_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - cydB_1 cydB_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - fbp_1 fbp_2 fbp_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - dauA_1 dauA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rluA_1 rluA_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - rspR_1 rspR_2 rspR_3 rspR_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - parB_1 parB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rpfC_1 rpfC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - htpX_1 htpX_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - xerD_1 xerD_2 xerD_3\n",
      "[15:30:10] Fixed 5 duplicate /gene - tsr_1 tsr_2 tsr_3 tsr_4 tsr_5\n",
      "[15:30:10] Fixed 2 duplicate /gene - pyrE_1 pyrE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - qseB_1 qseB_2\n",
      "[15:30:10] Fixed 5 duplicate /gene - cynR_1 cynR_2 cynR_3 cynR_4 cynR_5\n",
      "[15:30:10] Fixed 2 duplicate /gene - proA_1 proA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - glmU_1 glmU_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - queE_1 queE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ftsA_1 ftsA_2\n",
      "[15:30:10] Fixed 7 duplicate /gene - fabG_1 fabG_2 fabG_3 fabG_4 fabG_5 fabG_6 fabG_7\n",
      "[15:30:10] Fixed 3 duplicate /gene - pilT_1 pilT_2 pilT_3\n",
      "[15:30:10] Fixed 3 duplicate /gene - dosC_1 dosC_2 dosC_3\n",
      "[15:30:10] Fixed 4 duplicate /gene - exbD_1 exbD_2 exbD_3 exbD_4\n",
      "[15:30:10] Fixed 3 duplicate /gene - cheY_1 cheY_2 cheY_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - crp_1 crp_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - benM_1 benM_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - proP_1 proP_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - fadD3_1 fadD3_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - apc3_1 apc3_2\n",
      "[15:30:10] Fixed 7 duplicate /gene - bepA_1 bepA_2 bepA_3 bepA_4 bepA_5 bepA_6 bepA_7\n",
      "[15:30:10] Fixed 3 duplicate /gene - nagL_1 nagL_2 nagL_3\n",
      "[15:30:10] Fixed 4 duplicate /gene - lysN_1 lysN_2 lysN_3 lysN_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - slmA_1 slmA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - intA_1 intA_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - ohrR_1 ohrR_2 ohrR_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - nsrR_1 nsrR_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nasA_1 nasA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - lcfB_1 lcfB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - mlaE_1 mlaE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - soxC_1 soxC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - cnrB_1 cnrB_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - gloB_1 gloB_2 gloB_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - msrP_1 msrP_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - hmuU_1 hmuU_2 hmuU_3\n",
      "[15:30:10] Fixed 14 duplicate /gene - livH_1 livH_2 livH_3 livH_4 livH_5 livH_6 livH_7 livH_8 livH_9 livH_10 livH_11 livH_12 livH_13 livH_14\n",
      "[15:30:10] Fixed 2 duplicate /gene - cugP_1 cugP_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - btuF_1 btuF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - fdx_1 fdx_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - amiD_1 amiD_2\n",
      "[15:30:10] Fixed 15 duplicate /gene - tar_1 tar_2 tar_3 tar_4 tar_5 tar_6 tar_7 tar_8 tar_9 tar_10 tar_11 tar_12 tar_13 tar_14 tar_15\n",
      "[15:30:10] Fixed 3 duplicate /gene - hdaH_1 hdaH_2 hdaH_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - alkJ_1 alkJ_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - mftC_1 mftC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - bcr_1 bcr_2\n",
      "[15:30:10] Fixed 5 duplicate /gene - braC_1 braC_2 braC_3 braC_4 braC_5\n",
      "[15:30:10] Fixed 2 duplicate /gene - epsE_1 epsE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yijE_1 yijE_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - cph2_1 cph2_2 cph2_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - kup_1 kup_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - teaD_1 teaD_2 teaD_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - moeA_1 moeA_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - dmdC_1 dmdC_2 dmdC_3 dmdC_4\n",
      "[15:30:10] Fixed 2 duplicate /gene - pht2_1 pht2_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - fadD_1 fadD_2 fadD_3\n",
      "[15:30:10] Fixed 3 duplicate /gene - cheW_1 cheW_2 cheW_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - ssuA_1 ssuA_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - ftsH_1 ftsH_2 ftsH_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - lpxH_1 lpxH_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - tqsA_1 tqsA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - dadA1_1 dadA1_2\n",
      "[15:30:10] Fixed 3 duplicate /gene - nagK_1 nagK_2 nagK_3\n",
      "[15:30:10] Fixed 3 duplicate /gene - ompR_1 ompR_2 ompR_3\n",
      "[15:30:10] Fixed 2 duplicate /gene - thiD_1 thiD_2\n",
      "[15:30:10] Fixed 280 colliding /gene names.\n",
      "[15:30:10] Adding /locus_tag identifiers\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[15:30:11] Assigned 3409 locus_tags to CDS and RNA features.\n",
      "[15:30:11] Writing outputs to ./data/annotation/8//\n",
      "[15:30:22] Generating annotation statistics file\n",
      "[15:30:22] Generating Genbank and Sequin files\n",
      "[15:30:22] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.12 from https://github.com/tseemann/prokka' -Z \\.\\/data\\/annotation\\/8\\/\\/PROKKA_06012018\\.err -i \\.\\/data\\/annotation\\/8\\/\\/PROKKA_06012018\\.fsa 2> /dev/null\n",
      "[15:30:40] Deleting unwanted file: ./data/annotation/8//errorsummary.val\n",
      "[15:30:40] Deleting unwanted file: ./data/annotation/8//PROKKA_06012018.dr\n",
      "[15:30:40] Deleting unwanted file: ./data/annotation/8//PROKKA_06012018.fixedproducts\n",
      "[15:30:40] Deleting unwanted file: ./data/annotation/8//PROKKA_06012018.ecn\n",
      "[15:30:40] Deleting unwanted file: ./data/annotation/8//PROKKA_06012018.val\n",
      "[15:30:40] Repairing broken .GBK output that tbl2asn produces...\n",
      "[15:30:40] Running: sed 's/COORDINATES: profile/COORDINATES:profile/' < \\.\\/data\\/annotation\\/8\\/\\/PROKKA_06012018\\.gbf > \\.\\/data\\/annotation\\/8\\/\\/PROKKA_06012018\\.gbk\n",
      "[15:30:40] Deleting unwanted file: ./data/annotation/8//PROKKA_06012018.gbf\n",
      "[15:30:40] Output files:\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.fsa\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.gff\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.err\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.txt\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.tbl\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.faa\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.gbk\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.fna\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.sqn\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.tsv\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.log\n",
      "[15:30:40] ./data/annotation/8/PROKKA_06012018.ffn\n",
      "[15:30:40] Annotation finished successfully.\n",
      "[15:30:40] Walltime used: 12.90 minutes\n",
      "[15:30:40] If you use this result please cite the Prokka paper:\n",
      "[15:30:40] Seemann T (2014) Prokka: rapid prokaryotic genome annotation. Bioinformatics. 30(14):2068-9.\n",
      "[15:30:40] Type 'prokka --citation' for more details.\n",
      "[15:30:40] Thank you, come again.\n",
      "\tCommand being timed: \"/opt/prokka/bin/prokka ./data/bins/testlaura.8.fa --cpus 1 --outdir ./data/annotation/8/ --centre X --compliant\"\n",
      "\tUser time (seconds): 719.83\n",
      "\tSystem time (seconds): 40.14\n",
      "\tPercent of CPU this job got: 98%\n",
      "\tElapsed (wall clock) time (h:mm:ss or m:ss): 12:54.97\n",
      "\tAverage shared text size (kbytes): 0\n",
      "\tAverage unshared data size (kbytes): 0\n",
      "\tAverage stack size (kbytes): 0\n",
      "\tAverage total size (kbytes): 0\n",
      "\tMaximum resident set size (kbytes): 226692\n",
      "\tAverage resident set size (kbytes): 0\n",
      "\tMajor (requiring I/O) page faults: 24\n",
      "\tMinor (reclaiming a frame) page faults: 966539\n",
      "\tVoluntary context switches: 84112\n",
      "\tInvoluntary context switches: 33351\n",
      "\tSwaps: 0\n",
      "\tFile system inputs: 3992\n",
      "\tFile system outputs: 210872\n",
      "\tSocket messages sent: 0\n",
      "\tSocket messages received: 0\n",
      "\tSignals delivered: 0\n",
      "\tPage size (bytes): 4096\n",
      "\tExit status: 0\n"
     ]
    }
   ],
   "source": [
    "!time -v /opt/prokka/bin/prokka ./data/bins/testlaura.8.fa --cpus 1 --outdir ./data/annotation/8/ --centre X --compliant\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "(think and discuss these questions) <br>\n",
    "\n",
    "What did you do? <br>\n",
    "Where is your output? <br>\n",
    "What does your output look like? <br>\n",
    "Do you know what all the output files are or mean? <br>\n",
    "\n",
    "(HINT: you can look at the prokka files in the same way we looked at the reads and scaffold earlier)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>investigating prokka output</b><br>\n",
    "to investigate the prokka output you can use two webservers that both are able to place the annotations from prokka in KEGG pathways.<br>\n",
    "\n",
    "(1) prokka gives uniprot IDs in the gff files first we will collect these IDs,\n",
    "(2) since we have Uniprot IDs from prokka we are going to convert these to **KO IDs** that can be used by KEGG using this website:<br>\n",
    "http://www.uniprot.org/uploadlists/\n",
    "\n",
    "\n",
    "(3) Make sure to download the 'mapping table', this you can supply to KEGG. The 'target list' you can upload in the pathways.embl.de webpage. In these webpages (links below) you can investigate the metabolism of your specific bin (in the ideal world: your specific microbe) by investigating which genes make complete pathways.\n",
    "\n",
    "You may feel overwhelmed by the amount of pathways, modules and genes available to you. For this specific case, we are  interested in Nitrogen metabolism. You may have a look at the Dijkhuizen et al. 2018 paper on Azolla endophytes. In figure 4 of that paper, the nitrogen metabolism of multiple microbes is plotted. Next some hypotheses are derived and tested in the wetlab. Does your plot of the nitrogen metabolism overlap with the one published? Or did you maybe discover a new endophyte!\n",
    "\n",
    "http://www.genome.jp/kegg/tool/map_module.html<br>\n",
    "https://pathways.embl.de/ipath3.cgi?map=metabolic<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "view prokka output:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!less ....yourprokkaoutput.gff | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "take the uniprot IDs out of the gff file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!grep -o 'UniProt.*' ....yourprokkaoutput | cut -d';' -f1 | cut -d':' -f2 > listofuniprotIDs"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
