{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# General introduction\n",
    "Today, you will assemble and annotate a metagenome of a plant: *Azolla filiculoides*. *Azolla* is known for its endophytes: microorganisms that live inside the plant. These microorganisms live specifically inside a leaf cavity in every upper leaf lobe of the plant. One of these endophytes has been known for a long time: *Nostoc azollae* They are best known for fixing N^2 from the air, in service of the plant. The image included below illustrates the physiology of these leaf cavities. \n",
    "\n",
    "While the presence of *N. azollae* is well studied, the presence of other microbes in the *Azolla* leaf pocket is not. Today, we will create and analyse the metagenome of *A. filiculoides* plants, in search for these other \"neglected\" microbes. \n",
    "\n",
    "![image](http://theazollafoundation.org/wp-content/uploads/2013/05/Azolla-leaf-cavities-with-Anabaena.png)\n",
    "\n",
    "Hence, the open biological question we are answering today is \"which species live inside these leaf cavities and what are the metabolic capacities encoded on their genomes?\" Based on these genomes, you may be able to infer their function and think of a hypotheses to test. \n",
    "\n",
    "*Tip*: If you stumble upon a step with a long calculation time, first read ahead to see what steps come next, and otherwise you can google some information about the plant you are researching. Notice that this [paper](http://onlinelibrary.wiley.com/doi/10.1111/nph.14843/full) specifically describes the data that we will be using today. For a more general introduction into Azolla, perhaps have a look at [this short movie](https://youtu.be/O34gTsxyDq8) or [this short movie](https://youtu.be/OI4VV4M2-f4) (Dutch) recorded for Dutch local television.\n",
    "\n",
    "\n",
    "# Workflow\n",
    "\n",
    "## Acquire Samples\n",
    "A typical metagenomics workflow starts in the environment and/or the wetlab with a DNA extraction from some biological sample. This is also true for this workflow. We have extracted DNA from *Azolla* from the environment in two ways. Firstly, the whole Plant (P), secondly from the Leaf cavities (L). Both were collected in three biological replicates. Hence the samples we will work with are coded L1 L2 L3 and P1 P2 P3. The reasoning behind these two sampling types is that true endophytes are expected to be more abundant in leaf samples than in plant samples. So we may later use these different sampling types to infer if a microbial genome that we assembled originated from the leaf cavity, or from somewhere else in or on the plant. \n",
    "\n",
    "Extracted DNA is processed in a sequencing library; it is prepared in a way that a [Illumina sequencing](https://en.wikipedia.org/wiki/DNA_sequencing#Illumina_(Solexa)_sequencing) machine can process it. The sequencing machine then provides us with [FastQ](https://en.wikipedia.org/wiki/FASTQ_format) files, you may have handled these before, perhaps not. We will have a look at FastQ files later in this practical.\n",
    "\n",
    "## First Quality Control and processing.\n",
    "FastQ files contain the output of the Illumina sequencing machine: tens of millions of short DNA sequences. Where short is anything between 50 and 250 bp. These original short DNA sequences are further refered to as DNA reads, or just reads. FastQ files contain unprocessed output of the machine, every read consists of four lines: \n",
    "1. a header proceeded by the '@' character. \n",
    "2. the DNA sequence\n",
    "3. a comment line proceeded by a '+'\n",
    "4. a quality string.\n",
    "\n",
    "The quality string has the exact same length as the DNA read, every character in the quality string corresponds to some number between 0 and 40. The sequencing machine assigns this quality score based on \"how sure it is\" that this specific base is correct. To put it a bit loosly.\n",
    "\n",
    "These are two reads encoded in a FastQ file.\n",
    "\n",
    "> @NS500813:28:H3M3VAFXX:1:11101:19270:1015 1:N:0:CGATGT\n",
    "GNGGTGAAGAAATCAGCCATTCTAAACCAATTGCTCTCCAAGGATTATCAGGTGCTTTTTCACCTTGCGTCCAAGAAACCAAAATATTGAATATGAAGGGTAAGGTGGACATTCCTAATAAGAATGCACCAAGACTAGCGAGAACATTCC\n",
    "+\n",
    "A#AAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEAEE<EEEEEEEEEEEEEEEEEEEEEAEEEAEEEEEEEEEEEEEEEEEEEEAAAEEAEEEEEAAAAA6AEAEAAEEEEEEEEEEEEA6\n",
    "@NS500813:28:H3M3VAFXX:1:11101:24948:1015 1:N:0:CGATGT\n",
    "ANATTAATTATACAATCACTAATTTAGCAGAGCATTTAGAAAGTATAAGCCATGATGCAATTAACTATTATTTAAAAACCGAAAAGTTAACATCTCGTTTACTATGGGATAAGGTGAAAGAGGTAGTAGAACCTGATGGTAATGGGTACAT\n",
    "+\n",
    "A#AAAAEEEEE6EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE/EEAEEEEEEEAEEEEEEEEEEE<EEE<EEEEE66EEA<EEEEAEEEEEEEEEEEAE<E/AE/EEEEEAEAAEAAAEEEEAEEEE6<EEAEA<//AAEEAEE<AA\n",
    "\n",
    "FastQ files often contain some reads of poor quality, we definitively want to filter those out before starting our analyses. There are many ways available, for todays practical, this was done for you already with a tool called [trimmomatic](https://academic.oup.com/bioinformatics/article-abstract/30/15/2114/2390096).\n",
    "\n",
    "Before and after trimming, general Quality Controll (QC( of a FastQ file is assessed with a tool called FastQC. The FastQC reports are rendered as html pages so we can view them in our webbrowsers. You can find the FastQC reports for this practical here:\n",
    "\n",
    "### Filtering\n",
    "Finally, we chose here to filter the sequencing data for plant DNA, since we are only interested in microbial DNA. The sequencing data was rid of any plant DNA by mapping (aligning) the reads to a reference plant genome. Only the sequencing data that did not map anything was kept for further analysis.\n",
    "\n",
    "### laura add fastQC links\n",
    "\n",
    "\n",
    "## Assembly\n",
    "Assmbly is the process of combining these many short DNA reads in longer contiguous strings of DNA: contigs. For the assembly process we have used both sample types and all biological replicates as if they were one big sample. This yields the best assembly results. A metagenome assembly is different than a regular genome assembly in that it has DNA of multiple species in one sequencing file. Metagenome assembly was achieved with [SPAdes](http://online.liebertpub.com/doi/abs/10.1089/cmb.2012.0021). \n",
    "\n",
    "\n",
    "Some more elaborate information of Assembly\n",
    "> The process of assembly takes the millions of short reads in a fastq file, and processes these to thousands of longer nucleotide sequences. Typically the output is a fasta file with contiguous sequences called contigs (contigs.fasta) or a fasta file with multiple contigs scaffolded based on their relative position and orientation to each other called scaffolds, in a file scaffolds.fasta. Ideally, these scaffolds represent the genome of a single species in the metagenome. This is however, extremely unlikely. At least in my personal experience. In reality the assembly process wil produce thousands of longer scaffolds where a single species' genome is still represented by multiple sequences in the assembly fasta file.\n",
    "\n",
    "\n",
    "# Here you will take over the workflow\n",
    "We have provided you with a subset of the assembly and a subset of the reads. We have made subsets rather than the whole sets to keep this one day practical feasible.\n",
    "\n",
    "The complete workflow you will follow, is seen on this sketch: \n",
    "![image](./metagenomics/workflowsketch.png)\n",
    "\n",
    "After assembly of the contigs we want to know how abundant each contig is in the different metagenomic samples. To do this, we align the illumina reads to the contigs in a step called **backmapping** with a tool called BWA.\n",
    "\n",
    "We will first do some exercises to learn how to work in a notebook like this then we will continue the metagenomics workflow at the backmapping step. Whenever you are lost in commandline, error messages and output tablaes, have a look again at this flowchart. In these bio-informatics workflows it is essential to keep a feeling for where you are, where you are going going and where you come from. \n",
    "\n",
    "The programmes mentioned in the flowchart are preinstalled on this virtual machine you are currently working on:\n",
    "bwa: http://bio-bwa.sourceforge.net/ <br>\n",
    "samtools: http://samtools.sourceforge.net/ <br>\n",
    "binningmetabat: https://bitbucket.org/berkeleylab/metabat/overview <br>\n",
    "checkm: https://github.com/Ecogenomics/CheckM/wiki/Installation#how-to-install-checkm <br>\n",
    "prokka: http://www.vicbioinformatics.com/software.prokka.shtml <br>\n",
    "\n",
    "\n",
    "\n",
    "# todo: from here on filter out redundant data\n",
    "\n",
    "\n",
    "## binning\n",
    "A single metagenome assembly contains contigs representing DNA of many species in the original sample, but without any information on which contigs belongs to which species.\n",
    "\n",
    "\n",
    "To tackle this issue, we will categorise each individual scaffold in a 'bin' which then ideally comprises one species in the metagenome. This procedure is called binning. The binning procudure starts with 'backmapping'. This means mapping the original reads from each biological replicate separately to the assembly. For this, we use BWA (Burrows-wheeler alignment) which then produces a bam file containing all reads and their location on the scaffolds. These bam files then must be sorted for further processing. Then metabat uses these sorted bamfile to subdivide the scaffolds in bins. This will produce between 5 and 20 bins. Hence we have moved from millions of very short sequences with very little biological information (to lets say about) 10 long sequences with a lot of biological information. \n",
    "\n",
    "<h3> Annotation </h3>\n",
    "One bin must then ideally represent one species. The check wether this is the case, we can use checkM. CheckM assesses the completeness and contamination of bins based on marker genes (collocated sets of genes that are ubiquitous and single-copy within a phylogenetic lineage). \n",
    "\n",
    "For the sake of simplicity and time, we will isolate one bin per group and annotate this bin using prokka. Annotation is the process of adding functional information to parts of DNA sequence, i.e. making a table of genes and locations of these genes. These are often stored in a GFF file, the genomic feature format. Once you have this gff file, and a fasta for your specific bin/microbial species, we will visualise it using KEGG. Again, for more details please google kegg. Here we will use it to load our gff file and then explore the metabolic pathways present. \n",
    "\n",
    "<h2> Practical information </h2>\n",
    "This whole computer practicum will take a whole day, staff will be present all day to help you in this process. However, we are with very few so please assisst each other where possible. \n",
    "\n",
    "You will work in this python notebook, combining bash and python commands. Bash commands can be executed by appending an exclamation mark '!' to the command like so:\n",
    "\n",
    "> !whoami\n",
    "\n",
    "Or by typing \"%%bash\" at the start of your code block, like so:\n",
    "\n",
    "> %%bash\n",
    "\n",
    "\n",
    "You are working on the virtual machine, which you have used before, did you know you can check the computer power of this machine using bash commands:\n",
    "\n",
    "number of cpu threads\n",
    "\n",
    "> !nproc\n",
    "\n",
    "available GB RAM memory\n",
    "\n",
    "> !top\n",
    "\n",
    "\n",
    "The first bash commands we will use are 'mkdir' 'wget' and 'ls'. Go ahead and see what they do. If you wonder, most programms print a short summary of their function when presented the -h flag like so:\n",
    "\n",
    "> !ls -h\n",
    "\n",
    "or \n",
    "\n",
    "> !ls --help\n",
    "\n",
    "or \n",
    "\n",
    "> !mkdir -h\n",
    "\n",
    "\n",
    "If you made a directory you want to remove again you can use:\n",
    "\n",
    "> !rm -rf directory\n",
    "\n",
    "but be carefull with the rm command because you will not be able to recover the removed file.\n",
    "\n",
    "\n",
    "\n",
    "Not all steps are feasible on these humble web based machines. And some may fail for other reasons. Therefore, results of all steps are uploaded to a separate web server. Instructions on accessing this webserver will follow .\n",
    "\n",
    "\n",
    "Good luck! and remember Google is your friend!!!\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# How to work in a Jupyter notebook with the Bash language\n",
    "The webpage you are looking at, is called a jupyter notebook. It is a webpage on which you can write text (like this text) and also code. The code can ben executed an the output you see again within the same notebook! This may sound trivial, but it's really \"cool\". To put it in non-scientific terms. Code and text are entered in individual cells, a code cell or a text cell. This cell you are reading now is obviously a text cell. Now let's look at a code cell."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hello world\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"hello world\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So what happened in the above code cell. First I declared that I want to write code in bash by typing\n",
    "> %%bash\n",
    "\n",
    "then the first word you type in bash is always the command: in this case this is \n",
    "> echo\n",
    "\n",
    "this command 'echoes' whatever you give it in the terminal.\n",
    "\n",
    "after you 'call' the command, you give it an argument in this case that is \n",
    "> \"hello world\"\n",
    "\n",
    "This basic structure of 'command' 'arguments' commes back through this practical. \n",
    "\n",
    "*do* now try to change \"hello world\" to something else"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Often, an argument is a path to a file. To see what files we have, we have the `ls` command. (short for list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COO1_C002.ipynb               dna181_circos.conf\r\n",
      "COO3.ipynb                    dna181_denovo.vcf\r\n",
      "COO4+Pipeline+building.ipynb  dna181_illumina.bam_CNVs\r\n",
      "COO5a.ipynb                   \u001b[0m\u001b[01;36mHomo_sapiens.GRCh37.GATK.illumina\u001b[0m@\r\n",
      "COO5b.ipynb                   \u001b[01;34mmetagenomics\u001b[0m/\r\n",
      "COO6B_github.ipynb            metagenomics.ipynb\r\n",
      "COO6.ipynb                    NGS analyses.ipynb\r\n",
      "\u001b[01;34mCWLs\u001b[0m/                         sv.links\r\n",
      "\u001b[01;36mdatafiles\u001b[0m@                    \u001b[01;32mWES_analysis_pipeline.sh\u001b[0m*\r\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Files are black, folders are purple and trailed by a '/'   Now lets use the `ls` command to see what is in the metagenomics folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "workflowsketch.png\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls metagenomics/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are learning fast. So we now know what a `command` is, we know what an `argument` is. Finally we also need to know what options are. options are often provided in between the `command` and the `argument`. They look either like this\n",
    "> ls --size --human-readable\n",
    "\n",
    "or in shortened versions like this\n",
    "> ls -sh\n",
    "\n",
    "Note that the above two command are synonimous. Try out in the cell below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 640K\n",
      "4.0K assembly\n",
      "636K workflowsketch.png\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "ls --size --human-readable metagenomics/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Raw data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b> sequencing reads </b><br>\n",
    "The sequencing reads are already available in the folder 'metagenomics/reads', to keep the practical feasible we have made a subset of the original data. <br>\n",
    "\n",
    "In this reads folder you will find 12 files. <br>\n",
    "L and P stand for the leaf and plant samples. <br>\n",
    "L1 L2 L3 are the biological replicates. R1 and R2 represent the forward and reverse illumina reads. So 2 samples * 3 replicates * 2 directions = 12 files <br>\n",
    "Files are stored in a fastq format (.fastq) and compressed in gzip archives (.gz). The gzip compression format is widely used in the genomics field, fastq files often compress down to only a quarter of the filesize with gzip! You do not need to extract these. <br>\n",
    "\n",
    "A good practice is to always peek into a file just to check the format. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This command allows you to see the files in the reads directory.   'ls' is short for 'list'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L1.R1.fastq.gz\tL2.R2.fastq.gz\tP1.R1.fastq.gz\tP2.R2.fastq.gz\r\n",
      "L1.R2.fastq.gz\tL3.R1.fastq.gz\tP1.R2.fastq.gz\tP3.R1.fastq.gz\r\n",
      "L2.R1.fastq.gz\tL3.R2.fastq.gz\tP2.R1.fastq.gz\tP3.R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls metagenomics/reads/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To view in a gzipped file you can use the zcat command. \n",
    "\n",
    "The '|' sign is called a 'pipe' You can use it to feed the output of one programme to the next. Never forget to pipe the output to head when viewing/reading a file, like this \"| head\". The files we work with are big and will crash your jupyter-notebook instance if you don't use head."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@NS500813:28:H3M3VAFXX:1:11101:21566:1019 1:N:0:CAGATC\r\n",
      "AAAAAATTAAAATAATTGGCTGAAAATGAAAAAAGAAAGCTTGATCATATGTTAAGCCTTTTGTAACGGAACAAAGTGATTAAATTATCTCTACAGTGTTTTTTCGATTTATTGATGGATTAAGAACAACCGTCACACACCAATAGAATAT\r\n",
      "+\r\n",
      "AAAAAEEEE6EEEEEEEEEEEEEEEEEEEEEAEEEEEEEEE</EE<EEEEEEEE/AA6E<E/E/</EEEAE/EEEEEEE<EEE//EEEEE<EE/EEAE<E<E//AE/E<EE/<AEEEAE66<E<<E</<AA<E/AEE/E//<<EEEEE/AA\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:14840:1067 1:N:0:CAGATC\r\n",
      "GTGGGATTTCCAGTGATTTGAATTCTTACAAAAGTCGCTTGCTTTTGTTTTTATTGCCACCTATTTCGTTGAATAATTATTAGAAAGGGAACAAGGCTTTTTGACGATAAAAAGTTATTTTTAATTGGAGGTCGCTAGAATAGCTTTGTT\r\n",
      "+\r\n",
      "A/AAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEAEEAEEEEEEEEEEEEEAEEE/EE/EEEEEEAEEEEEEEEEAEEEEEEEEEA/EEEAEAA<EEEEEEEEEAAEA/A/AA/6A<E<AE/EA6<A\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:20510:1074 1:N:0:CAGATC\r\n",
      "CCCGGTGCTGAATGAGAATGATTCGTGTTTAGATTAGCAGGCGCTGATGGTAGTAGTAGCGGCAATGAGAGTGATTCGTATTTAGATCCAGGCAGGGGCGGAAGTTTCATAATGTGGAATGCGCCCTGGTGCGCTTTACAATGTAAAGC\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "!zcat metagenomics/reads/L1.R1.fastq.gz | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using grep we can select for lines in a file that contain only the word/key specified in the grep command. As you can see in the above command, the headers of the reads always start with the \"@\", therefor if we only want to see the headers we can simply grep for the \"@\" using the command below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@NS500813:28:H3M3VAFXX:1:11101:21566:1019 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:14840:1067 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:20510:1074 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:5773:1086 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:2702:1014 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:6665:1015 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:6357:1086 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:18737:1028 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:1677:1028 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:26277:1020 1:N:0:CAGATC\r\n",
      "grep: write error: Broken pipe\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "!zcat metagenomics/reads/L1.R1.fastq.gz | grep '@' | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now if we want to count the number of reads in the fastq file we can count the number of headers, since every entry has one header. To count the number of headers we will first have to read in the gzip file, then grep on the headers, and then we can use the wordcount command to count the number of lines."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1000000\r\n"
     ]
    }
   ],
   "source": [
    "!zcat metagenomics/reads/L1.R1.fastq.gz | grep '@' | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### wrap up\n",
    "So now you have executed some bash commands and investigated the output. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Assembled data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Calculating a de-novo metagenome assembly is beyond the scope of this practical, so we have done this for you. You will however analyse this assembly yourself! :)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "View the inside of the assembly directory:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scaffolds.fasta.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls metagenomics/assembly/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now your first assignment is to dowload the assemblies from the data repository, adapting the commands above. Below you will find a code block with the template command that you can finish to download the the assemblies. Use multiple code blocks if convenient. You can make new code blocks by clicking on the left side of a code block and then type \"a\" or \"b\" to create a code block above or below the original code block. You can delete a code block by clicking on the left side of a code block and typing \"d\" twice."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Edit this line to have a look into the assembly file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gzip: metagenomics/assembly/....assemblyfile.gz: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "!zcat metagenomics/assembly/....assemblyfile | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Grep the headers of the assembly file, what word/key would work for fastas when viewing the assembly entries above"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!zcat metagenomics/assembly/....assemblyfile | grep \"\" | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Count the scaffolds in the assembly like we did earlier with the fastq.gz file. \n",
    "\n",
    "Remember not to over-load the webinterface!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!zcat metagenomics/assembly/....assemblyfile | grep \"\" ......"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Rename the assembly file:   (mv is short for move)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mv metagenomics/assembly/....oldname  metagenomics/assembly/....newname"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check if the rename worked:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!ls metagenomics/assembly/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Did your assembly downloaded correctly?<br>\n",
    "Does the content of the fasta make sense?<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2>Length distributions of the scaffolds</h2>\n",
    "Now that we have the assembly, we will do some quick analyses to get a idea of the quality. First we will plot the length distribution of the scaffolds in the assembly. Luckily for us, the length of each sequence in the fasta is already embeded in each fasta header. We can easily extract these numbers and plot them in python.\n",
    "\n",
    "To plot the length distribution run the python code below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scaffolds.fasta.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls ./metagenomics/assembly/\n",
    "gunzip --keep ./metagenomics/assembly/scaffolds.fasta.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAnQAAAJhCAYAAADIci3nAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMS4xLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvAOZPmwAAIABJREFUeJzt3X1UlHX+//HXwHgTYoCDoShrqbmllUijKVaCUna3pW2abdsvJcv7IutsLSe3+ibf7FumlVqsqe3qdzetNG3Lasm0FE0GwkytoLvNo4GAKd6FMNfvD37MTxR0UJiLj/N8nNM5zmfmuq739WYYXn2um3FYlmUJAAAAxgqxuwAAAACcGQIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdEAzM3r0aKWkpNhdRi3nn3++pk+f3iTrasx1H+/4XjZ1b5OSkjR27NgmW38g7d+/X8OHD9e5554rh8OhH374QZWVlUpNTZXL5ZLD4dDatWv9Wpc/P+Pm+L4HTOK0uwAAzcf06dP16quv6ocffgjYNnNychQWFubXa9evX6+rrrpK33//vc4///xTvv6FF16Q1+s9wwpPVF+fli9fLqfz7PhYffnll7Vx40Zt2LBB7du3V/v27fXmm2/qH//4h9asWaOuXbuqXbt2dpcJ4P85Oz55ABirffv2jb7Oo0ePyul0KiIiotHXfTJnU8ApKChQr169dOmll9Ya69SpkxITE22sDEBdOOQKGOD1119XfHy8WrdurfPPP19Tp07VwYMHfc/XHOp76qmn1KFDB7Vr106jR4+u9Rqv16v09HS1b99e4eHhGjVqlGbPnu2bUXrttdc0bdo0/fjjj3I4HHI4HHriiSd8y1dUVOiBBx5Qu3btFBMTo4cfflhVVVUnrXvLli1KTExU69at1aNHDy1btuyE1xx/OG7lypXq06ePwsLCFBkZqX79+unzzz/XDz/8oKuuukqSdMEFF8jhcCgpKUnS/z9c99JLL+n8889Xq1atdPDgwXoP4z3//PPq1KmTwsLC9Pvf/14lJSW+5+paZsmSJXI4HKfs0/GHXI8ePapHH31UnTp1UsuWLdWzZ0/94x//qLVuh8OhefPm6a677lLbtm0VFxen//mf/zlpX48ePaqpU6eqc+fOatWqlTp27KhRo0bVes3SpUt1+eWXq3Xr1nK5XLr++uu1d+9eSdK///1vJSUlqV27doqIiNCgQYO0efPmWj+TBQsWaM2aNb4+JyUladq0afruu+/kcDh8M6T+7OPx9u7dq9tvv11t2rRRTEyMHnvsMR3/pUXr16/XwIED1bZtW7Vt21a9e/fWBx98cNL1AsGMGTqgmXvttdf04IMP6sUXX9TAgQO1c+dOTZ48WXv27NHixYt9r3vzzTc1ZswYrV27Vj/88INGjRqlLl266Mknn5QkzZ49Wy+++KJefvll9e/fX++8847+67/+y7f87bffrq+++kr/+7//q5ycHElSeHi47/mXXnpJjzzyiD777DPl5eXpzjvvVK9evTRmzJg66z58+LBuuOEG9e7dW5999pkOHTqk+++/X8XFxfXu688//6wRI0Zo+vTpGjFihI4cOaLPP/9cTqdTcXFxWrlypW655RZt3rxZcXFxatmypW/ZzZs3q23btnr77bcVGhqq1q1b17mNzZs3KywsTO+//75KS0t17733KjU1VatWrfLjp3HqPh0rPT1dCxcu1CuvvKLevXvrzTff1B//+EfFxMRoyJAhvtc9+eSTmj59up544gm9++67euCBB9S3b18lJyfXud6XXnpJy5Yt05IlS9S1a1cVFRVpw4YNvucXLVqk++67T3/5y1+0ePFiVVZW6uOPP/YF8AMHDmjSpEnq3bu3jh49qlmzZum6665TQUGBXC6XcnJyNHnyZO3evVvLli3z9fm///u/9dZbb2njxo0KDQ1t0D4eKzU1VVu3btU777yjmJgYPf3001q1apX69esnSaqqqtLNN9+s0aNH67XXXpMkffnll34fmgeCkgWgWbn77rutIUOG+B536dLFevnll2u9Zt26dZYkq6yszLIsyxo0aJB16aWX1nrNuHHjrP79+/sex8bGWo899lit19x+++1WaGio7/FTTz1ldenS5YSaunTpYv3ud7+rNTZ06FBr1KhR9e7H/PnzrTZt2vhqtCzL2rp1qyXJeuqpp2qtu+ZxXl6eJcn6/vvv61znp59+Wufzd999txUREWGVl5efMH5sL++++26rTZs21i+//OIb++CDDyxJ1jfffFPnMpZlWYsXL7aO/bisr0+DBg2y7rnnHsuyLOvgwYNWy5Ytrblz59Z6zbBhw6zk5GTfY0nWlClTar3mt7/9rfXoo4/W2QPLsqz777/fSk5Otrxeb53Px8XFWZMmTap3+eNVVVVZkZGR1pIlS3xjdfXh8ccft7p16+Z77O8+HvszLigosCRZH374oe/5X3/91YqNjfVtr6yszJJkffzxx37vAxDsOOQKNGN79uzRjz/+qKlTpyo8PNz33/XXXy9JKiws9L02Pj6+1rKdOnVSUVGRpOorFnft2qX+/fvXes2AAQP8ruVk66/L9u3bdfHFFysqKso3dskll5z0vLbLLrtMQ4cO1SWXXKLhw4frhRde0E8//eRXfRdffHG9M2XH6tmzZ60aBg4cKEnasWOHX9vxV2FhoSoqKnT11VfXGh80aJC2bdtWa6yhvR0zZoy2bt2q7t27a/z48XrrrbdUUVEhSSouLtZPP/2ka6+9tt7lv//+e911113q3r27zj33XJ177rnat2+ffvzxxybbxxrbt2+XpFrn4bVs2VJ9+/b1PY6KitLYsWM1dOhQXX/99ZoxY4a+/vrrBtUGBBsCHdCM1Vyh+cILLyg/P9/335YtW1RQUFDrhPVjDz9K1edm1Sxv/b/zk2rOAzsdJ1t/XSzLavD2QkNDtXr1aq1Zs0Z9+/bVW2+9pR49euhf//rXKZdt06ZNg7ZVn5CQkBPO5zp69Ohpr+/4HtTVl4b2Nj4+Xt9//72ee+45tWzZUg888IDi4+O1f//+erd7rJtuukn/+c9/NHfuXG3atEn5+fk677zzfKGwofzZx2Of88f8+fOVm5ura665RuvWrdMll1yizMzM06oPCAYEOqAZi4mJUVxcnL7++mt17979hP/qO0/seBEREYqNjdXGjRtrjW/atKnW45YtW57yQgd/9erVS9u3b9cvv/ziG9u2bZv27dt30uUcDof69eun9PR0ffLJJxo0aJAWLVrkq0/SGdW4Y8eOWsEnOztbUvUMnySdd9552rVrV61l8vLyaj32p0/du3dXq1attG7dulrjn3zyiXr16nXa9dcIDw/X8OHD9eKLL8rj8WjHjh1at26dzjvvPHXu3LneCwhKS0u1fft2Pfrooxo6dKh69uyp1q1bn/Tcxvqczj7WjNf0Xaq+4KbmfMRjXXLJJZo6dapWr16te+65R3/9618bXCMQLLgoAmjmMjIydM899ygyMlLDhg1TixYttGPHDq1evbpBMxYPPfSQHn/8cV100UXq16+f3n33XX344Ye1ZlIuuOAC/fzzz9q4caMuvPBChYWFnfaJ6H/4wx80bdo0/fGPf1RGRoYOHz6sBx54QOecc069y2RnZ+ujjz7Stddeq44dO6qgoEBffPGF7rnnHklSly5dFBISovfee0+33367WrVq1eBbkzgcDv2f//N/NH36dJWVlWnSpEm68cYbdeGFF0qSUlJS9Mwzz2jOnDm6/vrrtWbNmhOuzvWnT2FhYbr//vs1bdo0tW/fXvHx8XrjjTe0cuVK/fvf/25Qzcd79tlnFRsbq/j4eIWFhemf//ynQkND1aNHD0nS448/rgkTJigmJka33XabvF6vPv74Y40aNUrt2rVT+/btNX/+fHXr1k2lpaX605/+dNKfS31OZx+7d++um2++WZMmTVJmZqZiYmI0Y8YMlZeX+15TWFio+fPn63e/+53i4uK0a9cuffrpp0pISDi9hgFBgBk6oJm76667tGzZMr377rvq16+f+vbtqyeeeEKdOnVq0HrS0tI0efJkPfDAA+rTp482bdqkhx56qNYs37BhwzRixAjdeOONat++/Slvn3EyYWFheu+991RaWqp+/frpzjvv1IMPPqjzzjuv3mUiIiK0ceNG3XLLLbrwwguVmpqqO++8U9OmTZMk3xWRM2bMUMeOHXXLLbc0uK5+/frpyiuv1DXXXKOhQ4eqV69evhlAqTrQTZ8+XU8//bR69+6tNWvW6C9/+Uutdfjbp4yMDN17771KS0tTr169tGTJEi1ZsqTeqz/9de655+r555/XgAEDdOmll2rFihV666239Nvf/laSNHbsWL322mt68803FR8fr6uvvlqrV6+W0+lUSEiI3njjDX377be67LLLNHr0aKWlpaljx46nVcvp7OPChQsVHx+vm266SYMGDVKnTp00fPhw3/Nt2rRRQUGBRo0apR49euj3v/+9EhMTNWfOnNOqEQgGDsvfExoAnHVSU1O1ZcsW5ebm2l0KAOAMcMgVCBK7du3SihUrlJycrNDQUL3zzjv6+9//zqwHAJwFmKEDgkRRUZFuv/12ffHFFzpy5Ii6d++uKVOm6N5777W7NADAGSLQAQAAGI6LIgAAAAxHoAMAADAcgQ4AAMBwZ9VVrsff3b2xRUdHq6SkpEm3YQL6QA9q0Idq9IEe1KAP9KDGqfoQGxvbqNtjhg4AAMBwBDoAAADDEegAAAAMR6ADAAAwHIEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxnfKDzeDzKzMy0uwwAAADbOO0u4Ey53W653W67ywAAALCN8TN0AAAAwY5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOOO/+iuQioYnnjAWOn+VDZUAAAD8f8zQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhmuV96Lxer5YuXarDhw+ra9euSkpKsrskAACAZitggW7evHnKy8tTRESEZs6c6RvPz8/XokWL5PV6NWTIEA0bNkwej0dlZWUKDw+Xy+UKVIkAAABGCtgh16SkJKWnp9ca83q9WrBggdLT0zVr1ixt2LBBO3fu1K5du9SjRw/dfffd+vDDDwNVIgAAgJECFuh69uyp8PDwWmOFhYXq0KGDYmJi5HQ6lZiYqJycHLVr105t2rSpLjCE0/wAAABOxtZz6MrKymodUnW5XCooKNANN9yghQsX6quvvtLFF19c7/JZWVnKysqSJM2YMUPR0dFNWm9RHWNNvc3myOl0BuV+H4seVKMP1egDPahBH+hBjUD3wdZAZ1nWCWMOh0OtWrXShAkTTrl8SkqKUlJSfI9LSkoatT5/2LFNu0VHRwflfh+LHlSjD9XoAz2oQR/oQY1T9SE2NrZRt2fr8UyXy6XS0lLf49LSUkVFRdlYEQAAgHlsDXTdunXT7t27VVxcrMrKSmVnZ8vtdttZEgAAgHECdsh19uzZ2r59u8rLyzV+/HiNHDlSgwcPVmpqqjIyMuT1epWcnKy4uLgGrdfj8Sg3N1fjxo1rosoBAACat4AFurS0tDrHExISlJCQcNrrdbvdzOoBAICgxj1BAAAADEegAwAAMJzxgc7j8SgzM9PuMgAAAGxj633oGgPn0AEAgGBn/AwdAABAsCPQAQAAGI5ABwAAYDjjAx0XRQAAgGDHRREAAACGM36GDgAAINgR6AAAAAxHoAMAADCc8YGOiyIAAECw46IIAAAAwxk/QwcAABDsCHQAAACGI9ABAAAYjkAHAABgOOMDHVe5AgCAYMdVrgAAAIYzfoYOAAAg2BHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxnfKDjPnQAACDYcR86AAAAwxk/QwcAABDsCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDjjAx3fFAEAAIId3xQBAABgOONn6AAAAIIdgQ4AAMBwBDoAAADDEegAAAAMR6ADAAAwHIEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxnfKDzeDzKzMy0uwwAAADbOO0u4Ey53W653W67ywAAALCN8TN0AAAAwY5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4p90F1GXbtm1aunSpOnfurIEDB6pXr152lwQAANBsBSzQzZs3T3l5eYqIiNDMmTN94/n5+Vq0aJG8Xq+GDBmiYcOGyeFwqHXr1jp69KhcLlegSgQAADBSwA65JiUlKT09vdaY1+vVggULlJ6erlmzZmnDhg3auXOnLrroIqWnp+vOO+/UsmXLAlUiAACAkQIW6Hr27Knw8PBaY4WFherQoYNiYmLkdDqVmJionJwchYRUlxUeHq6jR48GqkQAAAAj2XoOXVlZWa1Dqi6XSwUFBfrss8+0ZcsWHTx4UNddd52NFQIAADR/tgY6y7JOGHM4HLriiit0xRVXnHL5rKwsZWVlSZJmzJih6OjoRq/xWEV1jDX1Npsjp9MZlPt9LHpQjT5Uow/0oAZ9oAc1At0HWwOdy+VSaWmp73FpaamioqL8Xj4lJUUpKSm+xyUlJY1anz/s2KbdoqOjg3K/j0UPqtGHavSBHtSgD/Sgxqn6EBsb26jbs/U+dN26ddPu3btVXFysyspKZWdny+1221kSAACAcQI2Qzd79mxt375d5eXlGj9+vEaOHKnBgwcrNTVVGRkZ8nq9Sk5OVlxcXIPW6/F4lJubq3HjxjVR5QAAAM1bwAJdWlpaneMJCQlKSEg47fW63W5m9QAAQFDjq78AAAAMR6ADAAAwnPGBzuPxKDMz0+4yAAAAbGPrbUsaA+fQAQCAYGf8DB0AAECwI9ABAAAYjkAHAABgOOMDHRdFAACAYMdFEQAAAIYzfoYOAAAg2BHoAAAADEegAwAAMJzxgY6LIgAAQLDjoggAAADDGT9DBwAAEOwIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhjA903LYEAAAEO25bAgAAYDjjZ+gAAACCHYEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHDGBzruQwcAAIId96EDAAAwnPEzdAAAAMGOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhjM+0PFNEQAAINjxTREAAACGM36GDgAAINgR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAMR6ADAAAwHIEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHDGBzqPx6PMzEy7ywAAALCN0+4CzpTb7Zbb7ba7DAAAANsYP0MHAAAQ7Ah0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIZrtoHuyJEjeuSRR5Sbm2t3KQAAAM2aM1AbmjdvnvLy8hQREaGZM2f6xvPz87Vo0SJ5vV4NGTJEw4YNkyStXLlSAwYMCFR5AAAAxgrYDF1SUpLS09NrjXm9Xi1YsEDp6emaNWuWNmzYoJ07d+qLL75Q586dFRkZGajyAAAAjBWwGbqePXuquLi41lhhYaE6dOigmJgYSVJiYqJycnJ05MgR/frrr9q5c6datmypPn36KCSk2R4dBgAAsFXAAl1dysrK5HK5fI9dLpcKCgp0zz33SJLWrl2rtm3b1hvmsrKylJWVJUmaMWOGoqOjm7TeojrGmnqbzZHT6QzK/T4WPahGH6rRB3pQgz7QgxqB7oOtgc6yrBPGHA6H799JSUknXT4lJUUpKSm+xyUlJY1Wm7/s2KbdoqOjg3K/j0UPqtGHavSBHtSgD/Sgxqn6EBsb26jbs/U4psvlUmlpqe9xaWmpoqKibKwIAADAPLYGum7dumn37t0qLi5WZWWlsrOz5Xa77SwJAADAOAE75Dp79mxt375d5eXlGj9+vEaOHKnBgwcrNTVVGRkZ8nq9Sk5OVlxcXIPW6/F4lJubq3HjxjVR5QAAAM1bwAJdWlpaneMJCQlKSEg47fW63W5m9QAAQFDjXiAAAACGI9ABAAAYzvhA5/F4lJmZaXcZAAAAtrH1PnSNgXPoAABAsDN+hg4AACDYEegAAAAMZ3yg4xw6AAAQ7DiHDgAAwHDGz9ABAAAEOwIdAACA4Qh0AAAAhiPQAQAAGM74QMdVrgAAINhxlSsAAIDhjJ+hAwAACHYEOgAAAMMR6AAAAAxHoAMAADCc8YGOq1wBAECw4ypXAAAAwxk/QwcAABDsCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhjM+0HEfOgAAEOy4Dx0AAIDhjJ+hAwAACHYEOgAAAMMR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAMR6ADAAAwnPGBjm+KAAAAwY5vigAAADCc8TN0AAAAwY5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhjM+0Hk8HmVmZtpdBgAAgG2cdhdwptxut9xut91lAAAA2Mb4GToAAIBg53egW7VqVZ3j//rXvxqtGAAAADSc34HurbfeatA4AAAAAuOU59B9+eWXkiSv1+v7d42ioiKdc845TVMZAAAA/HLKQPfyyy9LkioqKnz/liSHw6HIyEilpqY2XXUAAAA4pVMGurlz50qS5syZo8mTJzd5QQAAAGgYv29bcmyY83q9tZ4LCeFiWQAAALv4Hei+++47LViwQP/5z39UUVFR67mlS5c2emEAAADwj9+Bbu7cubr88ss1YcIEtWrVqilrAgAAQAP4HehKSkp0xx13yOFwNGU9AAAAaCC/T37r27evtmzZ0pS1AAAA4DT4PUN39OhRPffcc7rooosUGRlZ6zmufgUAALCP34Guc+fO6ty5c1PWAgAAgNPgd6AbMWJEU9YBAACA0+R3oDv+a7+OdckllzRKMQAAAGg4vwPdsV/7JUn79+9XZWWlXC6X5syZ0+iFAQAAwD8Nug/dsbxer9566y2dc845jV4UAAAA/Hfa39kVEhKiW2+9VStXrmzMegAAANBAfs/Q1eWLL75oku9x3blzp9577z2Vl5fr0ksv1bXXXtvo2wAAADhb+B3oJkyYUOtxRUWFKioqNHbsWL+WnzdvnvLy8hQREaGZM2f6xvPz87Vo0SJ5vV4NGTJEw4YNU+fOnXXffffJ6/UqMzPT3xIBAACCkt+BbsqUKbUet2rVSh07dlRYWJhfyyclJem6666rdS6e1+vVggUL9Nhjj8nlcunPf/6z3G63OnfuLI/Ho7ffflvXXXedvyUCAAAEJb8DXc+ePSVVh7B9+/YpIiKiQYdbe/bsqeLi4lpjhYWF6tChg2JiYiRJiYmJysnJUefOneV2u+V2u/X000/ryiuv9Hs7AAAAwcbvQHf48GEtWLBA2dnZqqqqUmhoqBITE5Wamur3LN3xysrK5HK5fI9dLpcKCgq0bds2ffbZZ6qsrFSfPn3qXT4rK0tZWVmSpBkzZig6Ovq06vBXUR1jTb3N5sjpdAblfh+LHlSjD9XoAz2oQR/oQY1A98HvQLdw4UIdOXJEzz33nNq3b689e/bo9ddf18KFC0/7u1wtyzphzOFwqFevXurVq9cpl09JSVFKSorvcUlJyWnVcSbs2KbdoqOjg3K/j0UPqtGHavSBHtSgD/Sgxqn6EBsb26jb8/uYaX5+vqZMmaLY2Fi1aNFCsbGxmjhxorZs2XLaG3e5XCotLfU9Li0tVVRU1GmvDwAAIBj5Hehatmyp/fv31xrbv3+/nM7Tv/NJt27dtHv3bhUXF6uyslLZ2dlyu92nvT4AAIBg5HcaGzx4sKZPn64bb7zRd8j13Xff1ZAhQ/xafvbs2dq+fbvKy8s1fvx4jRw5UoMHD1ZqaqoyMjLk9XqVnJysuLi4Bu2Ax+NRbm6uxo0b16DlAAAAzhZ+B7pbb71V7dq10/r161VWVqZ27drplltu0eDBg/1aPi0trc7xhIQEJSQk+FvGCWquhgUAAAhWfge6RYsWaeDAgZo2bZpv7Ouvv9Zrr72m0aNHN0VtAAAA8IPf59Bt2LBB3bp1qzXWtWtXrV+/vtGLAgAAgP/8DnQOh0Ner7fWmNfrrfPWI4Hk8Xj4ejAAABDU/A50F110kV5//XVfqPN6vXrjjTd00UUXNVlx/nC73VwQAQAAgprf59CNGTNGM2bM0Lhx43w3y4uKitIjjzzSlPUBAADgFPwOdC6XS88884wKCwtVWloql8ul7t27N+j7XAEAAND4GnRX4JCQEPXo0aOpagEAAMBpMH56jYsiAABAsDv97+1qJrixMAAACHbGz9ABAAAEOwIdAACA4Qh0AAAAhjM+0HFRBAAACHZcFAEAAGA442foAAAAgh2BDgAAwHAEOgAAAMMR6AAAAAxnfKDjKlcAABDsuMoVAADAcMbP0AEAAAQ7Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4YwPdNyHDgAABDvuQwcAAGA442foAAAAgh2BDgAAwHAEOgAAAMMR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAMZ3yg45siAABAsOObIgAAAAxnfKCzW9W9N9c5Hjp/VYArAQAAwcr4Q64AAADBjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhjA90Ho9HmZmZdpcBAABgG6fdBZxFi0etAAAXk0lEQVQpt9stt9ttdxkAAAC2MX6GDgAAINgR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAMR6ADAAAwHIEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAM57S7gPps3rxZeXl52r9/v4YOHarevXvbXRIAAECzFNBAN2/ePOXl5SkiIkIzZ870jefn52vRokXyer0aMmSIhg0bpn79+qlfv346cOCAFi9eTKADAACoR0APuSYlJSk9Pb3WmNfr1YIFC5Senq5Zs2Zpw4YN2rlzp+/55cuXa+jQoYEsEwAAwCgBDXQ9e/ZUeHh4rbHCwkJ16NBBMTExcjqdSkxMVE5OjizL0pIlSxQfH6+uXbsGskwAAACj2H4OXVlZmVwul++xy+VSQUGBVq9era1bt+rQoUP6+eefde21156wbFZWlrKysiRJM2bMUHR0dJPWWtSA1zZ1LXZyOp1n9f75gx5Uow/V6AM9qEEf6EGNQPfB9kBnWdYJYw6HQzfccINuuOGGky6bkpKilJQU3+OSkpJGr+90NadaGlt0dPRZvX/+oAfV6EM1+kAPatAHelDjVH2IjY1t1O3ZftsSl8ul0tJS3+PS0lJFRUXZWBEAAIBZbA903bp10+7du1VcXKzKykplZ2fL7XbbXRYAAIAxAnrIdfbs2dq+fbvKy8s1fvx4jRw5UoMHD1ZqaqoyMjLk9XqVnJysuLg4v9fp8XiUm5urcePGNWHlAAAAzVdAA11aWlqd4wkJCUpISDitdbrdbmb0AABAULP9kCsAAADODIEOAADAcMYHOo/Ho8zMTLvLAAAAsI3t96E7U831HLqqe28+YSx0/iobKgEAAGc742foAAAAgh2BDgAAwHAEOgAAAMMZH+i4KAIAAAQ7LooAAAAwnPEzdAAAAMGOQAcAAGA4Ah0AAIDhjA90XBQBAACCHRdFAAAAGM74GToAAIBgR6ADAAAwHIEOAADAcAQ6AAAAwxHoAAAADGd8oOO2JQAAINhx2xIAAADDGT9DBwAAEOwIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhjA903LYEAAAEO25bAgAAYDjjZ+gAAACCHYEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxnfKDjmyIAAECw45siAAAADGf8DB0AAECwI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYz/psiTFJ1780njIXOX2VDJQAA4GzCDB0AAIDhCHQAAACGI9ABAAAYzvhA5/F4lJmZaXcZAAAAtjH+ogi32y232213GQAAALYxfoYOAAAg2BHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAMZ/xXf52Nqu69+YSx0PmrbKgEAACYgBk6AAAAwxHoAAAADEegAwAAMByBDgAAwHBcFGGzui6AAAAAaAhm6AAAAAxHoAMAADBcszzkWlRUpOXLl+vQoUN66KGH7C4HAACgWQvYDN28efM0duzYEwJafn6+HnjgAU2ZMkVvv/22JCkmJkYTJkwIVGkAAABGC1igS0pKUnp6eq0xr9erBQsWKD09XbNmzdKGDRu0c+fOQJUEAABwVghYoOvZs6fCw8NrjRUWFqpDhw6KiYmR0+lUYmKicnJyAlUSAADAWcHWc+jKysrkcrl8j10ulwoKClReXq5//vOf+uGHH7RixQoNHz68zuWzsrKUlZUlSZoxY4aio6ObtN6iJl37yTX1vjWE0+lsVvXYgR5Uow/V6AM9qEEf6EGNQPfB1kBnWdYJYw6HQ23bttV99913yuVTUlKUkpLie1xSUtKo9TUnzWnfoqOjm1U9dqAH1ehDNfpAD2rQB3pQ41R9iI2NbdTt2XrbEpfLpdLSUt/j0tJSRUVF2VgRAACAeWwNdN26ddPu3btVXFysyspKZWdny+12N2gdHo9HmZmZTVQhAABA8xewQ66zZ8/W9u3bVV5ervHjx2vkyJEaPHiwUlNTlZGRIa/Xq+TkZMXFxTVovW63u8EhEAAA4GwSsECXlpZW53hCQoISEhICVQYAAMBZh6/+AgAAMByBDgAAwHDGBzouigAAAMHO1vvQNQYuigAAAMHO+Bk6AACAYEegAwAAMJzxgY5z6AAAQLDjHDoAAADDGT9DBwAAEOwIdAAAAIYj0AEAABiOQAcAAGA44wMdV7kCAIBgx1WuAAAAhjN+hg4AACDYEegAAAAMR6ADAAAwHIEOAADAcMZfFOHxeJSbm6tx48bZXUrAVd178wljofNX2VAJAACwk/GBjqtcAQBAsOOQKwAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGM/62JcFyH7q67jnn7+v8vTedv9vQimz/XgcAAALC+EDHfegAAECw45ArAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4428sHCzfFHEmzuTbIwAAQPNnfKDjmyIAAECw45ArAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOKfdBZwpj8ej3NxcjRs3zu5SjFJ17812lwAAABqJ8YHO7XbL7XbbXQYAAIBtOOQKAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOGcdhdQlyNHjujVV1+V0+lUr169dNVVV9ldEgAAQLMVsEA3b9485eXlKSIiQjNnzvSN5+fna9GiRfJ6vRoyZIiGDRumzZs3q3///nK73Zo1axaBDgAA4CQCdsg1KSlJ6enptca8Xq8WLFig9PR0zZo1Sxs2bNDOnTtVWlqq6Ojo6gJDOCoMAABwMgFLSz179lR4eHitscLCQnXo0EExMTFyOp1KTExUTk6OXC6XSktLJUmWZQWqRAAAACPZeg5dWVmZXC6X77HL5VJBQYGuv/56LVy4UHl5ebr88svrXT4rK0tZWVmSpBkzZvhm9ZpKUZOu3RxFwxP9el3MiuwzWt5f9W2nKbdb1IDt1ruOM+xjU6urvuNrcTqdTf57d7J66mJHv5xOp6ruvblZ1GKXQL4XmjP6YF4P/PmsOx2B7oOtga6u2TeHw6HWrVtr4sSJp1w+JSVFKSkpvsclJSWNWh/OTKB+Hnb93M/2/avL8bVER0c3q/oke/pV34d2c+tNU2qO7wU70IezoweNUf+p+hAbG3vG2ziWrSeoHXtoVZJKS0sVFRVlY0UAAADmsTXQdevWTbt371ZxcbEqKyuVnZ0tt9ttZ0kAAADGCdgh19mzZ2v79u0qLy/X+PHjNXLkSA0ePFipqanKyMiQ1+tVcnKy4uLiGrRej8ej3NxcjRs3rokqBwAAaN4CFujS0tLqHE9ISFBCQsJpr9ftdjOrBwAAgho3eQMAADAcgQ4AAMBwxgc6j8ejzMxMu8sAAACwja33oWsMnEMHAACCnfEzdAAAAMGOQAcAAGA4Ah0AAIDhjA90XBQBAACCHRdFAAAAGM74GToAAIBgR6ADAAAwHIEOAADAcMYHOi6KAAAAwY6LIgAAAAxn/AwdAABAsCPQAQAAGI5ABwAAYDgCHQAAgOGMD3Rc5QoAAIKdw7Isy+4iAAAAcPqMn6ELpEcffdTuEpoF+kAPatCHavSBHtSgD/SgRqD7QKADAAAwHIEOAADAcKFPPPHEE3YXYZKuXbvaXUKzQB/oQQ36UI0+0IMa9IEe1AhkH7goAgAAwHAccgUAADCc0+4CTJGfn69FixbJ6/VqyJAhGjZsmN0lNVhJSYnmzp2rX375RQ6HQykpKbrhhhu0bNkyffTRRzr33HMlSXfccYcSEhIkSStWrNCaNWsUEhKiMWPGKD4+XlL9/SguLtbs2bN14MABXXDBBZoyZYqcTqeOHj2qOXPm6LvvvlPbtm2Vlpam8847z5Y+TJo0Sa1bt1ZISIhCQ0M1Y8YMHThwQLNmzdKePXvUvn17PfjggwoPD5dlWVq0aJE+//xztWrVShMnTvRNoa9du1bLly+XJN16661KSkqSJH333XeaO3euKioq1KdPH40ZM0YOh6Pebdhh165dmjVrlu9xcXGxRo4cqYMHD57174V58+YpLy9PERERmjlzpiTZ+vM/2TYC2YPFixcrNzdXTqdTMTExmjhxotq0aaPi4mI9+OCDio2NlSRdeOGFuu+++xp9X+vrZ6D7YPfnYX3bCGQPZs2apV27dkmSDh06pLCwMD377LNn7Xuhvr+Nxn0uWDilqqoqa/LkydbPP/9sHT161Hr44Yetn376ye6yGqysrMz69ttvLcuyrEOHDln333+/9dNPP1lLly61Vq5cecLrf/rpJ+vhhx+2KioqrKKiImvy5MlWVVXVSfsxc+ZMa/369ZZlWVZmZqb1wQcfWJZlWe+//76VmZlpWZZlrV+/3nr++ecDsct1mjhxorVv375aY4sXL7ZWrFhhWZZlrVixwlq8eLFlWZaVm5trZWRkWF6v1/r666+tP//5z5ZlWVZ5ebk1adIkq7y8vNa/LcuyHn30Uevrr7+2vF6vlZGRYeXl5Z10G3arqqqyxo4daxUXFwfFe2Hbtm3Wt99+a02dOtU3ZufPv75tBLoH+fn5VmVlpa/WmvqKiopqve5YjbWvJ+tnU6qrD3b+DtS3jUD34Fh/+9vfrDfeeMOyrLP3vVDf30bTPhc45OqHwsJCdejQQTExMXI6nUpMTFROTo7dZTVYVFSUL+Gfc8456tSpk8rKyup9fU5OjhITE9WiRQudd9556tChgwoLC+vth2VZ2rZtm/r37y9JSkpK8vXJ4/H4/k+lf//++vLLL2U1o9M3c3JyNGjQIEnSoEGDatV99dVXy+FwqEePHjp48KD27t2r/Px8XXbZZQoPD1d4eLguu+wy5efna+/evTp8+LB69Oghh8Ohq6++2reu+rZht61bt6pDhw5q3759va85m94LPXv2PGFm1M6ff33bCHQPevfurdDQUElSjx49TvrZIKlR97W+fja1uvpQn0D8DtS3jaZ0sh5YlqWNGzdq4MCBJ12H6e+F+v42mva5QKDzQ1lZmVwul++xy+U65Yddc1dcXKzvv/9e3bt3lyR98MEHevjhhzVv3jwdOHBA0on73a5dO5WVldXbj/LycoWFhfn+KNS8/vh1hYaGKiwsTOXl5QHZ17pkZGTokUceUVZWliRp3759ioqKklT9y71//35J1XVHR0f7lqvZ14b25mTbsNuGDRtqfWAH23tBsvfnX9827LRmzZpah/qKi4v1pz/9SY8//rh27Ngh6eSfi43VT7vY9TvQ3PqwY8cORUREqGPHjr6xs/29cOzfRtM+FziHzg91zR44HA4bKmkcR44c0cyZMzV69GiFhYXp2muv1W233SZJWrp0qf7+979r4sSJ9c6anE4/mlMPn3rqKbVr10779u3T9OnTfeeD1KUhdTscjmY16+iPyspK5ebm6g9/+IMkBd174VQC8fNvbv1Yvny5QkNDddVVV0mq/iMzb948tW3bVt99952effZZzZw5s8n31a4e2Pk70Nw+P47/n72z/b1w/N/G+jTXzwVm6PzgcrlUWlrqe1xaWupL1KaprKzUzJkzddVVV+mKK66QJEVGRiokJEQhISEaMmSIvv32W0kn7ndZWZnatWtXbz/atm2rQ4cOqaqqqtbrj19XVVWVDh06ZNsFATU1RUREqG/fviosLFRERIRvOnvv3r2+E6JdLpdKSkp8y9bsa7t27U7oTVRUVJ29OXZ7dW3DTp9//rkuuOACRUZGSgq+90INO3/+9W3DDmvXrlVubq7uv/9+3x+PFi1aqG3btpKq76kVExOj3bt3N+q+1tdPO9j5O1DfNuxQVVWlzZs3KzEx0Td2Nr8X6vrbaNrnAoHOD926ddPu3btVXFysyspKZWdny+12211Wg1mWpVdeeUWdOnXSTTfd5Bs/9rj85s2bFRcXJ0lyu93Kzs7W0aNHVVxcrN27d6t79+719sPhcKhXr17atGmTpOo/DjV9uvzyy7V27VpJ0qZNm9SrVy9b/g/8yJEjOnz4sO/fX3zxhX7zm9/I7XZr3bp1kqR169apb9++kqp78Mknn8iyLH3zzTcKCwtTVFSU4uPjtWXLFh04cEAHDhzQli1bFB8fr6ioKJ1zzjn65ptvZFmWPvnkE18P6tuGnY7/P/Bgei8cy86ff33bCLT8/HytXLlSjzzyiFq1auUb379/v7xerySpqKhIu3fvVkxMTKPua339tIOdvwP1bcMOW7duVWxsbK1DhWfre6G+v42mfS5wY2E/5eXl6W9/+5u8Xq+Sk5N166232l1Sg3311Vf6y1/+ot/85je+P6B33HGHNmzYoB9++EEOh0Pt27fXfffd53vjLF++XB9//LFCQkI0evRo9enTR1L9/SgqKjrhMv0WLVqooqJCc+bM0ffff6/w8HClpaUpJiYm4D0oKirSc889J6n6/0CvvPJK3XrrrSovL9esWbNUUlKi6OhoTZ061Xfp+IIFC7Rlyxa1bNlSEydOVLdu3SRVn2e0YsUKSdWXpycnJ0uSvv32W82bN08VFRWKj49XamqqHA5Hvduwy6+//qoJEyZozpw5vsMLL7300ln/Xpg9e7a2b9+u8vJyRUREaOTIkerbt69tP/+TbSOQPVixYoUqKyt978maW1Js2rRJy5YtU2hoqEJCQjRixAjfH6PG3Nf6+hnoPmzbts3W34H6thHIHgwePFhz587VhRdeqGuvvdb32rP1vVDf38YLL7zQqM8FAh0AAIDhOOQKAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHYCz0qRJk/TFF18EdJvFxcUaOXKk72ayABAoBDoAOE12hEYAqAuBDgAAwHBOuwsAgKbk9Xq1atUqffTRRzp48KAuueQS3XfffQoPD1dxcbEmT56siRMnaunSpaqoqNCNN97ou9N/RUWF/vrXvyo3N1eRkZFKSkrS6tWr9corr+ill15SSUmJnnnmGYWEhOi2227TgAEDJEmffvppnesDgKZCoANwVlu9erVycnL0xBNP6Nxzz9WiRYv06quvKi0tzfear776Si+88IJ27dql9PR09evXT507d9Ybb7yhPXv26KWXXtKvv/6qp59+2rfMlClT9NVXX2ncuHG67LLLJFWfQ3ey9QFAU+GQK4CzWlZWlkaNGiWXy6UWLVpoxIgR+uyzz2pduDBixAi1bNlS559/vrp06aIff/xRkrRx40YNHz5c4eHhcrlcuv766/3aZn3rA4CmwgwdgLPanj179Nxzz/m+dFuSQkJCtG/fPt/jyMhI379btWqlI0eOSJL27t2r6Oho33Mul8uvbda3PgBoKgQ6AGc1l8ulCRMm6KKLLjrhuZpDpPWJjIxUaWmp73BpaWlpk9QIAGeKQ64AzmrXXHONXn/9de3Zs0eStH//fuXk5Pi17IABA/T222/rwIEDKisr0/vvv1/r+cjIyFOGQgAIBGboAJzVbrjhBknS9OnTtXfvXkVERGjAgAHq27fvKZe97bbbNH/+fE2ePFlRUVG68sortXbtWt/zw4YN08KFC7VkyRLdeuut6t+/f1PtBgCclMOyLMvuIgDABB9++KE2bNigJ5980u5SAKAWDrkCQD327t2rr776Sl6vV7t27dI777yjfv362V0WAJyAQ64AUI/KykrNnz9fxcXFCgsL08CBAzV06FC7ywKAE3DIFQAAwHAccgUAADAcgQ4AAMBwBDoAAADDEegAAAAMR6ADAAAwHIEOAADAcP8XDZ32cdXspacAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f4534df7f98>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "%matplotlib inline  \n",
    "plt.style.use('ggplot')\n",
    "\n",
    "f = open(\"./metagenomics/assembly/scaffolds.fasta\", \"r\")\n",
    "lines = f.readlines()\n",
    "f.close()\n",
    "\n",
    "data = []\n",
    "regexp = re.compile(\">\")\n",
    "\n",
    "for line in lines:\n",
    "    if re.search(regexp, line):\n",
    "        line = line.strip().split('_')\n",
    "        data.append(float(line[3]))\n",
    "        \n",
    "fig = plt.figure(figsize=(10,10))\n",
    "plt.hist(data, bins=100, log=True);\n",
    "plt.title(\"length distribution scaffolds\");\n",
    "plt.xlabel(\"length\");\n",
    "plt.ylabel(\"count\");"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Did you expect this distribution?<br>\n",
    "Why would there be so many short scaffolds?<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2> Mapping reads to scaffolds with bwa (Burrows-Wheeler Aligner) and sorting with Samtools</h2>\n",
    "We now have the reads and the assembly so we can start by mapping (aligning) the original reads back to the scaffolds created with the assembly. Doing this will allow us to see the read depth on the scaffolds.\n",
    "\n",
    "The reads are paired-end, which means that for each DNA fragment, we have sequence data from both ends. The sequences are therefore stored in two separate files (one for the data from each end). We’ll use bwa’s default settings to align the reads back and then Samtools allows us to work with the bwa results.\n",
    "\n",
    "<b>Assignment:</b><br>\n",
    "Use bwa to backmap the reads against the scaffolds and save the mapping as a .bam file.\n",
    "(because we want to map all the reads back we helped you a bit with the commands).\n",
    "\n",
    "(1) To run bwa we first need to make an index file of the scaffolds. <br>\n",
    "(2) Then we can run bwa and immediatly pipe the output of bwa through samtools view which will output a bam file <br>\n",
    "(3) Finally the bam file has to be sorted using Samtools sort.\n",
    "\n",
    "\n",
    "some background: By default mapping algorithms like bwa spit out a .sam file. The sam format is widely used and accepted by many different downstream programmes. Basically it's just a big table with a standardised format. Although a sam file is human readable, it is also rather bulky. The filesize of a single SAM file will quickly exceed the disk space you have on this virtual machine. Therefore, we convert it to a BAM file, which is a Binary sAM file. There is no loss of information, it is just stored much more efficiently. Naturally, a binary file is not human readable. Always store your files as bam files, if you need to have a look in the bam file you can view them using samtools view or any of the many downstream programmes available online."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "bwa help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Program: bwa (alignment via Burrows-Wheeler transformation)\r\n",
      "Version: 0.7.17-r1188\r\n",
      "Contact: Heng Li <lh3@sanger.ac.uk>\r\n",
      "\r\n",
      "Usage:   bwa <command> [options]\r\n",
      "\r\n",
      "Command: index         index sequences in the FASTA format\r\n",
      "         mem           BWA-MEM algorithm\r\n",
      "         fastmap       identify super-maximal exact matches\r\n",
      "         pemerge       merge overlapping paired ends (EXPERIMENTAL)\r\n",
      "         aln           gapped/ungapped alignment\r\n",
      "         samse         generate alignment (single ended)\r\n",
      "         sampe         generate alignment (paired ended)\r\n",
      "         bwasw         BWA-SW for long queries\r\n",
      "\r\n",
      "         shm           manage indices in shared memory\r\n",
      "         fa2pac        convert FASTA to PAC format\r\n",
      "         pac2bwt       generate BWT from PAC\r\n",
      "         pac2bwtgen    alternative algorithm for generating BWT\r\n",
      "         bwtupdate     update .bwt to the new format\r\n",
      "         bwt2sa        generate SA from BWT and Occ\r\n",
      "\r\n",
      "Note: To use BWA, you need to first index the genome with `bwa index'.\r\n",
      "      There are three alignment algorithms in BWA: `mem', `bwasw', and\r\n",
      "      `aln/samse/sampe'. If you are not sure which to use, try `bwa mem'\r\n",
      "      first. Please `man ./bwa.1' for the manual.\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!bwa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "samtools help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.7 (using htslib 1.7-2)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!samtools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>run bwa index on default to make index files of your scaffolds (so no need for options)</b>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "bwa index help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!bwa index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run bwa index, finish the code: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!bwa index ...."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 4.47 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=1207810830, availableWord=96985660\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 99999998 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 199999998 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 299999998 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 399999998 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 499999998 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 597255246 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 684196110 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 761465326 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 830138430 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 891171246 characters processed.\n",
      "[BWTIncConstructFromPacked] 110 iterations done. 945413326 characters processed.\n",
      "[BWTIncConstructFromPacked] 120 iterations done. 993619838 characters processed.\n",
      "[BWTIncConstructFromPacked] 130 iterations done. 1036461870 characters processed.\n",
      "[BWTIncConstructFromPacked] 140 iterations done. 1074535966 characters processed.\n",
      "[BWTIncConstructFromPacked] 150 iterations done. 1108372270 characters processed.\n",
      "[BWTIncConstructFromPacked] 160 iterations done. 1138442030 characters processed.\n",
      "[BWTIncConstructFromPacked] 170 iterations done. 1165164094 characters processed.\n",
      "[BWTIncConstructFromPacked] 180 iterations done. 1188910686 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 189 iterations.\n",
      "[bwa_index] 324.60 seconds elapse.\n",
      "[bwa_index] Update BWT... 2.66 sec\n",
      "[bwa_index] Pack forward-only FASTA... 3.15 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 142.79 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index ./metagenomics/assembly/scaffolds.fasta\n",
      "[main] Real time: 480.825 sec; CPU: 477.678 sec\n"
     ]
    }
   ],
   "source": [
    "!bwa index ./metagenomics/assembly/scaffolds.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>check the output of bwa</b><br>\n",
    "there should now be multiple files in the assembly folder <br>\n",
    "think about what command you could use to look at the files in a directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>run bwa mem for backmapping</b>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For this part we will use a bash for loop to run bwa mem and samtools view on all the reads in metagenomics/reads and  create BAM output files in a newly made directory\n",
    "\n",
    "Then we use a bash for loop to run samtools sort on all the files created by samtools view\n",
    "\n",
    "Step by step instruction:<br>\n",
    "(1) make a new directory for the samtools view results, give this a usefull name<br>\n",
    "(2) finish the for loop and run both bwa mem and samtools view<br>\n",
    "(3) make a new directory for the samtools sort results, give this a usefull name<br>\n",
    "(4) finish the for loop and run samtools sort<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "bwa mem help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Usage: bwa mem [options] <idxbase> <in1.fq> [in2.fq]\r\n",
      "\r\n",
      "Algorithm options:\r\n",
      "\r\n",
      "       -t INT        number of threads [1]\r\n",
      "       -k INT        minimum seed length [19]\r\n",
      "       -w INT        band width for banded alignment [100]\r\n",
      "       -d INT        off-diagonal X-dropoff [100]\r\n",
      "       -r FLOAT      look for internal seeds inside a seed longer than {-k} * FLOAT [1.5]\r\n",
      "       -y INT        seed occurrence for the 3rd round seeding [20]\r\n",
      "       -c INT        skip seeds with more than INT occurrences [500]\r\n",
      "       -D FLOAT      drop chains shorter than FLOAT fraction of the longest overlapping chain [0.50]\r\n",
      "       -W INT        discard a chain if seeded bases shorter than INT [0]\r\n",
      "       -m INT        perform at most INT rounds of mate rescues for each read [50]\r\n",
      "       -S            skip mate rescue\r\n",
      "       -P            skip pairing; mate rescue performed unless -S also in use\r\n",
      "\r\n",
      "Scoring options:\r\n",
      "\r\n",
      "       -A INT        score for a sequence match, which scales options -TdBOELU unless overridden [1]\r\n",
      "       -B INT        penalty for a mismatch [4]\r\n",
      "       -O INT[,INT]  gap open penalties for deletions and insertions [6,6]\r\n",
      "       -E INT[,INT]  gap extension penalty; a gap of size k cost '{-O} + {-E}*k' [1,1]\r\n",
      "       -L INT[,INT]  penalty for 5'- and 3'-end clipping [5,5]\r\n",
      "       -U INT        penalty for an unpaired read pair [17]\r\n",
      "\r\n",
      "       -x STR        read type. Setting -x changes multiple parameters unless overridden [null]\r\n",
      "                     pacbio: -k17 -W40 -r10 -A1 -B1 -O1 -E1 -L0  (PacBio reads to ref)\r\n",
      "                     ont2d: -k14 -W20 -r10 -A1 -B1 -O1 -E1 -L0  (Oxford Nanopore 2D-reads to ref)\r\n",
      "                     intractg: -B9 -O16 -L5  (intra-species contigs to ref)\r\n",
      "\r\n",
      "Input/output options:\r\n",
      "\r\n",
      "       -p            smart pairing (ignoring in2.fq)\r\n",
      "       -R STR        read group header line such as '@RG\\tID:foo\\tSM:bar' [null]\r\n",
      "       -H STR/FILE   insert STR to header if it starts with @; or insert lines in FILE [null]\r\n",
      "       -o FILE       sam file to output results to [stdout]\r\n",
      "       -j            treat ALT contigs as part of the primary assembly (i.e. ignore <idxbase>.alt file)\r\n",
      "       -5            for split alignment, take the alignment with the smallest coordinate as primary\r\n",
      "       -q            don't modify mapQ of supplementary alignments\r\n",
      "       -K INT        process INT input bases in each batch regardless of nThreads (for reproducibility) []\r\n",
      "\r\n",
      "       -v INT        verbosity level: 1=error, 2=warning, 3=message, 4+=debugging [3]\r\n",
      "       -T INT        minimum score to output [30]\r\n",
      "       -h INT[,INT]  if there are <INT hits with score >80% of the max score, output all in XA [5,200]\r\n",
      "       -a            output all alignments for SE or unpaired PE\r\n",
      "       -C            append FASTA/FASTQ comment to SAM output\r\n",
      "       -V            output the reference FASTA header in the XR tag\r\n",
      "       -Y            use soft clipping for supplementary alignments\r\n",
      "       -M            mark shorter split hits as secondary\r\n",
      "\r\n",
      "       -I FLOAT[,FLOAT[,INT[,INT]]]\r\n",
      "                     specify the mean, standard deviation (10% of the mean if absent), max\r\n",
      "                     (4 sigma from the mean if absent) and min of the insert size distribution.\r\n",
      "                     FR orientation only. [inferred]\r\n",
      "\r\n",
      "Note: Please read the man page for detailed description of the command line and options.\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!bwa mem"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "samtools view help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools view"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘./metagenomics/mapping’: File exists\r\n"
     ]
    }
   ],
   "source": [
    "mkdir ./metagenomics/mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L1.R1.fastq.gz  L2.R2.fastq.gz  P1.R1.fastq.gz  P2.R2.fastq.gz\r\n",
      "L1.R2.fastq.gz  L3.R1.fastq.gz  P1.R2.fastq.gz  P3.R1.fastq.gz\r\n",
      "L2.R1.fastq.gz  L3.R2.fastq.gz  P2.R1.fastq.gz  P3.R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls ./metagenomics/reads/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L1\n",
      "L2\n",
      "L3\n",
      "P1\n",
      "P2\n",
      "P3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 136468 sequences (20000127 bp)...\n",
      "[M::process] read 136216 sequences (20000016 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 47226, 42, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (175, 314, 463)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1039)\n",
      "[M::mem_pestat] mean and std.dev: (291.19, 146.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1327)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 422, 527)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 863)\n",
      "[M::mem_pestat] mean and std.dev: (439.10, 140.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1031)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (37, 95, 236)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 634)\n",
      "[M::mem_pestat] mean and std.dev: (118.16, 117.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 833)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136468 reads in 24.231 CPU sec, 11.999 real sec\n",
      "[M::process] read 136280 sequences (20000219 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (24, 47108, 43, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (177, 352, 480)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1086)\n",
      "[M::mem_pestat] mean and std.dev: (328.57, 184.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1389)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (361, 422, 531)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 871)\n",
      "[M::mem_pestat] mean and std.dev: (440.76, 141.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1041)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (34, 60, 178)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 466)\n",
      "[M::mem_pestat] mean and std.dev: (92.51, 90.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 610)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136216 reads in 24.790 CPU sec, 12.237 real sec\n",
      "[M::process] read 136398 sequences (20000087 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 47019, 48, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (129, 239, 490)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1212)\n",
      "[M::mem_pestat] mean and std.dev: (297.35, 229.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1573)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 421, 525)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 857)\n",
      "[M::mem_pestat] mean and std.dev: (437.45, 140.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1023)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (25, 68, 198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 544)\n",
      "[M::mem_pestat] mean and std.dev: (85.49, 100.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 717)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136280 reads in 24.781 CPU sec, 12.230 real sec\n",
      "[M::process] read 136376 sequences (20000275 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (22, 47092, 47, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (127, 374, 491)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1219)\n",
      "[M::mem_pestat] mean and std.dev: (301.30, 197.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1583)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 420, 522)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (33, 848)\n",
      "[M::mem_pestat] mean and std.dev: (436.20, 138.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1011)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (39, 68, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 465)\n",
      "[M::mem_pestat] mean and std.dev: (110.05, 105.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 607)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136398 reads in 24.896 CPU sec, 12.273 real sec\n",
      "[M::process] read 137072 sequences (20000092 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (23, 47318, 51, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (213, 325, 453)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 933)\n",
      "[M::mem_pestat] mean and std.dev: (321.00, 162.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1173)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (358, 419, 520)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 844)\n",
      "[M::mem_pestat] mean and std.dev: (434.71, 137.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1006)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (50, 79, 235)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 605)\n",
      "[M::mem_pestat] mean and std.dev: (130.40, 136.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 790)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136376 reads in 25.032 CPU sec, 12.352 real sec\n",
      "[M::process] read 137546 sequences (20000283 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (21, 47226, 32, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (190, 335, 489)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1087)\n",
      "[M::mem_pestat] mean and std.dev: (300.53, 155.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1386)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (358, 419, 516)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 832)\n",
      "[M::mem_pestat] mean and std.dev: (433.14, 134.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 990)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (28, 70, 280)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 784)\n",
      "[M::mem_pestat] mean and std.dev: (103.57, 112.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1036)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137072 reads in 25.316 CPU sec, 12.495 real sec\n",
      "[M::process] read 137150 sequences (20000151 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 47671, 37, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (171, 420, 806)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2076)\n",
      "[M::mem_pestat] mean and std.dev: (369.36, 252.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2711)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 418, 512)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (47, 822)\n",
      "[M::mem_pestat] mean and std.dev: (431.08, 133.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 977)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (35, 68, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 851)\n",
      "[M::mem_pestat] mean and std.dev: (144.06, 172.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1123)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137546 reads in 25.160 CPU sec, 12.417 real sec\n",
      "[M::process] read 136626 sequences (20000138 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 47595, 45, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 163, 441)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1103)\n",
      "[M::mem_pestat] mean and std.dev: (231.70, 195.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1434)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 420, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 833)\n",
      "[M::mem_pestat] mean and std.dev: (433.36, 134.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 991)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (34, 70, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 490)\n",
      "[M::mem_pestat] mean and std.dev: (104.44, 111.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 642)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137150 reads in 24.719 CPU sec, 12.173 real sec\n",
      "[M::process] read 136540 sequences (20000220 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (20, 47454, 39, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (141, 333, 437)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1029)\n",
      "[M::mem_pestat] mean and std.dev: (277.63, 143.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1325)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (358, 419, 518)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (38, 838)\n",
      "[M::mem_pestat] mean and std.dev: (433.70, 136.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 998)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (50, 88, 343)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 929)\n",
      "[M::mem_pestat] mean and std.dev: (167.43, 216.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1222)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136626 reads in 24.932 CPU sec, 12.281 real sec\n",
      "[M::process] read 136502 sequences (20000252 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (19, 47197, 43, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (313, 391, 488)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 838)\n",
      "[M::mem_pestat] mean and std.dev: (371.82, 115.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1013)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 420, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 833)\n",
      "[M::mem_pestat] mean and std.dev: (434.44, 135.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 991)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (43, 91, 255)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 679)\n",
      "[M::mem_pestat] mean and std.dev: (137.75, 145.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 891)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136540 reads in 24.940 CPU sec, 12.310 real sec\n",
      "[M::process] read 136616 sequences (20000189 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (19, 47543, 39, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (106, 336, 444)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1120)\n",
      "[M::mem_pestat] mean and std.dev: (287.53, 184.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1458)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 419, 515)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (41, 831)\n",
      "[M::mem_pestat] mean and std.dev: (432.01, 134.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 989)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (39, 88, 263)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 711)\n",
      "[M::mem_pestat] mean and std.dev: (138.91, 157.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 935)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136502 reads in 24.896 CPU sec, 12.277 real sec\n",
      "[M::process] read 137040 sequences (20000196 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (15, 47479, 35, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (294, 334, 476)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 840)\n",
      "[M::mem_pestat] mean and std.dev: (354.40, 152.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1022)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (358, 419, 515)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (44, 829)\n",
      "[M::mem_pestat] mean and std.dev: (432.15, 135.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 986)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (17, 84, 239)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 683)\n",
      "[M::mem_pestat] mean and std.dev: (114.36, 122.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 905)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136616 reads in 24.655 CPU sec, 12.156 real sec\n",
      "[M::process] read 137472 sequences (20000097 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (21, 47515, 44, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (159, 365, 531)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1275)\n",
      "[M::mem_pestat] mean and std.dev: (373.55, 276.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1647)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (356, 418, 510)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (48, 818)\n",
      "[M::mem_pestat] mean and std.dev: (430.22, 132.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 972)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (45, 102, 268)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 714)\n",
      "[M::mem_pestat] mean and std.dev: (134.15, 153.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 937)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137040 reads in 24.974 CPU sec, 12.300 real sec\n",
      "[M::process] read 85698 sequences (12437560 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (21, 47748, 40, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (177, 275, 412)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 882)\n",
      "[M::mem_pestat] mean and std.dev: (275.70, 138.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1117)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (355, 417, 508)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (49, 814)\n",
      "[M::mem_pestat] mean and std.dev: (428.02, 131.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 967)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (39, 99, 381)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1065)\n",
      "[M::mem_pestat] mean and std.dev: (204.74, 239.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1407)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137472 reads in 24.953 CPU sec, 12.358 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 29624, 25, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (86, 218, 400)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1028)\n",
      "[M::mem_pestat] mean and std.dev: (239.54, 151.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1342)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 418, 510)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (51, 816)\n",
      "[M::mem_pestat] mean and std.dev: (429.81, 132.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 969)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (20, 45, 69)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 167)\n",
      "[M::mem_pestat] mean and std.dev: (46.30, 36.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 216)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 85698 reads in 15.629 CPU sec, 7.763 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 2 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/L1.R1.fastq.gz ./metagenomics/reads/L1.R2.fastq.gz\n",
      "[main] Real time: 182.818 sec; CPU: 365.705 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 136982 sequences (20000186 bp)...\n",
      "[M::process] read 136780 sequences (20000237 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (105, 46303, 55, 9)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 206, 335)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 767)\n",
      "[M::mem_pestat] mean and std.dev: (239.04, 137.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 983)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (322, 379, 525)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 931)\n",
      "[M::mem_pestat] mean and std.dev: (419.07, 151.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1134)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (33, 50, 114)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 276)\n",
      "[M::mem_pestat] mean and std.dev: (56.87, 49.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 357)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136982 reads in 26.197 CPU sec, 12.982 real sec\n",
      "[M::process] read 137056 sequences (20000166 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (109, 45861, 44, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (125, 197, 356)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 818)\n",
      "[M::mem_pestat] mean and std.dev: (230.26, 136.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1049)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (323, 379, 526)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 932)\n",
      "[M::mem_pestat] mean and std.dev: (419.59, 150.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1135)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (40, 60, 145)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 355)\n",
      "[M::mem_pestat] mean and std.dev: (78.74, 72.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 460)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136780 reads in 26.885 CPU sec, 13.282 real sec\n",
      "[M::process] read 137430 sequences (20000062 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (113, 45785, 60, 7)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (149, 265, 394)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 884)\n",
      "[M::mem_pestat] mean and std.dev: (264.88, 139.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1129)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 378, 525)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 933)\n",
      "[M::mem_pestat] mean and std.dev: (417.45, 150.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1137)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (37, 74, 215)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 571)\n",
      "[M::mem_pestat] mean and std.dev: (107.40, 110.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 749)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137056 reads in 27.147 CPU sec, 13.389 real sec\n",
      "[M::process] read 137934 sequences (20000113 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (96, 45930, 55, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (109, 224, 377)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 913)\n",
      "[M::mem_pestat] mean and std.dev: (240.95, 150.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1181)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (322, 379, 523)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 925)\n",
      "[M::mem_pestat] mean and std.dev: (416.65, 148.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1126)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (37, 99, 259)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 703)\n",
      "[M::mem_pestat] mean and std.dev: (137.88, 149.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 925)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137430 reads in 27.345 CPU sec, 13.501 real sec\n",
      "[M::process] read 137222 sequences (20000226 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (91, 46634, 50, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (141, 246, 355)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 783)\n",
      "[M::mem_pestat] mean and std.dev: (246.49, 143.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 997)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 376, 515)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 905)\n",
      "[M::mem_pestat] mean and std.dev: (411.56, 144.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1100)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (22, 42, 157)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 427)\n",
      "[M::mem_pestat] mean and std.dev: (77.91, 84.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 562)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137934 reads in 27.061 CPU sec, 13.360 real sec\n",
      "[M::process] read 137098 sequences (20000120 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (98, 45985, 54, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (113, 204, 311)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 707)\n",
      "[M::mem_pestat] mean and std.dev: (220.22, 130.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 905)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 377, 520)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 920)\n",
      "[M::mem_pestat] mean and std.dev: (414.52, 148.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1120)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (39, 91, 189)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 489)\n",
      "[M::mem_pestat] mean and std.dev: (98.17, 99.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 639)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137222 reads in 26.765 CPU sec, 13.222 real sec\n",
      "[M::process] read 137210 sequences (20000117 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (107, 45843, 54, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (101, 212, 323)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 767)\n",
      "[M::mem_pestat] mean and std.dev: (225.47, 152.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 989)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 377, 519)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 915)\n",
      "[M::mem_pestat] mean and std.dev: (413.66, 146.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1113)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (51, 93, 278)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 732)\n",
      "[M::mem_pestat] mean and std.dev: (139.18, 123.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 959)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137098 reads in 26.989 CPU sec, 13.335 real sec\n",
      "[M::process] read 137730 sequences (20000248 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (83, 46118, 47, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (121, 236, 372)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 874)\n",
      "[M::mem_pestat] mean and std.dev: (247.55, 142.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1125)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 378, 520)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 918)\n",
      "[M::mem_pestat] mean and std.dev: (414.49, 147.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1117)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (39, 105, 336)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 930)\n",
      "[M::mem_pestat] mean and std.dev: (158.63, 193.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1227)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137210 reads in 27.119 CPU sec, 13.384 real sec\n",
      "[M::process] read 137974 sequences (20000073 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (92, 46078, 52, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (107, 195, 343)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 815)\n",
      "[M::mem_pestat] mean and std.dev: (235.84, 152.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1051)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 376, 515)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 907)\n",
      "[M::mem_pestat] mean and std.dev: (409.81, 145.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1103)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (41, 70, 190)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 488)\n",
      "[M::mem_pestat] mean and std.dev: (94.68, 92.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 637)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137730 reads in 27.363 CPU sec, 13.501 real sec\n",
      "[M::process] read 137168 sequences (20000022 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (81, 46258, 47, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (135, 233, 344)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 762)\n",
      "[M::mem_pestat] mean and std.dev: (244.28, 136.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 971)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 377, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 913)\n",
      "[M::mem_pestat] mean and std.dev: (412.11, 146.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1111)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (31, 88, 236)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 646)\n",
      "[M::mem_pestat] mean and std.dev: (120.26, 128.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 851)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137974 reads in 27.228 CPU sec, 13.446 real sec\n",
      "[M::process] read 137180 sequences (20000063 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (92, 45893, 57, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 227, 366)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 894)\n",
      "[M::mem_pestat] mean and std.dev: (233.37, 151.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1158)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 378, 522)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 924)\n",
      "[M::mem_pestat] mean and std.dev: (415.80, 148.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1125)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (26, 97, 239)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 665)\n",
      "[M::mem_pestat] mean and std.dev: (131.77, 130.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 878)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137168 reads in 27.379 CPU sec, 13.515 real sec\n",
      "[M::process] read 137152 sequences (20000134 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (103, 45869, 48, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (105, 205, 366)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 888)\n",
      "[M::mem_pestat] mean and std.dev: (232.38, 148.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1149)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (322, 379, 524)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 928)\n",
      "[M::mem_pestat] mean and std.dev: (417.20, 150.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1130)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (28, 178, 459)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1321)\n",
      "[M::mem_pestat] mean and std.dev: (200.16, 209.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1752)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137180 reads in 26.769 CPU sec, 13.199 real sec\n",
      "[M::process] read 137406 sequences (20000020 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (110, 46041, 57, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (113, 202, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 851)\n",
      "[M::mem_pestat] mean and std.dev: (238.66, 152.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1097)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (322, 379, 522)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 922)\n",
      "[M::mem_pestat] mean and std.dev: (417.17, 149.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1122)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (33, 101, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 759)\n",
      "[M::mem_pestat] mean and std.dev: (170.37, 186.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1001)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137152 reads in 26.812 CPU sec, 13.226 real sec\n",
      "[M::process] read 77678 sequences (11272005 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (95, 46386, 51, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 209, 325)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 771)\n",
      "[M::mem_pestat] mean and std.dev: (213.05, 120.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 994)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (322, 378, 522)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 922)\n",
      "[M::mem_pestat] mean and std.dev: (416.45, 148.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1122)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (33, 95, 157)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 405)\n",
      "[M::mem_pestat] mean and std.dev: (95.49, 86.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 529)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137406 reads in 26.321 CPU sec, 13.038 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (57, 25935, 42, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (147, 218, 335)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 711)\n",
      "[M::mem_pestat] mean and std.dev: (246.02, 134.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 899)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 377, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 913)\n",
      "[M::mem_pestat] mean and std.dev: (412.66, 146.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1111)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (22, 97, 352)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1012)\n",
      "[M::mem_pestat] mean and std.dev: (130.38, 147.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1342)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 77678 reads in 15.307 CPU sec, 7.584 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 2 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/L2.R1.fastq.gz ./metagenomics/reads/L2.R2.fastq.gz\n",
      "[main] Real time: 196.940 sec; CPU: 394.435 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 136284 sequences (20000131 bp)...\n",
      "[M::process] read 136212 sequences (20000147 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (20, 48042, 46, 7)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (138, 230, 410)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 954)\n",
      "[M::mem_pestat] mean and std.dev: (226.05, 139.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1226)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (354, 414, 469)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (124, 699)\n",
      "[M::mem_pestat] mean and std.dev: (412.84, 109.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 849)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (30, 71, 214)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 582)\n",
      "[M::mem_pestat] mean and std.dev: (83.28, 84.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 766)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136284 reads in 25.054 CPU sec, 12.409 real sec\n",
      "[M::process] read 136180 sequences (20000186 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (24, 47814, 42, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 279, 438)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 900)\n",
      "[M::mem_pestat] mean and std.dev: (293.42, 162.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1131)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (354, 413, 465)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (132, 687)\n",
      "[M::mem_pestat] mean and std.dev: (411.47, 106.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 836)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (20, 83, 265)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 755)\n",
      "[M::mem_pestat] mean and std.dev: (147.80, 169.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1000)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136212 reads in 25.572 CPU sec, 12.625 real sec\n",
      "[M::process] read 136256 sequences (20000058 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (23, 47617, 46, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (309, 429, 504)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 894)\n",
      "[M::mem_pestat] mean and std.dev: (355.35, 150.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1089)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (354, 415, 464)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (134, 684)\n",
      "[M::mem_pestat] mean and std.dev: (410.90, 105.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 834)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (34, 89, 304)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 844)\n",
      "[M::mem_pestat] mean and std.dev: (135.71, 165.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1114)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136180 reads in 26.005 CPU sec, 12.833 real sec\n",
      "[M::process] read 136300 sequences (20000019 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (20, 47602, 33, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (290, 373, 484)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 872)\n",
      "[M::mem_pestat] mean and std.dev: (364.74, 132.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (353, 413, 463)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (133, 683)\n",
      "[M::mem_pestat] mean and std.dev: (410.28, 105.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 832)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (36, 74, 149)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 375)\n",
      "[M::mem_pestat] mean and std.dev: (90.34, 84.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 488)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136256 reads in 25.766 CPU sec, 12.722 real sec\n",
      "[M::process] read 136764 sequences (20000164 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (23, 47592, 39, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (281, 379, 461)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 821)\n",
      "[M::mem_pestat] mean and std.dev: (344.00, 156.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1001)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 413, 460)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (136, 676)\n",
      "[M::mem_pestat] mean and std.dev: (409.23, 104.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 828)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (18, 47, 213)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 603)\n",
      "[M::mem_pestat] mean and std.dev: (79.21, 85.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 798)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136300 reads in 25.905 CPU sec, 12.785 real sec\n",
      "[M::process] read 137242 sequences (20000201 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (16, 47686, 52, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 430, 643)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1225)\n",
      "[M::mem_pestat] mean and std.dev: (394.54, 132.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1516)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (355, 414, 461)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (143, 673)\n",
      "[M::mem_pestat] mean and std.dev: (410.77, 102.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (2, 819)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (42, 76, 209)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 543)\n",
      "[M::mem_pestat] mean and std.dev: (112.73, 108.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 710)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136764 reads in 26.108 CPU sec, 12.881 real sec\n",
      "[M::process] read 136890 sequences (20000151 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 47903, 46, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (162, 227, 283)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 525)\n",
      "[M::mem_pestat] mean and std.dev: (173.00, 87.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 646)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (353, 412, 452)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (155, 650)\n",
      "[M::mem_pestat] mean and std.dev: (407.45, 97.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (19, 795)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (20, 67, 139)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 377)\n",
      "[M::mem_pestat] mean and std.dev: (75.12, 76.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 496)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137242 reads in 26.090 CPU sec, 12.879 real sec\n",
      "[M::process] read 136554 sequences (20000097 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (28, 48068, 52, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 359, 521)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1159)\n",
      "[M::mem_pestat] mean and std.dev: (339.33, 160.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1478)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (353, 413, 454)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (151, 656)\n",
      "[M::mem_pestat] mean and std.dev: (407.87, 98.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (15, 801)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (21, 67, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 504)\n",
      "[M::mem_pestat] mean and std.dev: (86.29, 110.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 665)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136890 reads in 25.688 CPU sec, 12.658 real sec\n",
      "[M::process] read 136376 sequences (20000037 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (15, 47794, 44, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (282, 358, 410)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 666)\n",
      "[M::mem_pestat] mean and std.dev: (329.93, 133.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 864)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 413, 457)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (142, 667)\n",
      "[M::mem_pestat] mean and std.dev: (408.28, 101.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (2, 815)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (24, 100, 303)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 861)\n",
      "[M::mem_pestat] mean and std.dev: (165.76, 205.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1140)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136554 reads in 25.972 CPU sec, 12.810 real sec\n",
      "[M::process] read 136368 sequences (20000075 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (16, 47747, 45, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (103, 283, 455)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1159)\n",
      "[M::mem_pestat] mean and std.dev: (298.44, 249.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1511)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (354, 413, 459)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 669)\n",
      "[M::mem_pestat] mean and std.dev: (409.99, 102.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (2, 818)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (54, 89, 188)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 456)\n",
      "[M::mem_pestat] mean and std.dev: (111.55, 88.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 590)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136376 reads in 25.609 CPU sec, 12.644 real sec\n",
      "[M::process] read 136384 sequences (20000198 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (21, 47975, 44, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (238, 353, 405)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 739)\n",
      "[M::mem_pestat] mean and std.dev: (307.60, 132.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 906)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 413, 456)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (144, 664)\n",
      "[M::mem_pestat] mean and std.dev: (408.03, 101.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (3, 813)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (32, 69, 157)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 407)\n",
      "[M::mem_pestat] mean and std.dev: (68.87, 60.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 532)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136368 reads in 25.522 CPU sec, 12.576 real sec\n",
      "[M::process] read 136692 sequences (20000286 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (16, 47806, 51, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (156, 325, 444)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1020)\n",
      "[M::mem_pestat] mean and std.dev: (274.87, 139.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1308)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (351, 412, 456)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (141, 666)\n",
      "[M::mem_pestat] mean and std.dev: (407.43, 102.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 817)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (46, 86, 196)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 496)\n",
      "[M::mem_pestat] mean and std.dev: (105.11, 99.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 646)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136384 reads in 25.945 CPU sec, 12.811 real sec\n",
      "[M::process] read 137214 sequences (20000030 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (27, 47961, 41, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (260, 303, 434)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 782)\n",
      "[M::mem_pestat] mean and std.dev: (324.62, 153.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 956)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (351, 412, 450)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (153, 648)\n",
      "[M::mem_pestat] mean and std.dev: (405.74, 96.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (19, 793)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (25, 65, 135)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 355)\n",
      "[M::mem_pestat] mean and std.dev: (71.86, 66.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 465)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136692 reads in 25.914 CPU sec, 12.795 real sec\n",
      "[M::process] read 88284 sequences (12837174 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 48275, 42, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (126, 323, 497)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1239)\n",
      "[M::mem_pestat] mean and std.dev: (340.00, 300.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1610)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 412, 449)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (158, 643)\n",
      "[M::mem_pestat] mean and std.dev: (406.67, 95.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (26, 788)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (29, 80, 245)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 677)\n",
      "[M::mem_pestat] mean and std.dev: (127.63, 154.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 893)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137214 reads in 25.645 CPU sec, 12.679 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 30789, 24, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 411, 448)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (160, 640)\n",
      "[M::mem_pestat] mean and std.dev: (404.91, 93.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (31, 779)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (36, 107, 625)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1803)\n",
      "[M::mem_pestat] mean and std.dev: (290.91, 397.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2392)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 88284 reads in 16.945 CPU sec, 8.402 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 2 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/L3.R1.fastq.gz ./metagenomics/reads/L3.R2.fastq.gz\n",
      "[main] Real time: 189.663 sec; CPU: 379.482 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 137262 sequences (20000156 bp)...\n",
      "[M::process] read 137112 sequences (20000128 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 30733, 65, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (373, 434, 552)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (15, 910)\n",
      "[M::mem_pestat] mean and std.dev: (456.86, 143.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1089)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (53, 107, 300)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 794)\n",
      "[M::mem_pestat] mean and std.dev: (170.79, 185.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1041)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137262 reads in 31.443 CPU sec, 15.601 real sec\n",
      "[M::process] read 137086 sequences (20000279 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 30816, 72, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 434, 557)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (2, 927)\n",
      "[M::mem_pestat] mean and std.dev: (458.15, 147.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1112)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (32, 74, 178)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 470)\n",
      "[M::mem_pestat] mean and std.dev: (84.79, 80.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 616)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137112 reads in 32.113 CPU sec, 15.892 real sec\n",
      "[M::process] read 137156 sequences (20000066 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 30585, 71, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (373, 435, 556)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (7, 922)\n",
      "[M::mem_pestat] mean and std.dev: (458.97, 144.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1105)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (35, 107, 288)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 794)\n",
      "[M::mem_pestat] mean and std.dev: (166.55, 181.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1047)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137086 reads in 32.068 CPU sec, 15.868 real sec\n",
      "[M::process] read 137402 sequences (20000032 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 30749, 69, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 434, 553)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (7, 917)\n",
      "[M::mem_pestat] mean and std.dev: (455.87, 143.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1099)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (27, 75, 214)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 588)\n",
      "[M::mem_pestat] mean and std.dev: (129.04, 131.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 775)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137156 reads in 32.146 CPU sec, 15.897 real sec\n",
      "[M::process] read 137326 sequences (20000182 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 30567, 75, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 434, 553)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 915)\n",
      "[M::mem_pestat] mean and std.dev: (456.41, 144.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1096)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (29, 70, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 500)\n",
      "[M::mem_pestat] mean and std.dev: (83.95, 80.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 657)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137402 reads in 32.384 CPU sec, 16.026 real sec\n",
      "[M::process] read 137434 sequences (20000236 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 31085, 68, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 434, 553)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 915)\n",
      "[M::mem_pestat] mean and std.dev: (456.21, 143.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1096)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (39, 110, 211)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 555)\n",
      "[M::mem_pestat] mean and std.dev: (103.03, 95.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 727)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137326 reads in 32.079 CPU sec, 15.871 real sec\n",
      "[M::process] read 138138 sequences (20000065 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 30988, 75, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 433, 549)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (12, 907)\n",
      "[M::mem_pestat] mean and std.dev: (453.86, 142.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1086)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (30, 84, 143)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 369)\n",
      "[M::mem_pestat] mean and std.dev: (91.28, 87.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 482)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137434 reads in 32.359 CPU sec, 16.010 real sec\n",
      "[M::process] read 138474 sequences (20000166 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31015, 81, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 433, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 899)\n",
      "[M::mem_pestat] mean and std.dev: (453.55, 140.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1075)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (43, 154, 387)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1075)\n",
      "[M::mem_pestat] mean and std.dev: (185.37, 199.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1419)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 138138 reads in 32.545 CPU sec, 16.100 real sec\n",
      "[M::process] read 139192 sequences (20000068 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 31035, 63, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 433, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 897)\n",
      "[M::mem_pestat] mean and std.dev: (453.06, 138.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1072)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (32, 74, 209)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 563)\n",
      "[M::mem_pestat] mean and std.dev: (116.66, 127.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 740)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 138474 reads in 32.386 CPU sec, 16.024 real sec\n",
      "[M::process] read 138224 sequences (20000157 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 31491, 71, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 431, 545)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 897)\n",
      "[M::mem_pestat] mean and std.dev: (450.01, 139.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1073)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (42, 84, 192)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 492)\n",
      "[M::mem_pestat] mean and std.dev: (89.20, 82.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 642)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 139192 reads in 32.448 CPU sec, 16.053 real sec\n",
      "[M::process] read 137768 sequences (20000048 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 31317, 59, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 432, 545)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 895)\n",
      "[M::mem_pestat] mean and std.dev: (451.48, 139.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1070)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (34, 76, 263)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 721)\n",
      "[M::mem_pestat] mean and std.dev: (116.08, 132.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 950)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 138224 reads in 32.000 CPU sec, 15.831 real sec\n",
      "[M::process] read 137556 sequences (20000058 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 30996, 69, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 433, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 900)\n",
      "[M::mem_pestat] mean and std.dev: (453.21, 141.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1076)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (35, 74, 164)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 422)\n",
      "[M::mem_pestat] mean and std.dev: (98.15, 94.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 551)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137768 reads in 32.233 CPU sec, 15.925 real sec\n",
      "[M::process] read 137636 sequences (20000290 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 31018, 67, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 432, 546)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (18, 898)\n",
      "[M::mem_pestat] mean and std.dev: (450.88, 141.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1074)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (39, 99, 326)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 900)\n",
      "[M::mem_pestat] mean and std.dev: (161.15, 188.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1187)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137556 reads in 32.393 CPU sec, 16.005 real sec\n",
      "[M::process] read 72234 sequences (10509348 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 31337, 61, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 433, 549)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (12, 907)\n",
      "[M::mem_pestat] mean and std.dev: (452.67, 142.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1086)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (51, 82, 320)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 858)\n",
      "[M::mem_pestat] mean and std.dev: (147.91, 171.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1127)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137636 reads in 32.352 CPU sec, 16.046 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 16364, 41, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 433, 546)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 896)\n",
      "[M::mem_pestat] mean and std.dev: (452.91, 140.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1071)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (60, 103, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 879)\n",
      "[M::mem_pestat] mean and std.dev: (158.78, 166.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1152)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 72234 reads in 16.869 CPU sec, 8.364 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 2 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/P1.R1.fastq.gz ./metagenomics/reads/P1.R2.fastq.gz\n",
      "[main] Real time: 234.384 sec; CPU: 469.557 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 136800 sequences (20000199 bp)...\n",
      "[M::process] read 136704 sequences (20000075 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 36438, 44, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 443, 555)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 907)\n",
      "[M::mem_pestat] mean and std.dev: (460.70, 141.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1083)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (30, 70, 253)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 699)\n",
      "[M::mem_pestat] mean and std.dev: (109.75, 123.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 922)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136800 reads in 28.254 CPU sec, 14.010 real sec\n",
      "[M::process] read 136754 sequences (20000195 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 36369, 61, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 444, 557)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 913)\n",
      "[M::mem_pestat] mean and std.dev: (461.49, 142.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1091)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (45, 101, 504)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1422)\n",
      "[M::mem_pestat] mean and std.dev: (163.06, 245.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1881)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136704 reads in 28.934 CPU sec, 14.306 real sec\n",
      "[M::process] read 136778 sequences (20000126 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 36175, 58, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 443, 554)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (29, 904)\n",
      "[M::mem_pestat] mean and std.dev: (460.35, 140.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1079)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (34, 74, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 481)\n",
      "[M::mem_pestat] mean and std.dev: (108.46, 112.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 630)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136754 reads in 28.804 CPU sec, 14.239 real sec\n",
      "[M::process] read 137018 sequences (20000061 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 36091, 72, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (377, 442, 554)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 908)\n",
      "[M::mem_pestat] mean and std.dev: (458.83, 141.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1085)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (37, 78, 231)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 619)\n",
      "[M::mem_pestat] mean and std.dev: (110.65, 116.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 813)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136778 reads in 29.109 CPU sec, 14.392 real sec\n",
      "[M::process] read 136854 sequences (20000221 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 36208, 60, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 443, 554)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (29, 904)\n",
      "[M::mem_pestat] mean and std.dev: (460.00, 139.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1079)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (53, 160, 364)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 986)\n",
      "[M::mem_pestat] mean and std.dev: (195.30, 206.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1297)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137018 reads in 29.153 CPU sec, 14.390 real sec\n",
      "[M::process] read 136920 sequences (20000114 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 36482, 40, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (376, 442, 550)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 898)\n",
      "[M::mem_pestat] mean and std.dev: (457.01, 140.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1072)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (37, 101, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 481)\n",
      "[M::mem_pestat] mean and std.dev: (115.38, 105.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 629)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136854 reads in 28.923 CPU sec, 14.302 real sec\n",
      "[M::process] read 137448 sequences (20000111 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 36482, 64, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (124, 301, 1039)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2869)\n",
      "[M::mem_pestat] mean and std.dev: (503.18, 547.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3784)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (378, 442, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (38, 888)\n",
      "[M::mem_pestat] mean and std.dev: (457.05, 138.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1058)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (38, 72, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 482)\n",
      "[M::mem_pestat] mean and std.dev: (96.37, 96.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 630)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 136920 reads in 29.186 CPU sec, 14.431 real sec\n",
      "[M::process] read 137916 sequences (20000180 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 36526, 60, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (377, 441, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (37, 887)\n",
      "[M::mem_pestat] mean and std.dev: (455.77, 137.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1057)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (58, 140, 368)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 988)\n",
      "[M::mem_pestat] mean and std.dev: (170.50, 186.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1298)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137448 reads in 29.332 CPU sec, 14.501 real sec\n",
      "[M::process] read 138310 sequences (20000197 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 36869, 61, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (380, 442, 549)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 887)\n",
      "[M::mem_pestat] mean and std.dev: (457.77, 136.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1056)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (33, 74, 239)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 651)\n",
      "[M::mem_pestat] mean and std.dev: (123.28, 145.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 857)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137916 reads in 29.198 CPU sec, 14.433 real sec\n",
      "[M::process] read 139056 sequences (20000173 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 36865, 62, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (374, 440, 544)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 884)\n",
      "[M::mem_pestat] mean and std.dev: (452.73, 136.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1054)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (36, 81, 189)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 495)\n",
      "[M::mem_pestat] mean and std.dev: (96.39, 92.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 648)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 138310 reads in 29.358 CPU sec, 14.513 real sec\n",
      "[M::process] read 137238 sequences (20000209 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 36959, 48, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (376, 440, 542)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (44, 874)\n",
      "[M::mem_pestat] mean and std.dev: (452.11, 134.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1040)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (32, 100, 337)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 947)\n",
      "[M::mem_pestat] mean and std.dev: (129.64, 144.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1252)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 139056 reads in 29.481 CPU sec, 14.576 real sec\n",
      "[M::process] read 137142 sequences (20000234 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 36704, 68, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (374, 441, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 893)\n",
      "[M::mem_pestat] mean and std.dev: (454.77, 138.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (29, 83, 441)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1265)\n",
      "[M::mem_pestat] mean and std.dev: (172.65, 237.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1677)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137238 reads in 28.884 CPU sec, 14.255 real sec\n",
      "[M::process] read 137232 sequences (20000057 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 36412, 67, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 432, 869)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2363)\n",
      "[M::mem_pestat] mean and std.dev: (562.90, 500.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3110)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (375, 440, 545)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (35, 885)\n",
      "[M::mem_pestat] mean and std.dev: (454.04, 138.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1055)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (33, 87, 255)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 699)\n",
      "[M::mem_pestat] mean and std.dev: (114.88, 127.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 921)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137142 reads in 29.148 CPU sec, 14.409 real sec\n",
      "[M::process] read 77830 sequences (11346582 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 36535, 44, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (377, 441, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (35, 890)\n",
      "[M::mem_pestat] mean and std.dev: (455.72, 137.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1061)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (32, 80, 247)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 677)\n",
      "[M::mem_pestat] mean and std.dev: (118.55, 141.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 892)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137232 reads in 29.156 CPU sec, 14.442 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 20678, 33, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (375, 439, 545)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (35, 885)\n",
      "[M::mem_pestat] mean and std.dev: (453.48, 137.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1055)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (18, 50, 211)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 597)\n",
      "[M::mem_pestat] mean and std.dev: (101.03, 118.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 790)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 77830 reads in 16.554 CPU sec, 8.226 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 2 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/P2.R1.fastq.gz ./metagenomics/reads/P2.R2.fastq.gz\n",
      "[main] Real time: 212.399 sec; CPU: 425.240 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 137266 sequences (20000094 bp)...\n",
      "[M::process] read 137098 sequences (20000097 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 33423, 65, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 440, 546)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 896)\n",
      "[M::mem_pestat] mean and std.dev: (450.26, 143.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1071)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (38, 88, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1265)\n",
      "[M::mem_pestat] mean and std.dev: (167.62, 221.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1674)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137266 reads in 29.945 CPU sec, 14.855 real sec\n",
      "[M::process] read 137200 sequences (20000168 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 33057, 55, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (260, 429, 676)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1508)\n",
      "[M::mem_pestat] mean and std.dev: (299.73, 193.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1924)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 440, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 899)\n",
      "[M::mem_pestat] mean and std.dev: (450.77, 144.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1075)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (27, 77, 258)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 720)\n",
      "[M::mem_pestat] mean and std.dev: (122.42, 140.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 951)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137098 reads in 30.566 CPU sec, 15.123 real sec\n",
      "[M::process] read 137298 sequences (20000189 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (16, 33184, 55, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (137, 367, 425)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1001)\n",
      "[M::mem_pestat] mean and std.dev: (321.44, 222.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1289)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 441, 546)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 896)\n",
      "[M::mem_pestat] mean and std.dev: (450.78, 143.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1071)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (38, 96, 312)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 860)\n",
      "[M::mem_pestat] mean and std.dev: (158.52, 178.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1134)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137200 reads in 30.527 CPU sec, 15.077 real sec\n",
      "[M::process] read 137344 sequences (20000099 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 33547, 69, 9)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (131, 323, 452)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (299.69, 165.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1415)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 441, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (16, 901)\n",
      "[M::mem_pestat] mean and std.dev: (449.70, 144.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1078)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (45, 90, 263)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 699)\n",
      "[M::mem_pestat] mean and std.dev: (118.83, 124.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 917)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137298 reads in 30.622 CPU sec, 15.148 real sec\n",
      "[M::process] read 137568 sequences (20000130 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 33561, 69, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (213, 371, 480)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1014)\n",
      "[M::mem_pestat] mean and std.dev: (303.00, 145.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1281)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 439, 543)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 889)\n",
      "[M::mem_pestat] mean and std.dev: (448.28, 143.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1062)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (48, 79, 327)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 885)\n",
      "[M::mem_pestat] mean and std.dev: (138.23, 173.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1164)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137344 reads in 30.916 CPU sec, 15.287 real sec\n",
      "[M::process] read 138204 sequences (20000225 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 33494, 65, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (95, 272, 470)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1220)\n",
      "[M::mem_pestat] mean and std.dev: (267.46, 170.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1595)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 439, 544)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 892)\n",
      "[M::mem_pestat] mean and std.dev: (448.70, 142.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (20, 60, 193)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 539)\n",
      "[M::mem_pestat] mean and std.dev: (86.09, 105.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 712)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137568 reads in 30.755 CPU sec, 15.186 real sec\n",
      "[M::process] read 139026 sequences (20000266 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 33751, 62, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 566, 719)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1515)\n",
      "[M::mem_pestat] mean and std.dev: (543.67, 301.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1913)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 439, 542)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (32, 882)\n",
      "[M::mem_pestat] mean and std.dev: (448.67, 139.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1052)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (39, 64, 229)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 609)\n",
      "[M::mem_pestat] mean and std.dev: (114.25, 133.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 799)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 138204 reads in 30.881 CPU sec, 15.259 real sec\n",
      "[M::process] read 138412 sequences (20000141 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 33945, 76, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (90, 321, 826)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2298)\n",
      "[M::mem_pestat] mean and std.dev: (434.82, 499.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3034)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (368, 437, 535)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 869)\n",
      "[M::mem_pestat] mean and std.dev: (443.28, 138.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1036)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (41, 89, 174)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 440)\n",
      "[M::mem_pestat] mean and std.dev: (93.58, 79.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 573)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 139026 reads in 31.000 CPU sec, 15.316 real sec\n",
      "[M::process] read 137634 sequences (20000156 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 33974, 60, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 438, 538)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (31, 876)\n",
      "[M::mem_pestat] mean and std.dev: (443.96, 139.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1045)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (37, 75, 281)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 769)\n",
      "[M::mem_pestat] mean and std.dev: (131.33, 162.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1013)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 138412 reads in 30.559 CPU sec, 15.091 real sec\n",
      "[M::process] read 137590 sequences (20000069 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 33764, 54, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (304, 355, 433)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 691)\n",
      "[M::mem_pestat] mean and std.dev: (337.22, 109.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 820)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (368, 438, 537)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (30, 875)\n",
      "[M::mem_pestat] mean and std.dev: (443.66, 140.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1044)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (54, 137, 479)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1329)\n",
      "[M::mem_pestat] mean and std.dev: (237.60, 286.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1754)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137634 reads in 30.504 CPU sec, 15.069 real sec\n",
      "[M::process] read 137592 sequences (20000141 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 33711, 66, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (291, 358, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 969)\n",
      "[M::mem_pestat] mean and std.dev: (360.50, 148.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1195)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 439, 541)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 883)\n",
      "[M::mem_pestat] mean and std.dev: (446.74, 140.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1054)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (38, 94, 358)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 998)\n",
      "[M::mem_pestat] mean and std.dev: (140.57, 157.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1318)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137590 reads in 30.500 CPU sec, 15.088 real sec\n",
      "[M::process] read 137656 sequences (20000043 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (16, 33543, 56, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (177, 406, 511)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1179)\n",
      "[M::mem_pestat] mean and std.dev: (332.07, 205.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1513)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 438, 540)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 882)\n",
      "[M::mem_pestat] mean and std.dev: (445.26, 142.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1053)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (59, 99, 261)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 665)\n",
      "[M::mem_pestat] mean and std.dev: (145.65, 154.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 867)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137592 reads in 30.626 CPU sec, 15.132 real sec\n",
      "[M::process] read 137636 sequences (20000050 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (15, 33668, 73, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (316, 342, 531)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 961)\n",
      "[M::mem_pestat] mean and std.dev: (379.64, 136.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1176)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 438, 539)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (29, 879)\n",
      "[M::mem_pestat] mean and std.dev: (445.54, 140.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1049)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (34, 83, 266)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 730)\n",
      "[M::mem_pestat] mean and std.dev: (138.68, 152.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 962)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137656 reads in 30.778 CPU sec, 15.225 real sec\n",
      "[M::process] read 72476 sequences (10483902 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 33669, 64, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (272, 399, 469)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 863)\n",
      "[M::mem_pestat] mean and std.dev: (324.64, 155.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1060)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (368, 438, 538)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 878)\n",
      "[M::mem_pestat] mean and std.dev: (444.41, 140.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1048)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (30, 74, 231)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 633)\n",
      "[M::mem_pestat] mean and std.dev: (108.95, 131.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 834)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 137636 reads in 30.847 CPU sec, 15.296 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 17676, 40, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 438, 537)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (36, 871)\n",
      "[M::mem_pestat] mean and std.dev: (444.15, 138.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1038)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (20, 78, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 959)\n",
      "[M::mem_pestat] mean and std.dev: (121.06, 167.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1272)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 72476 reads in 16.296 CPU sec, 8.079 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 2 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/P3.R1.fastq.gz ./metagenomics/reads/P3.R2.fastq.gz\n",
      "[main] Real time: 223.183 sec; CPU: 447.090 sec\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "samples=$(find ./metagenomics/reads -name '*.fastq.gz' -type f -printf '%P\\n' | cut -d '.' -f 1 | sort | uniq)\n",
    "for i in $samples\n",
    "    do echo $i\n",
    "    bwa mem -t 2  ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/\"$i\".R1.fastq.gz ./metagenomics/reads/\"$i\".R2.fastq.gz | samtools view -h -b > ./metagenomics/mapping/mapped.\"$i\".bam \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!bwa index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "samtools sort help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools sort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "for i in ./metagenomics/mapping/*.bam ;\n",
    "    do samtools sort  -o \"$i.sorted\" \"$i\"\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove the samtools view directory:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/laura/gitprojects/metagenomicspractical\n",
      "2.0G\t./metagenomics/assembly\n",
      "946M\t./metagenomics/reads\n",
      "4.0K\t./metagenomics/sorted\n",
      "2.8G\t./metagenomics/mapping\n",
      "5.7G\t./metagenomics/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "pwd\n",
    "du -h ./metagenomics/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!rm -rf ....yourdirectoryname"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "samtools help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools view"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run samtools view to view your BAM files:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools view ....yourbamsortedbam | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2> Binning with MetaBat </h2>\n",
    "\n",
    "Now that we have the reads, the scaffolds and the mapping of the reads on the scaffolds we can continue with the binning. In metagenomics, binning is the process of grouping reads or contigs and assigning them to operational taxonomic units. Binning methods can be based on either compositional features or alignment (similarity), or both. Metabat uses both the contig depth and tetra-nucleotide frequencies to bin the contigs. Every bin will represent one operational taxonomic unit that can be found in the metagenome.\n",
    "\n",
    "<b>Assignment:</b><br>\n",
    "Run the script provided in /metagenomics/scripts/jgi_summarize_bam_contig_depths to calculate the average depth per contig. <br>\n",
    "Then run metabat to bin the contigs, dont forget to include the previously included contig depths in the metabat command."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/sh: 1: ./metagenomics/scripts/jgi_summarize_bam_contig_depths: not found\r\n"
     ]
    }
   ],
   "source": [
    "!./metagenomics/scripts/jgi_summarize_bam_contig_depths -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "calculate the contig depth per scaffold<br>\n",
    "hint: a glob looks like this directoryname/* and includes all files included in the directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!./metagenomics/scripts/jgi_summarize_bam_contig_depths --outputDepth "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "metabat help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!metabat -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a new directory:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run metabat:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!metabat -i -o -t"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2> Checking completeness and contamination using CheckM </h2>\n",
    "\n",
    "Now that we have the bins made with metabat we can check them for contamination and completeness (quality), for this we will use CheckM. CheckM provides a set of tools for assessing the quality of genomes recovered from isolates, single cells, or metagenomes. It provides robust estimates of genome completeness and contamination by using collocated sets of genes that are ubiquitous and single-copy within a phylogenetic lineage (also called marker/signature genes). http://ecogenomics.github.io/CheckM/\n",
    "\n",
    "As you will be able to see in the checkm help pages, checkm has a workflow (lineage_wf) that will run all nessecary steps to asses bin quality. \n",
    "\n",
    "Lineage_wf (lineage-specific workflow) steps: <br>\n",
    "- The tree command places genome bins into a reference genome tree. <br>\n",
    "- The lineage_set command creates a marker file indicating lineage-specific marker sets suitable for evaluating each genome. <br>\n",
    "- This marker file is passed to the analyze command in order to identify marker genes and estimate the completeness and contamination of each genome bin.  <br>\n",
    "- Finally, the qa command can be used to produce different tables summarizing the quality of each genome bin. <br>\n",
    "\n",
    "\n",
    "<b>Assignment:</b><br>\n",
    "Sadly for this exercise the virtual machines we are using are not powerfull enough, therefore we provide the results of the first steps of the CheckM workflow up until the qa command (lineage_wf). Scan the help pages of CheckM to find out the correct command to finish the CheckM analysis. \n",
    "\n",
    "(OPTIONAL: you can try to find out what the limiting factor is of lineage_wf using !/usr/bin/time --verbose) <br>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!checkm -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!checkm lineage_wf -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "checkm qa help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!checkm qa -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run checkm qa:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!checkm qa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "(think and discuss these questions) <br>\n",
    "\n",
    "What did you do? <br>\n",
    "Where is your output? <br>\n",
    "What does your output look like? <br>\n",
    "What can you say about the bins with this output? <br>\n",
    "What can you say about lineages of the bins?<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2>Genome annotation with Prokka</h2>\n",
    "\n",
    "Now that we some extra information about our bins, we can continue to analyze the high quality bins. The final CheckM results will give you a good overview of the bins with low contamination and high completeness and also shows the lowest taxonomic rank of the bin. Pick a bin that you think is interesting to further study.\n",
    "\n",
    "With this bin we are going to do some genome annotation. Whole genome annotation is the process of identifying features of interest in a set of genomic DNA sequences, and labelling them with useful information. Prokka is a software tool to annotate bacterial, archaeal and viral genomes quickly and produce standards-compliant output files.\n",
    "\n",
    "<b>Assignment:</b><br>\n",
    "Have a look at the prokka help pages below and come up with the right command to run prokka. (HINT: look at the Usage tag to run prokka on default mode, think of threads, use --centre X --compliant to stop prokkas complaints about ugly contig names, direct the output towars metagenomics/prokka)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!prokka"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a directory for the prokka output:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "prokka help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!prokka -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run prokka:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "!prokka --cpus --outdir "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "(think and discuss these questions) <br>\n",
    "\n",
    "What did you do? <br>\n",
    "Where is your output? <br>\n",
    "What does your output look like? <br>\n",
    "Do you know what all the output files are or mean? <br>\n",
    "\n",
    "(HINT: you can look at the prokka files in the same way we looked at the reads and scaffold earlier)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>investigating prokka output</b><br>\n",
    "to investigate the prokka output you can use two webservers that both are able to place the annotations from prokka in KEGG pathways.<br>\n",
    "\n",
    "(1) prokka gives uniprot IDs in the gff files first we will collect these IDs,\n",
    "(2) since we have Uniprot IDs from prokka we are going to convert these to KO IDs that can be used by KEGG using this website:<br>\n",
    "http://www.uniprot.org/uploadlists/\n",
    "\n",
    "(3) then you can put your IDs in both websites and investigate the pathways\n",
    "http://www.genome.jp/kegg/tool/map_module.html<br>\n",
    "http://pathways.embl.de/iPath2.cgi#<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "view prokka output:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!less ....yourprokkaoutput.gff | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "take the uniprot IDs out of the gff file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!grep -o 'UniProt.*' ....yourprokkaoutput | cut -d';' -f1 | cut -d':' -f2 > listofuniprotIDs"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
