{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# General introduction\n",
    "Today, you will assemble and annotate a metagenome of a plant: *Azolla filiculoides*. *Azolla* is known for its endophytes: microorganisms that live inside the plant. These microorganisms live specifically inside a leaf cavity in every upper leaf lobe of the plant. One of these endophytes has been known for a long time: *Nostoc azollae* They are best known for fixing N^2 from the air, in service of the plant. The image included below illustrates the physiology of these leaf cavities. \n",
    "\n",
    "While the presence of *N. azollae* is well studied, the presence of other microbes in the *Azolla* leaf pocket is not. Today, we will create and analyse the metagenome of *A. filiculoides* plants, in search for these other \"neglected\" microbes. \n",
    "\n",
    "![image](http://theazollafoundation.org/wp-content/uploads/2013/05/Azolla-leaf-cavities-with-Anabaena.png)\n",
    "\n",
    "Hence, the open biological question we are answering today is \"which species live inside these leaf cavities and what are the metabolic capacities encoded on their genomes?\" Based on these genomes, you may be able to infer their function and think of a hypotheses to test. \n",
    "\n",
    "*Tip*: If you stumble upon a step with a long calculation time, first read ahead to see what steps come next, and otherwise you can google some information about the plant you are researching. Notice that this [paper](http://onlinelibrary.wiley.com/doi/10.1111/nph.14843/full) specifically describes the data that we will be using today. For a more general introduction into Azolla, perhaps have a look at [this short movie](https://youtu.be/O34gTsxyDq8) or [this short movie](https://youtu.be/OI4VV4M2-f4) (Dutch) recorded for Dutch local television.\n",
    "\n",
    "\n",
    "# Workflow\n",
    "\n",
    "## Acquire Samples\n",
    "A typical metagenomics workflow starts in the environment and/or the wetlab with a DNA extraction from some biological sample. This is also true for this workflow. We have extracted DNA from *Azolla* from the environment in two ways. Firstly, the whole Plant (P), secondly from the Leaf cavities (L). Both were collected in three biological replicates. Hence the samples we will work with are coded L1 L2 L3 and P1 P2 P3. The reasoning behind these two sampling types is that true endophytes are expected to be more abundant in leaf samples than in plant samples. So we may later use these different sampling types to infer if a microbial genome that we assembled originated from the leaf cavity, or from somewhere else in or on the plant. \n",
    "\n",
    "Extracted DNA is processed in a sequencing library; it is prepared in a way that a [Illumina sequencing](https://en.wikipedia.org/wiki/DNA_sequencing#Illumina_(Solexa)_sequencing) machine can process it. The sequencing machine then provides us with [FastQ](https://en.wikipedia.org/wiki/FASTQ_format) files, you may have handled these before, perhaps not. We will have a look at FastQ files later in this practical.\n",
    "\n",
    "## First Quality Control and processing.\n",
    "FastQ files contain the output of the Illumina sequencing machine: tens of millions of short DNA sequences. Where short is anything between 50 and 250 bp. These original short DNA sequences are further refered to as DNA reads, or just reads. FastQ files contain unprocessed output of the machine, every read consists of four lines: \n",
    "1. a header proceeded by the '@' character. \n",
    "2. the DNA sequence\n",
    "3. a comment line proceeded by a '+'\n",
    "4. a quality string.\n",
    "\n",
    "The quality string has the exact same length as the DNA read, every character in the quality string corresponds to some number between 0 and 40. The sequencing machine assigns this quality score based on \"how sure it is\" that this specific base is correct. To put it a bit loosly.\n",
    "\n",
    "These are two reads encoded in a FastQ file.\n",
    "\n",
    "> @NS500813:28:H3M3VAFXX:1:11101:19270:1015 1:N:0:CGATGT\n",
    "GNGGTGAAGAAATCAGCCATTCTAAACCAATTGCTCTCCAAGGATTATCAGGTGCTTTTTCACCTTGCGTCCAAGAAACCAAAATATTGAATATGAAGGGTAAGGTGGACATTCCTAATAAGAATGCACCAAGACTAGCGAGAACATTCC\n",
    "+\n",
    "A#AAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEAEE<EEEEEEEEEEEEEEEEEEEEEAEEEAEEEEEEEEEEEEEEEEEEEEAAAEEAEEEEEAAAAA6AEAEAAEEEEEEEEEEEEA6\n",
    "@NS500813:28:H3M3VAFXX:1:11101:24948:1015 1:N:0:CGATGT\n",
    "ANATTAATTATACAATCACTAATTTAGCAGAGCATTTAGAAAGTATAAGCCATGATGCAATTAACTATTATTTAAAAACCGAAAAGTTAACATCTCGTTTACTATGGGATAAGGTGAAAGAGGTAGTAGAACCTGATGGTAATGGGTACAT\n",
    "+\n",
    "A#AAAAEEEEE6EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE/EEAEEEEEEEAEEEEEEEEEEE<EEE<EEEEE66EEA<EEEEAEEEEEEEEEEEAE<E/AE/EEEEEAEAAEAAAEEEEAEEEE6<EEAEA<//AAEEAEE<AA\n",
    "\n",
    "FastQ files often contain some reads of poor quality, we definitively want to filter those out before starting our analyses. There are many ways available, for todays practical, this was done for you already with a tool called [trimmomatic](https://academic.oup.com/bioinformatics/article-abstract/30/15/2114/2390096).\n",
    "\n",
    "Before and after trimming, general Quality Controll (QC( of a FastQ file is assessed with a tool called FastQC. The FastQC reports are rendered as html pages so we can view them in our webbrowsers. You can find the FastQC reports for this practical here:\n",
    "\n",
    "### Filtering\n",
    "Finally, we chose here to filter the sequencing data for plant DNA, since we are only interested in microbial DNA. The sequencing data was rid of any plant DNA by mapping (aligning) the reads to a reference plant genome. Only the sequencing data that did not map anything was kept for further analysis.\n",
    "\n",
    "### laura add fastQC links\n",
    "\n",
    "\n",
    "## Assembly\n",
    "Assmbly is the process of combining these many short DNA reads in longer contiguous strings of DNA: contigs. For the assembly process we have used both sample types and all biological replicates as if they were one big sample. This yields the best assembly results. A metagenome assembly is different than a regular genome assembly in that it has DNA of multiple species in one sequencing file. Metagenome assembly was achieved with [SPAdes](http://online.liebertpub.com/doi/abs/10.1089/cmb.2012.0021). \n",
    "\n",
    "Some more elaborate information of Assembly\n",
    "> The process of assembly takes the millions of short reads in a fastq file, and processes these to thousands of longer nucleotide sequences. Typically the output is a fasta file with contiguous sequences called contigs (contigs.fasta) or a fasta file with multiple contigs scaffolded based on their relative position and orientation to each other called scaffolds, in a file scaffolds.fasta. Ideally, these scaffolds represent the genome of a single species in the metagenome. This is however, extremely unlikely. At least in my personal experience. In reality the assembly process wil produce thousands of longer scaffolds where a single species' genome is still represented by multiple sequences in the assembly fasta file.\n",
    "\n",
    "Also, you probably know what a fasta file is. It is a text file with DNA or protein sequences. A fasta file can contain multiple sequences. Fasta files are similar to fastq files, but a bit simpeler. There are just two types of lines. The header line is preceded by a `>` and the lines after contain the DNA sequence. The DNA sequence can continue until the next `>` or until the end of the document. We may have a quick look at a fasta file:\n",
    "> \\>Sequenceheader <br>\n",
    ">  ACTCATGAGACTAGACA\n",
    "\n",
    "\n",
    "# Here you will take over the workflow\n",
    "We have provided you with a subset of the assembly and a subset of the reads. We have made subsets rather than the whole sets to keep this one day practical feasible. The complete workflow you will follow, is seen on this sketch below. Whenever you are lost in commandline, error messages and output tablaes, have a look again at this flowchart. In these bio-informatics workflows it is essential to keep a feeling for where you are, where you are going and where you come from.  \n",
    "![image](./metagenomics/workflowsketch.png)\n",
    "\n",
    "The programmes mentioned in the flowchart are preinstalled on this virtual machine you are currently working on:\n",
    "bwa: http://bio-bwa.sourceforge.net/ <br>\n",
    "samtools: http://samtools.sourceforge.net/ <br>\n",
    "binningmetabat: https://bitbucket.org/berkeleylab/metabat/overview <br>\n",
    "checkm: https://github.com/Ecogenomics/CheckM/wiki/Installation#how-to-install-checkm <br>\n",
    "prokka: http://www.vicbioinformatics.com/software.prokka.shtml <br>\n",
    "\n",
    "\n",
    "## Jupyter and Bash Basics\n",
    "Before we will get our hands dirty with the 'real data', we will first do some exercises to learn how to work in a notebook like this then we will continue the metagenomics workflow. To keep all information central, I will continue here with the explaination of the workflow. The exercises will be available in another notebook after you have read the full introduction.\n",
    "\n",
    "## Backmapping\n",
    "After assembly of the scaffolds we want to know how abundant each scaffold is in the different metagenomic samples. To do this, we align the illumina reads (FastQ files) to the scaffolds (fasta) of the assembly in a step called **backmapping** with a tool called BWA (Burrows-wheeler aligner). BWA will provide us then with a table of which reads mapped on the metagenome assembly and their specific coördinates. This table is stored in a `.bam` file. Finally, these tables in the `.bam` files must be sorted by coördinate. This we achieve with a tool called `samtools`.\n",
    "\n",
    "## Binning\n",
    "A single metagenome assembly contains scaffolds representing DNA of many species in the original sample, but without any information on which scaffold belongs to which species. To tackle this issue, we will categorise each individual scaffold in a 'bin' which then ideally comprises one species in the metagenome. This procedure is called binning. The binning procudure needs the bam files we will make in the 'backmapping' step. The tool we use for binning is called `metabat`. Metabat uses the sorted bamfiles to subdivide the scaffolds in bins. This will produce between 5 and 20 bins. Hence we have moved from millions of very short sequences with very little biological information to thousands of longer scaffolds with more biological information, to dozens of long sequences with a lot of biological information. \n",
    "\n",
    "## QC\n",
    "One bin ideally represents one species. To check whether this is the case, we can use the tool `checkM`. CheckM assesses the completeness and contamination of bins based on single-copy-marker-genes (collocated sets of genes that are ubiquitous and single-copy within a phylogenetic lineage). \n",
    "\n",
    "## Annotation\n",
    "For the sake of simplicity and time, we will isolate one bin per group and annotate this bin using prokka. Annotation is the process of adding functional information to parts of DNA sequence, i.e. making a table of genes and locations of these genes. These are often stored in a GFF file, the genomic feature format. Once you have this gff file, and a fasta for your specific bin/microbial species, we will visualise it using KEGG. Again, for more details please google kegg. Here we will use it to load our gff file and then explore the metabolic pathways present. \n",
    "\n",
    "\n",
    "# Final remarks\n",
    "Not all steps are feasible on these humble web based machines. Be paitient and don't overload the machines, if we crash the server we all loose acces to the notebooks.\n",
    "\n",
    "Good luck! and remember Google is your friend!!!"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# How to work in a Jupyter notebook with the Bash language\n",
    "The webpage you are looking at, is called a jupyter notebook. It is a webpage on which you can write text (like this text) and also code. The code can ben executed an the output you see again within the same notebook! This may sound trivial, but it's really \"cool\". To put it in non-scientific terms. Code and text are entered in individual cells, a code cell or a text cell. This cell you are reading now is obviously a text cell. Now let's look at a code cell."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hello world\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"hello world\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So what happened in the above code cell. First I declared that I want to write code in bash by typing\n",
    "> %%bash\n",
    "\n",
    "then the first word you type in bash is always the command: in this case this is \n",
    "> echo\n",
    "\n",
    "this command 'echoes' whatever you give it in the terminal.\n",
    "\n",
    "after you 'call' the command, you give it an argument in this case that is \n",
    "> \"hello world\"\n",
    "\n",
    "This basic structure of 'command' 'arguments' commes back through this practical. \n",
    "\n",
    "*do* now try to change \"hello world\" to something else"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Often, an argument is a path to a file. To see what files we have, we have the `ls` command. (short for list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COO1_C002.ipynb               dna181_circos.conf\r\n",
      "COO3.ipynb                    dna181_denovo.vcf\r\n",
      "COO4+Pipeline+building.ipynb  dna181_illumina.bam_CNVs\r\n",
      "COO5a.ipynb                   \u001b[0m\u001b[01;36mHomo_sapiens.GRCh37.GATK.illumina\u001b[0m@\r\n",
      "COO5b.ipynb                   \u001b[01;34mmetagenomics\u001b[0m/\r\n",
      "COO6B_github.ipynb            metagenomics.ipynb\r\n",
      "COO6.ipynb                    NGS analyses.ipynb\r\n",
      "\u001b[01;34mCWLs\u001b[0m/                         sv.links\r\n",
      "\u001b[01;36mdatafiles\u001b[0m@                    \u001b[01;32mWES_analysis_pipeline.sh\u001b[0m*\r\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Files are black, folders are purple and trailed by a '/'   Now lets use the `ls` command to see what is in the metagenomics folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "assembly\n",
      "mapping\n",
      "reads\n",
      "sorted\n",
      "workflowsketch.png\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls metagenomics/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are learning fast. So we now know what a `command` is, we know what an `argument` is. Finally we also need to know what options are. options are often provided in between the `command` and the `argument`. They look either like this\n",
    "> ls --size --human-readable\n",
    "\n",
    "or in shortened versions like this\n",
    "> ls -sh\n",
    "\n",
    "Note that the above two command are synonimous. Try out in the cell below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 652K\n",
      "4.0K assembly\n",
      "4.0K mapping\n",
      "4.0K reads\n",
      "4.0K sorted\n",
      "636K workflowsketch.png\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "ls --size --human-readable metagenomics/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finally, you may also choose not to specify `%%bash` at the beginning of a cell, but the precede a command with an exclamation mark instead. Like so"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 652K\r\n",
      "4.0K assembly  4.0K mapping  4.0K reads  4.0K sorted  636K workflowsketch.png\r\n"
     ]
    }
   ],
   "source": [
    "!ls -sh ./metagenomics/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Wether you choose to use `%%bash` or `!` . Both work. Realise however that a `!` only works for that specific line you type in, and `%%bash` works for the entire cell. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "<h2> OLD Practical information </h2>\n",
    "This whole computer practicum will take a whole day, staff will be present all day to help you in this process. However, we are with very few so please assisst each other where possible. \n",
    "\n",
    "You will work in this python notebook, combining bash and python commands. Bash commands can be executed by appending an exclamation mark '!' to the command like so:\n",
    "\n",
    "> !whoami\n",
    "\n",
    "Or by typing \"%%bash\" at the start of your code block, like so:\n",
    "\n",
    "> %%bash\n",
    "\n",
    "\n",
    "You are working on the virtual machine, which you have used before, did you know you can check the computer power of this machine using bash commands:\n",
    "\n",
    "number of cpu threads\n",
    "\n",
    "> !nproc\n",
    "\n",
    "available GB RAM memory\n",
    "\n",
    "> !top\n",
    "\n",
    "\n",
    "The first bash commands we will use are 'mkdir' 'wget' and 'ls'. Go ahead and see what they do. If you wonder, most programms print a short summary of their function when presented the -h flag like so:\n",
    "\n",
    "> !ls -h\n",
    "\n",
    "or \n",
    "\n",
    "> !ls --help\n",
    "\n",
    "or \n",
    "\n",
    "> !mkdir -h\n",
    "\n",
    "\n",
    "If you made a directory you want to remove again you can use:\n",
    "\n",
    "> !rm -rf directory\n",
    "\n",
    "but be carefull with the rm command because you will not be able to recover the removed file.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Raw data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b> sequencing reads </b><br>\n",
    "The sequencing reads are already available in the folder 'metagenomics/reads', to keep the practical feasible we have made a subset of the original data. <br>\n",
    "\n",
    "In this reads folder you will find 12 files. <br>\n",
    "L and P stand for the leaf and plant samples. <br>\n",
    "L1 L2 L3 are the biological replicates. R1 and R2 represent the forward and reverse illumina reads. So 2 samples * 3 replicates * 2 directions = 12 files <br>\n",
    "Files are stored in a fastq format (.fastq) and compressed in gzip archives (.gz). The gzip compression format is widely used in the genomics field, fastq files often compress down to only a quarter of the filesize with gzip! You do not need to extract these. <br>\n",
    "\n",
    "A good practice is to always peek into a file just to check the format. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This command allows you to see the files in the reads directory.   'ls' is short for 'list'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L1.R1.fastq.gz\tL2.R2.fastq.gz\tP1.R1.fastq.gz\tP2.R2.fastq.gz\r\n",
      "L1.R2.fastq.gz\tL3.R1.fastq.gz\tP1.R2.fastq.gz\tP3.R1.fastq.gz\r\n",
      "L2.R1.fastq.gz\tL3.R2.fastq.gz\tP2.R1.fastq.gz\tP3.R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls metagenomics/reads/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To view in a gzipped file you can use the zcat command. \n",
    "\n",
    "The '|' sign is called a 'pipe' You can use it to feed the output of one programme to the next. Never forget to pipe the output to head when viewing/reading a file, like this \"| head\". The files we work with are big and will crash your jupyter-notebook instance if you don't use head."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@NS500813:28:H3M3VAFXX:1:11101:21566:1019 1:N:0:CAGATC\r\n",
      "AAAAAATTAAAATAATTGGCTGAAAATGAAAAAAGAAAGCTTGATCATATGTTAAGCCTTTTGTAACGGAACAAAGTGATTAAATTATCTCTACAGTGTTTTTTCGATTTATTGATGGATTAAGAACAACCGTCACACACCAATAGAATAT\r\n",
      "+\r\n",
      "AAAAAEEEE6EEEEEEEEEEEEEEEEEEEEEAEEEEEEEEE</EE<EEEEEEEE/AA6E<E/E/</EEEAE/EEEEEEE<EEE//EEEEE<EE/EEAE<E<E//AE/E<EE/<AEEEAE66<E<<E</<AA<E/AEE/E//<<EEEEE/AA\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:14840:1067 1:N:0:CAGATC\r\n",
      "GTGGGATTTCCAGTGATTTGAATTCTTACAAAAGTCGCTTGCTTTTGTTTTTATTGCCACCTATTTCGTTGAATAATTATTAGAAAGGGAACAAGGCTTTTTGACGATAAAAAGTTATTTTTAATTGGAGGTCGCTAGAATAGCTTTGTT\r\n",
      "+\r\n",
      "A/AAAEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEAEEAEEEEEEEEEEEEEAEEE/EE/EEEEEEAEEEEEEEEEAEEEEEEEEEA/EEEAEAA<EEEEEEEEEAAEA/A/AA/6A<E<AE/EA6<A\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:20510:1074 1:N:0:CAGATC\r\n",
      "CCCGGTGCTGAATGAGAATGATTCGTGTTTAGATTAGCAGGCGCTGATGGTAGTAGTAGCGGCAATGAGAGTGATTCGTATTTAGATCCAGGCAGGGGCGGAAGTTTCATAATGTGGAATGCGCCCTGGTGCGCTTTACAATGTAAAGC\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "!zcat metagenomics/reads/L1.R1.fastq.gz | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using grep we can select for lines in a file that contain only the word/key specified in the grep command. As you can see in the above command, the headers of the reads always start with the \"@\", therefor if we only want to see the headers we can simply grep for the \"@\" using the command below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@NS500813:28:H3M3VAFXX:1:11101:21566:1019 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:14840:1067 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:20510:1074 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:5773:1086 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:2702:1014 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:6665:1015 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:6357:1086 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:18737:1028 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:1677:1028 1:N:0:CAGATC\r\n",
      "@NS500813:28:H3M3VAFXX:1:11101:26277:1020 1:N:0:CAGATC\r\n",
      "grep: write error: Broken pipe\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "!zcat metagenomics/reads/L1.R1.fastq.gz | grep '@' | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now if we want to count the number of reads in the fastq file we can count the number of headers, since every entry has one header. To count the number of headers we will first have to read in the gzip file, then grep on the headers, and then we can use the wordcount command to count the number of lines."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1000000\r\n"
     ]
    }
   ],
   "source": [
    "!zcat metagenomics/reads/L1.R1.fastq.gz | grep '@' | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### wrap up\n",
    "So now you have executed some bash commands and investigated the output. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Assembled data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Calculating a de-novo metagenome assembly is beyond the scope of this practical, so we have done this for you. You will however analyse this assembly yourself! :)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "View the inside of the assembly directory:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "allscaffolds.fasta.gz  scaffolds.fasta.ann  scaffolds.fasta.pac\r\n",
      "scaffolds.fasta        scaffolds.fasta.bwt  scaffolds.fasta.sa\r\n",
      "scaffolds.fasta.amb    scaffolds.fasta.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls metagenomics/assembly/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now your first assignment is to dowload the assemblies from the data repository, adapting the commands above. Below you will find a code block with the template command that you can finish to download the the assemblies. Use multiple code blocks if convenient. You can make new code blocks by clicking on the left side of a code block and then type \"a\" or \"b\" to create a code block above or below the original code block. You can delete a code block by clicking on the left side of a code block and typing \"d\" twice."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Edit this line to have a look into the assembly file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gzip: metagenomics/assembly/....assemblyfile.gz: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "!zcat metagenomics/assembly/....assemblyfile | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Grep the headers of the assembly file, what word/key would work for fastas when viewing the assembly entries above"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!zcat metagenomics/assembly/....assemblyfile | grep \"\" | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Count the scaffolds in the assembly like we did earlier with the fastq.gz file. \n",
    "\n",
    "Remember not to over-load the webinterface!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!zcat metagenomics/assembly/....assemblyfile | grep \"\" ......"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Rename the assembly file:   (mv is short for move)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mv metagenomics/assembly/....oldname  metagenomics/assembly/....newname"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check if the rename worked:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!ls metagenomics/assembly/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Did your assembly downloaded correctly?<br>\n",
    "Does the content of the fasta make sense?<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2>Length distributions of the scaffolds</h2>\n",
    "Now that we have the assembly, we will do some quick analyses to get a idea of the quality. First we will plot the length distribution of the scaffolds in the assembly. Luckily for us, the length of each sequence in the fasta is already embeded in each fasta header. We can easily extract these numbers and plot them in python.\n",
    "\n",
    "To plot the length distribution run the python code below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scaffolds.fasta.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls ./metagenomics/assembly/\n",
    "gunzip --keep ./metagenomics/assembly/scaffolds.fasta.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAnQAAAJhCAYAAADIci3nAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMS4xLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvAOZPmwAAIABJREFUeJzt3X1UlHX+//HXwHgTYoCDoShrqbmllUijKVaCUna3pW2abdsvJcv7IutsLSe3+ibf7FumlVqsqe3qdzetNG3Lasm0FE0GwkytoLvNo4GAKd6FMNfvD37MTxR0UJiLj/N8nNM5zmfmuq739WYYXn2um3FYlmUJAAAAxgqxuwAAAACcGQIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdEAzM3r0aKWkpNhdRi3nn3++pk+f3iTrasx1H+/4XjZ1b5OSkjR27NgmW38g7d+/X8OHD9e5554rh8OhH374QZWVlUpNTZXL5ZLD4dDatWv9Wpc/P+Pm+L4HTOK0uwAAzcf06dP16quv6ocffgjYNnNychQWFubXa9evX6+rrrpK33//vc4///xTvv6FF16Q1+s9wwpPVF+fli9fLqfz7PhYffnll7Vx40Zt2LBB7du3V/v27fXmm2/qH//4h9asWaOuXbuqXbt2dpcJ4P85Oz55ABirffv2jb7Oo0ePyul0KiIiotHXfTJnU8ApKChQr169dOmll9Ya69SpkxITE22sDEBdOOQKGOD1119XfHy8WrdurfPPP19Tp07VwYMHfc/XHOp76qmn1KFDB7Vr106jR4+u9Rqv16v09HS1b99e4eHhGjVqlGbPnu2bUXrttdc0bdo0/fjjj3I4HHI4HHriiSd8y1dUVOiBBx5Qu3btFBMTo4cfflhVVVUnrXvLli1KTExU69at1aNHDy1btuyE1xx/OG7lypXq06ePwsLCFBkZqX79+unzzz/XDz/8oKuuukqSdMEFF8jhcCgpKUnS/z9c99JLL+n8889Xq1atdPDgwXoP4z3//PPq1KmTwsLC9Pvf/14lJSW+5+paZsmSJXI4HKfs0/GHXI8ePapHH31UnTp1UsuWLdWzZ0/94x//qLVuh8OhefPm6a677lLbtm0VFxen//mf/zlpX48ePaqpU6eqc+fOatWqlTp27KhRo0bVes3SpUt1+eWXq3Xr1nK5XLr++uu1d+9eSdK///1vJSUlqV27doqIiNCgQYO0efPmWj+TBQsWaM2aNb4+JyUladq0afruu+/kcDh8M6T+7OPx9u7dq9tvv11t2rRRTEyMHnvsMR3/pUXr16/XwIED1bZtW7Vt21a9e/fWBx98cNL1AsGMGTqgmXvttdf04IMP6sUXX9TAgQO1c+dOTZ48WXv27NHixYt9r3vzzTc1ZswYrV27Vj/88INGjRqlLl266Mknn5QkzZ49Wy+++KJefvll9e/fX++8847+67/+y7f87bffrq+++kr/+7//q5ycHElSeHi47/mXXnpJjzzyiD777DPl5eXpzjvvVK9evTRmzJg66z58+LBuuOEG9e7dW5999pkOHTqk+++/X8XFxfXu688//6wRI0Zo+vTpGjFihI4cOaLPP/9cTqdTcXFxWrlypW655RZt3rxZcXFxatmypW/ZzZs3q23btnr77bcVGhqq1q1b17mNzZs3KywsTO+//75KS0t17733KjU1VatWrfLjp3HqPh0rPT1dCxcu1CuvvKLevXvrzTff1B//+EfFxMRoyJAhvtc9+eSTmj59up544gm9++67euCBB9S3b18lJyfXud6XXnpJy5Yt05IlS9S1a1cVFRVpw4YNvucXLVqk++67T3/5y1+0ePFiVVZW6uOPP/YF8AMHDmjSpEnq3bu3jh49qlmzZum6665TQUGBXC6XcnJyNHnyZO3evVvLli3z9fm///u/9dZbb2njxo0KDQ1t0D4eKzU1VVu3btU777yjmJgYPf3001q1apX69esnSaqqqtLNN9+s0aNH67XXXpMkffnll34fmgeCkgWgWbn77rutIUOG+B536dLFevnll2u9Zt26dZYkq6yszLIsyxo0aJB16aWX1nrNuHHjrP79+/sex8bGWo899lit19x+++1WaGio7/FTTz1ldenS5YSaunTpYv3ud7+rNTZ06FBr1KhR9e7H/PnzrTZt2vhqtCzL2rp1qyXJeuqpp2qtu+ZxXl6eJcn6/vvv61znp59+Wufzd999txUREWGVl5efMH5sL++++26rTZs21i+//OIb++CDDyxJ1jfffFPnMpZlWYsXL7aO/bisr0+DBg2y7rnnHsuyLOvgwYNWy5Ytrblz59Z6zbBhw6zk5GTfY0nWlClTar3mt7/9rfXoo4/W2QPLsqz777/fSk5Otrxeb53Px8XFWZMmTap3+eNVVVVZkZGR1pIlS3xjdfXh8ccft7p16+Z77O8+HvszLigosCRZH374oe/5X3/91YqNjfVtr6yszJJkffzxx37vAxDsOOQKNGN79uzRjz/+qKlTpyo8PNz33/XXXy9JKiws9L02Pj6+1rKdOnVSUVGRpOorFnft2qX+/fvXes2AAQP8ruVk66/L9u3bdfHFFysqKso3dskll5z0vLbLLrtMQ4cO1SWXXKLhw4frhRde0E8//eRXfRdffHG9M2XH6tmzZ60aBg4cKEnasWOHX9vxV2FhoSoqKnT11VfXGh80aJC2bdtWa6yhvR0zZoy2bt2q7t27a/z48XrrrbdUUVEhSSouLtZPP/2ka6+9tt7lv//+e911113q3r27zj33XJ177rnat2+ffvzxxybbxxrbt2+XpFrn4bVs2VJ9+/b1PY6KitLYsWM1dOhQXX/99ZoxY4a+/vrrBtUGBBsCHdCM1Vyh+cILLyg/P9/335YtW1RQUFDrhPVjDz9K1edm1Sxv/b/zk2rOAzsdJ1t/XSzLavD2QkNDtXr1aq1Zs0Z9+/bVW2+9pR49euhf//rXKZdt06ZNg7ZVn5CQkBPO5zp69Ohpr+/4HtTVl4b2Nj4+Xt9//72ee+45tWzZUg888IDi4+O1f//+erd7rJtuukn/+c9/NHfuXG3atEn5+fk677zzfKGwofzZx2Of88f8+fOVm5ura665RuvWrdMll1yizMzM06oPCAYEOqAZi4mJUVxcnL7++mt17979hP/qO0/seBEREYqNjdXGjRtrjW/atKnW45YtW57yQgd/9erVS9u3b9cvv/ziG9u2bZv27dt30uUcDof69eun9PR0ffLJJxo0aJAWLVrkq0/SGdW4Y8eOWsEnOztbUvUMnySdd9552rVrV61l8vLyaj32p0/du3dXq1attG7dulrjn3zyiXr16nXa9dcIDw/X8OHD9eKLL8rj8WjHjh1at26dzjvvPHXu3LneCwhKS0u1fft2Pfrooxo6dKh69uyp1q1bn/Tcxvqczj7WjNf0Xaq+4KbmfMRjXXLJJZo6dapWr16te+65R3/9618bXCMQLLgoAmjmMjIydM899ygyMlLDhg1TixYttGPHDq1evbpBMxYPPfSQHn/8cV100UXq16+f3n33XX344Ye1ZlIuuOAC/fzzz9q4caMuvPBChYWFnfaJ6H/4wx80bdo0/fGPf1RGRoYOHz6sBx54QOecc069y2RnZ+ujjz7Stddeq44dO6qgoEBffPGF7rnnHklSly5dFBISovfee0+33367WrVq1eBbkzgcDv2f//N/NH36dJWVlWnSpEm68cYbdeGFF0qSUlJS9Mwzz2jOnDm6/vrrtWbNmhOuzvWnT2FhYbr//vs1bdo0tW/fXvHx8XrjjTe0cuVK/fvf/25Qzcd79tlnFRsbq/j4eIWFhemf//ynQkND1aNHD0nS448/rgkTJigmJka33XabvF6vPv74Y40aNUrt2rVT+/btNX/+fHXr1k2lpaX605/+dNKfS31OZx+7d++um2++WZMmTVJmZqZiYmI0Y8YMlZeX+15TWFio+fPn63e/+53i4uK0a9cuffrpp0pISDi9hgFBgBk6oJm76667tGzZMr377rvq16+f+vbtqyeeeEKdOnVq0HrS0tI0efJkPfDAA+rTp482bdqkhx56qNYs37BhwzRixAjdeOONat++/Slvn3EyYWFheu+991RaWqp+/frpzjvv1IMPPqjzzjuv3mUiIiK0ceNG3XLLLbrwwguVmpqqO++8U9OmTZMk3xWRM2bMUMeOHXXLLbc0uK5+/frpyiuv1DXXXKOhQ4eqV69evhlAqTrQTZ8+XU8//bR69+6tNWvW6C9/+Uutdfjbp4yMDN17771KS0tTr169tGTJEi1ZsqTeqz/9de655+r555/XgAEDdOmll2rFihV666239Nvf/laSNHbsWL322mt68803FR8fr6uvvlqrV6+W0+lUSEiI3njjDX377be67LLLNHr0aKWlpaljx46nVcvp7OPChQsVHx+vm266SYMGDVKnTp00fPhw3/Nt2rRRQUGBRo0apR49euj3v/+9EhMTNWfOnNOqEQgGDsvfExoAnHVSU1O1ZcsW5ebm2l0KAOAMcMgVCBK7du3SihUrlJycrNDQUL3zzjv6+9//zqwHAJwFmKEDgkRRUZFuv/12ffHFFzpy5Ii6d++uKVOm6N5777W7NADAGSLQAQAAGI6LIgAAAAxHoAMAADAcgQ4AAMBwZ9VVrsff3b2xRUdHq6SkpEm3YQL6QA9q0Idq9IEe1KAP9KDGqfoQGxvbqNtjhg4AAMBwBDoAAADDEegAAAAMR6ADAAAwHIEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxnfKDzeDzKzMy0uwwAAADbOO0u4Ey53W653W67ywAAALCN8TN0AAAAwY5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOOO/+iuQioYnnjAWOn+VDZUAAAD8f8zQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhmuV96Lxer5YuXarDhw+ra9euSkpKsrskAACAZitggW7evHnKy8tTRESEZs6c6RvPz8/XokWL5PV6NWTIEA0bNkwej0dlZWUKDw+Xy+UKVIkAAABGCtgh16SkJKWnp9ca83q9WrBggdLT0zVr1ixt2LBBO3fu1K5du9SjRw/dfffd+vDDDwNVIgAAgJECFuh69uyp8PDwWmOFhYXq0KGDYmJi5HQ6lZiYqJycHLVr105t2rSpLjCE0/wAAABOxtZz6MrKymodUnW5XCooKNANN9yghQsX6quvvtLFF19c7/JZWVnKysqSJM2YMUPR0dFNWm9RHWNNvc3myOl0BuV+H4seVKMP1egDPahBH+hBjUD3wdZAZ1nWCWMOh0OtWrXShAkTTrl8SkqKUlJSfI9LSkoatT5/2LFNu0VHRwflfh+LHlSjD9XoAz2oQR/oQY1T9SE2NrZRt2fr8UyXy6XS0lLf49LSUkVFRdlYEQAAgHlsDXTdunXT7t27VVxcrMrKSmVnZ8vtdttZEgAAgHECdsh19uzZ2r59u8rLyzV+/HiNHDlSgwcPVmpqqjIyMuT1epWcnKy4uLgGrdfj8Sg3N1fjxo1rosoBAACat4AFurS0tDrHExISlJCQcNrrdbvdzOoBAICgxj1BAAAADEegAwAAMJzxgc7j8SgzM9PuMgAAAGxj633oGgPn0AEAgGBn/AwdAABAsCPQAQAAGI5ABwAAYDjjAx0XRQAAgGDHRREAAACGM36GDgAAINgR6AAAAAxHoAMAADCc8YGOiyIAAECw46IIAAAAwxk/QwcAABDsCHQAAACGI9ABAAAYjkAHAABgOOMDHVe5AgCAYMdVrgAAAIYzfoYOAAAg2BHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxnfKDjPnQAACDYcR86AAAAwxk/QwcAABDsCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDjjAx3fFAEAAIId3xQBAABgOONn6AAAAIIdgQ4AAMBwBDoAAADDEegAAAAMR6ADAAAwHIEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxnfKDzeDzKzMy0uwwAAADbOO0u4Ey53W653W67ywAAALCN8TN0AAAAwY5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4p90F1GXbtm1aunSpOnfurIEDB6pXr152lwQAANBsBSzQzZs3T3l5eYqIiNDMmTN94/n5+Vq0aJG8Xq+GDBmiYcOGyeFwqHXr1jp69KhcLlegSgQAADBSwA65JiUlKT09vdaY1+vVggULlJ6erlmzZmnDhg3auXOnLrroIqWnp+vOO+/UsmXLAlUiAACAkQIW6Hr27Knw8PBaY4WFherQoYNiYmLkdDqVmJionJwchYRUlxUeHq6jR48GqkQAAAAj2XoOXVlZWa1Dqi6XSwUFBfrss8+0ZcsWHTx4UNddd52NFQIAADR/tgY6y7JOGHM4HLriiit0xRVXnHL5rKwsZWVlSZJmzJih6OjoRq/xWEV1jDX1Npsjp9MZlPt9LHpQjT5Uow/0oAZ9oAc1At0HWwOdy+VSaWmp73FpaamioqL8Xj4lJUUpKSm+xyUlJY1anz/s2KbdoqOjg3K/j0UPqtGHavSBHtSgD/Sgxqn6EBsb26jbs/U+dN26ddPu3btVXFysyspKZWdny+1221kSAACAcQI2Qzd79mxt375d5eXlGj9+vEaOHKnBgwcrNTVVGRkZ8nq9Sk5OVlxcXIPW6/F4lJubq3HjxjVR5QAAAM1bwAJdWlpaneMJCQlKSEg47fW63W5m9QAAQFDjq78AAAAMR6ADAAAwnPGBzuPxKDMz0+4yAAAAbGPrbUsaA+fQAQCAYGf8DB0AAECwI9ABAAAYjkAHAABgOOMDHRdFAACAYMdFEQAAAIYzfoYOAAAg2BHoAAAADEegAwAAMJzxgY6LIgAAQLDjoggAAADDGT9DBwAAEOwIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhjA903LYEAAAEO25bAgAAYDjjZ+gAAACCHYEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHDGBzruQwcAAIId96EDAAAwnPEzdAAAAMGOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhjM+0PFNEQAAINjxTREAAACGM36GDgAAINgR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAMR6ADAAAwHIEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHDGBzqPx6PMzEy7ywAAALCN0+4CzpTb7Zbb7ba7DAAAANsYP0MHAAAQ7Ah0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIZrtoHuyJEjeuSRR5Sbm2t3KQAAAM2aM1AbmjdvnvLy8hQREaGZM2f6xvPz87Vo0SJ5vV4NGTJEw4YNkyStXLlSAwYMCFR5AAAAxgrYDF1SUpLS09NrjXm9Xi1YsEDp6emaNWuWNmzYoJ07d+qLL75Q586dFRkZGajyAAAAjBWwGbqePXuquLi41lhhYaE6dOigmJgYSVJiYqJycnJ05MgR/frrr9q5c6datmypPn36KCSk2R4dBgAAsFXAAl1dysrK5HK5fI9dLpcKCgp0zz33SJLWrl2rtm3b1hvmsrKylJWVJUmaMWOGoqOjm7TeojrGmnqbzZHT6QzK/T4WPahGH6rRB3pQgz7QgxqB7oOtgc6yrBPGHA6H799JSUknXT4lJUUpKSm+xyUlJY1Wm7/s2KbdoqOjg3K/j0UPqtGHavSBHtSgD/Sgxqn6EBsb26jbs/U4psvlUmlpqe9xaWmpoqKibKwIAADAPLYGum7dumn37t0qLi5WZWWlsrOz5Xa77SwJAADAOAE75Dp79mxt375d5eXlGj9+vEaOHKnBgwcrNTVVGRkZ8nq9Sk5OVlxcXIPW6/F4lJubq3HjxjVR5QAAAM1bwAJdWlpaneMJCQlKSEg47fW63W5m9QAAQFDjXiAAAACGI9ABAAAYzvhA5/F4lJmZaXcZAAAAtrH1PnSNgXPoAABAsDN+hg4AACDYEegAAAAMZ3yg4xw6AAAQ7DiHDgAAwHDGz9ABAAAEOwIdAACA4Qh0AAAAhiPQAQAAGM74QMdVrgAAINhxlSsAAIDhjJ+hAwAACHYEOgAAAMMR6AAAAAxHoAMAADCc8YGOq1wBAECw4ypXAAAAwxk/QwcAABDsCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhjM+0HEfOgAAEOy4Dx0AAIDhjJ+hAwAACHYEOgAAAMMR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAMR6ADAAAwnPGBjm+KAAAAwY5vigAAADCc8TN0AAAAwY5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhjM+0Hk8HmVmZtpdBgAAgG2cdhdwptxut9xut91lAAAA2Mb4GToAAIBg53egW7VqVZ3j//rXvxqtGAAAADSc34HurbfeatA4AAAAAuOU59B9+eWXkiSv1+v7d42ioiKdc845TVMZAAAA/HLKQPfyyy9LkioqKnz/liSHw6HIyEilpqY2XXUAAAA4pVMGurlz50qS5syZo8mTJzd5QQAAAGgYv29bcmyY83q9tZ4LCeFiWQAAALv4Hei+++47LViwQP/5z39UUVFR67mlS5c2emEAAADwj9+Bbu7cubr88ss1YcIEtWrVqilrAgAAQAP4HehKSkp0xx13yOFwNGU9AAAAaCC/T37r27evtmzZ0pS1AAAA4DT4PUN39OhRPffcc7rooosUGRlZ6zmufgUAALCP34Guc+fO6ty5c1PWAgAAgNPgd6AbMWJEU9YBAACA0+R3oDv+a7+OdckllzRKMQAAAGg4vwPdsV/7JUn79+9XZWWlXC6X5syZ0+iFAQAAwD8Nug/dsbxer9566y2dc845jV4UAAAA/Hfa39kVEhKiW2+9VStXrmzMegAAANBAfs/Q1eWLL75oku9x3blzp9577z2Vl5fr0ksv1bXXXtvo2wAAADhb+B3oJkyYUOtxRUWFKioqNHbsWL+WnzdvnvLy8hQREaGZM2f6xvPz87Vo0SJ5vV4NGTJEw4YNU+fOnXXffffJ6/UqMzPT3xIBAACCkt+BbsqUKbUet2rVSh07dlRYWJhfyyclJem6666rdS6e1+vVggUL9Nhjj8nlcunPf/6z3G63OnfuLI/Ho7ffflvXXXedvyUCAAAEJb8DXc+ePSVVh7B9+/YpIiKiQYdbe/bsqeLi4lpjhYWF6tChg2JiYiRJiYmJysnJUefOneV2u+V2u/X000/ryiuv9Hs7AAAAwcbvQHf48GEtWLBA2dnZqqqqUmhoqBITE5Wamur3LN3xysrK5HK5fI9dLpcKCgq0bds2ffbZZ6qsrFSfPn3qXT4rK0tZWVmSpBkzZig6Ovq06vBXUR1jTb3N5sjpdAblfh+LHlSjD9XoAz2oQR/oQY1A98HvQLdw4UIdOXJEzz33nNq3b689e/bo9ddf18KFC0/7u1wtyzphzOFwqFevXurVq9cpl09JSVFKSorvcUlJyWnVcSbs2KbdoqOjg3K/j0UPqtGHavSBHtSgD/Sgxqn6EBsb26jb8/uYaX5+vqZMmaLY2Fi1aNFCsbGxmjhxorZs2XLaG3e5XCotLfU9Li0tVVRU1GmvDwAAIBj5Hehatmyp/fv31xrbv3+/nM7Tv/NJt27dtHv3bhUXF6uyslLZ2dlyu92nvT4AAIBg5HcaGzx4sKZPn64bb7zRd8j13Xff1ZAhQ/xafvbs2dq+fbvKy8s1fvx4jRw5UoMHD1ZqaqoyMjLk9XqVnJysuLi4Bu2Ax+NRbm6uxo0b16DlAAAAzhZ+B7pbb71V7dq10/r161VWVqZ27drplltu0eDBg/1aPi0trc7xhIQEJSQk+FvGCWquhgUAAAhWfge6RYsWaeDAgZo2bZpv7Ouvv9Zrr72m0aNHN0VtAAAA8IPf59Bt2LBB3bp1qzXWtWtXrV+/vtGLAgAAgP/8DnQOh0Ner7fWmNfrrfPWI4Hk8Xj4ejAAABDU/A50F110kV5//XVfqPN6vXrjjTd00UUXNVlx/nC73VwQAQAAgprf59CNGTNGM2bM0Lhx43w3y4uKitIjjzzSlPUBAADgFPwOdC6XS88884wKCwtVWloql8ul7t27N+j7XAEAAND4GnRX4JCQEPXo0aOpagEAAMBpMH56jYsiAABAsDv97+1qJrixMAAACHbGz9ABAAAEOwIdAACA4Qh0AAAAhjM+0HFRBAAACHZcFAEAAGA442foAAAAgh2BDgAAwHAEOgAAAMMR6AAAAAxnfKDjKlcAABDsuMoVAADAcMbP0AEAAAQ7Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4YwPdNyHDgAABDvuQwcAAGA442foAAAAgh2BDgAAwHAEOgAAAMMR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAMZ3yg45siAABAsOObIgAAAAxnfKCzW9W9N9c5Hjp/VYArAQAAwcr4Q64AAADBjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhjA90Ho9HmZmZdpcBAABgG6fdBZxFi0etAAAXk0lEQVQpt9stt9ttdxkAAAC2MX6GDgAAINgR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAMR6ADAAAwHIEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAM57S7gPps3rxZeXl52r9/v4YOHarevXvbXRIAAECzFNBAN2/ePOXl5SkiIkIzZ870jefn52vRokXyer0aMmSIhg0bpn79+qlfv346cOCAFi9eTKADAACoR0APuSYlJSk9Pb3WmNfr1YIFC5Senq5Zs2Zpw4YN2rlzp+/55cuXa+jQoYEsEwAAwCgBDXQ9e/ZUeHh4rbHCwkJ16NBBMTExcjqdSkxMVE5OjizL0pIlSxQfH6+uXbsGskwAAACj2H4OXVlZmVwul++xy+VSQUGBVq9era1bt+rQoUP6+eefde21156wbFZWlrKysiRJM2bMUHR0dJPWWtSA1zZ1LXZyOp1n9f75gx5Uow/V6AM9qEEf6EGNQPfB9kBnWdYJYw6HQzfccINuuOGGky6bkpKilJQU3+OSkpJGr+90NadaGlt0dPRZvX/+oAfV6EM1+kAPatAHelDjVH2IjY1t1O3ZftsSl8ul0tJS3+PS0lJFRUXZWBEAAIBZbA903bp10+7du1VcXKzKykplZ2fL7XbbXRYAAIAxAnrIdfbs2dq+fbvKy8s1fvx4jRw5UoMHD1ZqaqoyMjLk9XqVnJysuLg4v9fp8XiUm5urcePGNWHlAAAAzVdAA11aWlqd4wkJCUpISDitdbrdbmb0AABAULP9kCsAAADODIEOAADAcMYHOo/Ho8zMTLvLAAAAsI3t96E7U831HLqqe28+YSx0/iobKgEAAGc742foAAAAgh2BDgAAwHAEOgAAAMMZH+i4KAIAAAQ7LooAAAAwnPEzdAAAAMGOQAcAAGA4Ah0AAIDhjA90XBQBAACCHRdFAAAAGM74GToAAIBgR6ADAAAwHIEOAADAcAQ6AAAAwxHoAAAADGd8oOO2JQAAINhx2xIAAADDGT9DBwAAEOwIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhjA903LYEAAAEO25bAgAAYDjjZ+gAAACCHYEOAADAcAQ6AAAAwxHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxnfKDjmyIAAECw45siAAAADGf8DB0AAECwI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYz/psiTFJ1780njIXOX2VDJQAA4GzCDB0AAIDhCHQAAACGI9ABAAAYzvhA5/F4lJmZaXcZAAAAtjH+ogi32y232213GQAAALYxfoYOAAAg2BHoAAAADEegAwAAMByBDgAAwHAEOgAAAMMR6AAAAAxHoAMAADAcgQ4AAMBwBDoAAADDEegAAAAMZ/xXf52Nqu69+YSx0PmrbKgEAACYgBk6AAAAwxHoAAAADEegAwAAMByBDgAAwHBcFGGzui6AAAAAaAhm6AAAAAxHoAMAADBcszzkWlRUpOXLl+vQoUN66KGH7C4HAACgWQvYDN28efM0duzYEwJafn6+HnjgAU2ZMkVvv/22JCkmJkYTJkwIVGkAAABGC1igS0pKUnp6eq0xr9erBQsWKD09XbNmzdKGDRu0c+fOQJUEAABwVghYoOvZs6fCw8NrjRUWFqpDhw6KiYmR0+lUYmKicnJyAlUSAADAWcHWc+jKysrkcrl8j10ulwoKClReXq5//vOf+uGHH7RixQoNHz68zuWzsrKUlZUlSZoxY4aio6ObtN6iJl37yTX1vjWE0+lsVvXYgR5Uow/V6AM9qEEf6EGNQPfB1kBnWdYJYw6HQ23bttV99913yuVTUlKUkpLie1xSUtKo9TUnzWnfoqOjm1U9dqAH1ehDNfpAD2rQB3pQ41R9iI2NbdTt2XrbEpfLpdLSUt/j0tJSRUVF2VgRAACAeWwNdN26ddPu3btVXFysyspKZWdny+12N2gdHo9HmZmZTVQhAABA8xewQ66zZ8/W9u3bVV5ervHjx2vkyJEaPHiwUlNTlZGRIa/Xq+TkZMXFxTVovW63u8EhEAAA4GwSsECXlpZW53hCQoISEhICVQYAAMBZh6/+AgAAMByBDgAAwHDGBzouigAAAMHO1vvQNQYuigAAAMHO+Bk6AACAYEegAwAAMJzxgY5z6AAAQLDjHDoAAADDGT9DBwAAEOwIdAAAAIYj0AEAABiOQAcAAGA44wMdV7kCAIBgx1WuAAAAhjN+hg4AACDYEegAAAAMR6ADAAAwHIEOAADAcMZfFOHxeJSbm6tx48bZXUrAVd178wljofNX2VAJAACwk/GBjqtcAQBAsOOQKwAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGM/62JcFyH7q67jnn7+v8vTedv9vQimz/XgcAAALC+EDHfegAAECw45ArAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4428sHCzfFHEmzuTbIwAAQPNnfKDjmyIAAECw45ArAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOKfdBZwpj8ej3NxcjRs3zu5SjFJ17812lwAAABqJ8YHO7XbL7XbbXQYAAIBtOOQKAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOEIdAAAAIYj0AEAABiOQAcAAGA4Ah0AAIDhCHQAAACGI9ABAAAYjkAHAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHQAAgOGcdhdQlyNHjujVV1+V0+lUr169dNVVV9ldEgAAQLMVsEA3b9485eXlKSIiQjNnzvSN5+fna9GiRfJ6vRoyZIiGDRumzZs3q3///nK73Zo1axaBDgAA4CQCdsg1KSlJ6enptca8Xq8WLFig9PR0zZo1Sxs2bNDOnTtVWlqq6Ojo6gJDOCoMAABwMgFLSz179lR4eHitscLCQnXo0EExMTFyOp1KTExUTk6OXC6XSktLJUmWZQWqRAAAACPZeg5dWVmZXC6X77HL5VJBQYGuv/56LVy4UHl5ebr88svrXT4rK0tZWVmSpBkzZvhm9ZpKUZOu3RxFwxP9el3MiuwzWt5f9W2nKbdb1IDt1ruOM+xjU6urvuNrcTqdTf57d7J66mJHv5xOp6ruvblZ1GKXQL4XmjP6YF4P/PmsOx2B7oOtga6u2TeHw6HWrVtr4sSJp1w+JSVFKSkpvsclJSWNWh/OTKB+Hnb93M/2/avL8bVER0c3q/oke/pV34d2c+tNU2qO7wU70IezoweNUf+p+hAbG3vG2ziWrSeoHXtoVZJKS0sVFRVlY0UAAADmsTXQdevWTbt371ZxcbEqKyuVnZ0tt9ttZ0kAAADGCdgh19mzZ2v79u0qLy/X+PHjNXLkSA0ePFipqanKyMiQ1+tVcnKy4uLiGrRej8ej3NxcjRs3rokqBwAAaN4CFujS0tLqHE9ISFBCQsJpr9ftdjOrBwAAgho3eQMAADAcgQ4AAMBwxgc6j8ejzMxMu8sAAACwja33oWsMnEMHAACCnfEzdAAAAMGOQAcAAGA4Ah0AAIDhjA90XBQBAACCHRdFAAAAGM74GToAAIBgR6ADAAAwHIEOAADAcMYHOi6KAAAAwY6LIgAAAAxn/AwdAABAsCPQAQAAGI5ABwAAYDgCHQAAgOGMD3Rc5QoAAIKdw7Isy+4iAAAAcPqMn6ELpEcffdTuEpoF+kAPatCHavSBHtSgD/SgRqD7QKADAAAwHIEOAADAcKFPPPHEE3YXYZKuXbvaXUKzQB/oQQ36UI0+0IMa9IEe1AhkH7goAgAAwHAccgUAADCc0+4CTJGfn69FixbJ6/VqyJAhGjZsmN0lNVhJSYnmzp2rX375RQ6HQykpKbrhhhu0bNkyffTRRzr33HMlSXfccYcSEhIkSStWrNCaNWsUEhKiMWPGKD4+XlL9/SguLtbs2bN14MABXXDBBZoyZYqcTqeOHj2qOXPm6LvvvlPbtm2Vlpam8847z5Y+TJo0Sa1bt1ZISIhCQ0M1Y8YMHThwQLNmzdKePXvUvn17PfjggwoPD5dlWVq0aJE+//xztWrVShMnTvRNoa9du1bLly+XJN16661KSkqSJH333XeaO3euKioq1KdPH40ZM0YOh6Pebdhh165dmjVrlu9xcXGxRo4cqYMHD57174V58+YpLy9PERERmjlzpiTZ+vM/2TYC2YPFixcrNzdXTqdTMTExmjhxotq0aaPi4mI9+OCDio2NlSRdeOGFuu+++xp9X+vrZ6D7YPfnYX3bCGQPZs2apV27dkmSDh06pLCwMD377LNn7Xuhvr+Nxn0uWDilqqoqa/LkydbPP/9sHT161Hr44Yetn376ye6yGqysrMz69ttvLcuyrEOHDln333+/9dNPP1lLly61Vq5cecLrf/rpJ+vhhx+2KioqrKKiImvy5MlWVVXVSfsxc+ZMa/369ZZlWVZmZqb1wQcfWJZlWe+//76VmZlpWZZlrV+/3nr++ecDsct1mjhxorVv375aY4sXL7ZWrFhhWZZlrVixwlq8eLFlWZaVm5trZWRkWF6v1/r666+tP//5z5ZlWVZ5ebk1adIkq7y8vNa/LcuyHn30Uevrr7+2vF6vlZGRYeXl5Z10G3arqqqyxo4daxUXFwfFe2Hbtm3Wt99+a02dOtU3ZufPv75tBLoH+fn5VmVlpa/WmvqKiopqve5YjbWvJ+tnU6qrD3b+DtS3jUD34Fh/+9vfrDfeeMOyrLP3vVDf30bTPhc45OqHwsJCdejQQTExMXI6nUpMTFROTo7dZTVYVFSUL+Gfc8456tSpk8rKyup9fU5OjhITE9WiRQudd9556tChgwoLC+vth2VZ2rZtm/r37y9JSkpK8vXJ4/H4/k+lf//++vLLL2U1o9M3c3JyNGjQIEnSoEGDatV99dVXy+FwqEePHjp48KD27t2r/Px8XXbZZQoPD1d4eLguu+wy5efna+/evTp8+LB69Oghh8Ohq6++2reu+rZht61bt6pDhw5q3759va85m94LPXv2PGFm1M6ff33bCHQPevfurdDQUElSjx49TvrZIKlR97W+fja1uvpQn0D8DtS3jaZ0sh5YlqWNGzdq4MCBJ12H6e+F+v42mva5QKDzQ1lZmVwul++xy+U65Yddc1dcXKzvv/9e3bt3lyR98MEHevjhhzVv3jwdOHBA0on73a5dO5WVldXbj/LycoWFhfn+KNS8/vh1hYaGKiwsTOXl5QHZ17pkZGTokUceUVZWliRp3759ioqKklT9y71//35J1XVHR0f7lqvZ14b25mTbsNuGDRtqfWAH23tBsvfnX9827LRmzZpah/qKi4v1pz/9SY8//rh27Ngh6eSfi43VT7vY9TvQ3PqwY8cORUREqGPHjr6xs/29cOzfRtM+FziHzg91zR44HA4bKmkcR44c0cyZMzV69GiFhYXp2muv1W233SZJWrp0qf7+979r4sSJ9c6anE4/mlMPn3rqKbVr10779u3T9OnTfeeD1KUhdTscjmY16+iPyspK5ebm6g9/+IMkBd174VQC8fNvbv1Yvny5QkNDddVVV0mq/iMzb948tW3bVt99952effZZzZw5s8n31a4e2Pk70Nw+P47/n72z/b1w/N/G+jTXzwVm6PzgcrlUWlrqe1xaWupL1KaprKzUzJkzddVVV+mKK66QJEVGRiokJEQhISEaMmSIvv32W0kn7ndZWZnatWtXbz/atm2rQ4cOqaqqqtbrj19XVVWVDh06ZNsFATU1RUREqG/fviosLFRERIRvOnvv3r2+E6JdLpdKSkp8y9bsa7t27U7oTVRUVJ29OXZ7dW3DTp9//rkuuOACRUZGSgq+90INO3/+9W3DDmvXrlVubq7uv/9+3x+PFi1aqG3btpKq76kVExOj3bt3N+q+1tdPO9j5O1DfNuxQVVWlzZs3KzEx0Td2Nr8X6vrbaNrnAoHOD926ddPu3btVXFysyspKZWdny+12211Wg1mWpVdeeUWdOnXSTTfd5Bs/9rj85s2bFRcXJ0lyu93Kzs7W0aNHVVxcrN27d6t79+719sPhcKhXr17atGmTpOo/DjV9uvzyy7V27VpJ0qZNm9SrVy9b/g/8yJEjOnz4sO/fX3zxhX7zm9/I7XZr3bp1kqR169apb9++kqp78Mknn8iyLH3zzTcKCwtTVFSU4uPjtWXLFh04cEAHDhzQli1bFB8fr6ioKJ1zzjn65ptvZFmWPvnkE18P6tuGnY7/P/Bgei8cy86ff33bCLT8/HytXLlSjzzyiFq1auUb379/v7xerySpqKhIu3fvVkxMTKPua339tIOdvwP1bcMOW7duVWxsbK1DhWfre6G+v42mfS5wY2E/5eXl6W9/+5u8Xq+Sk5N166232l1Sg3311Vf6y1/+ot/85je+P6B33HGHNmzYoB9++EEOh0Pt27fXfffd53vjLF++XB9//LFCQkI0evRo9enTR1L9/SgqKjrhMv0WLVqooqJCc+bM0ffff6/w8HClpaUpJiYm4D0oKirSc889J6n6/0CvvPJK3XrrrSovL9esWbNUUlKi6OhoTZ061Xfp+IIFC7Rlyxa1bNlSEydOVLdu3SRVn2e0YsUKSdWXpycnJ0uSvv32W82bN08VFRWKj49XamqqHA5Hvduwy6+//qoJEyZozpw5vsMLL7300ln/Xpg9e7a2b9+u8vJyRUREaOTIkerbt69tP/+TbSOQPVixYoUqKyt978maW1Js2rRJy5YtU2hoqEJCQjRixAjfH6PG3Nf6+hnoPmzbts3W34H6thHIHgwePFhz587VhRdeqGuvvdb32rP1vVDf38YLL7zQqM8FAh0AAIDhOOQKAABgOAIdAACA4Qh0AAAAhiPQAQAAGI5ABwAAYDgCHYCz0qRJk/TFF18EdJvFxcUaOXKk72ayABAoBDoAOE12hEYAqAuBDgAAwHBOuwsAgKbk9Xq1atUqffTRRzp48KAuueQS3XfffQoPD1dxcbEmT56siRMnaunSpaqoqNCNN97ou9N/RUWF/vrXvyo3N1eRkZFKSkrS6tWr9corr+ill15SSUmJnnnmGYWEhOi2227TgAEDJEmffvppnesDgKZCoANwVlu9erVycnL0xBNP6Nxzz9WiRYv06quvKi0tzfear776Si+88IJ27dql9PR09evXT507d9Ybb7yhPXv26KWXXtKvv/6qp59+2rfMlClT9NVXX2ncuHG67LLLJFWfQ3ey9QFAU+GQK4CzWlZWlkaNGiWXy6UWLVpoxIgR+uyzz2pduDBixAi1bNlS559/vrp06aIff/xRkrRx40YNHz5c4eHhcrlcuv766/3aZn3rA4CmwgwdgLPanj179Nxzz/m+dFuSQkJCtG/fPt/jyMhI379btWqlI0eOSJL27t2r6Oho33Mul8uvbda3PgBoKgQ6AGc1l8ulCRMm6KKLLjrhuZpDpPWJjIxUaWmp73BpaWlpk9QIAGeKQ64AzmrXXHONXn/9de3Zs0eStH//fuXk5Pi17IABA/T222/rwIEDKisr0/vvv1/r+cjIyFOGQgAIBGboAJzVbrjhBknS9OnTtXfvXkVERGjAgAHq27fvKZe97bbbNH/+fE2ePFlRUVG68sortXbtWt/zw4YN08KFC7VkyRLdeuut6t+/f1PtBgCclMOyLMvuIgDABB9++KE2bNigJ5980u5SAKAWDrkCQD327t2rr776Sl6vV7t27dI777yjfv362V0WAJyAQ64AUI/KykrNnz9fxcXFCgsL08CBAzV06FC7ywKAE3DIFQAAwHAccgUAADAcgQ4AAMBwBDoAAADDEegAAAAMR6ADAAAwHIEOAADAcP8XDZ32cdXspacAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f4534df7f98>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "%matplotlib inline  \n",
    "plt.style.use('ggplot')\n",
    "\n",
    "f = open(\"./metagenomics/assembly/scaffolds.fasta\", \"r\")\n",
    "lines = f.readlines()\n",
    "f.close()\n",
    "\n",
    "data = []\n",
    "regexp = re.compile(\">\")\n",
    "\n",
    "for line in lines:\n",
    "    if re.search(regexp, line):\n",
    "        line = line.strip().split('_')\n",
    "        data.append(float(line[3]))\n",
    "        \n",
    "fig = plt.figure(figsize=(10,10))\n",
    "plt.hist(data, bins=100, log=True);\n",
    "plt.title(\"length distribution scaffolds\");\n",
    "plt.xlabel(\"length\");\n",
    "plt.ylabel(\"count\");"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Did you expect this distribution?<br>\n",
    "Why would there be so many short scaffolds?<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2> Mapping reads to scaffolds with bwa (Burrows-Wheeler Aligner) and sorting with Samtools</h2>\n",
    "We now have the reads and the assembly so we can start by mapping (aligning) the original reads back to the scaffolds created with the assembly. Doing this will allow us to see the read depth on the scaffolds.\n",
    "\n",
    "The reads are paired-end, which means that for each DNA fragment, we have sequence data from both ends. The sequences are therefore stored in two separate files (one for the data from each end). We’ll use bwa’s default settings to align the reads back and then Samtools allows us to work with the bwa results.\n",
    "\n",
    "<b>Assignment:</b><br>\n",
    "Use bwa to backmap the reads against the scaffolds and save the mapping as a .bam file.\n",
    "(because we want to map all the reads back we helped you a bit with the commands).\n",
    "\n",
    "(1) To run bwa we first need to make an index file of the scaffolds. <br>\n",
    "(2) Then we can run bwa and immediatly pipe the output of bwa through samtools view which will output a bam file <br>\n",
    "(3) Finally the bam file has to be sorted using Samtools sort.\n",
    "\n",
    "\n",
    "some background: By default mapping algorithms like bwa spit out a .sam file. The sam format is widely used and accepted by many different downstream programmes. Basically it's just a big table with a standardised format. Although a sam file is human readable, it is also rather bulky. The filesize of a single SAM file will quickly exceed the disk space you have on this virtual machine. Therefore, we convert it to a BAM file, which is a Binary sAM file. There is no loss of information, it is just stored much more efficiently. Naturally, a binary file is not human readable. Always store your files as bam files, if you need to have a look in the bam file you can view them using samtools view or any of the many downstream programmes available online."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "bwa help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Program: bwa (alignment via Burrows-Wheeler transformation)\r\n",
      "Version: 0.7.17-r1188\r\n",
      "Contact: Heng Li <lh3@sanger.ac.uk>\r\n",
      "\r\n",
      "Usage:   bwa <command> [options]\r\n",
      "\r\n",
      "Command: index         index sequences in the FASTA format\r\n",
      "         mem           BWA-MEM algorithm\r\n",
      "         fastmap       identify super-maximal exact matches\r\n",
      "         pemerge       merge overlapping paired ends (EXPERIMENTAL)\r\n",
      "         aln           gapped/ungapped alignment\r\n",
      "         samse         generate alignment (single ended)\r\n",
      "         sampe         generate alignment (paired ended)\r\n",
      "         bwasw         BWA-SW for long queries\r\n",
      "\r\n",
      "         shm           manage indices in shared memory\r\n",
      "         fa2pac        convert FASTA to PAC format\r\n",
      "         pac2bwt       generate BWT from PAC\r\n",
      "         pac2bwtgen    alternative algorithm for generating BWT\r\n",
      "         bwtupdate     update .bwt to the new format\r\n",
      "         bwt2sa        generate SA from BWT and Occ\r\n",
      "\r\n",
      "Note: To use BWA, you need to first index the genome with `bwa index'.\r\n",
      "      There are three alignment algorithms in BWA: `mem', `bwasw', and\r\n",
      "      `aln/samse/sampe'. If you are not sure which to use, try `bwa mem'\r\n",
      "      first. Please `man ./bwa.1' for the manual.\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!bwa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "samtools help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.7 (using htslib 1.7-2)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!samtools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>run bwa index on default to make index files of your scaffolds (so no need for options)</b>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "bwa index help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!bwa index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run bwa index, finish the code: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!bwa index ...."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 4.47 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=1207810830, availableWord=96985660\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 99999998 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 199999998 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 299999998 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 399999998 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 499999998 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 597255246 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 684196110 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 761465326 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 830138430 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 891171246 characters processed.\n",
      "[BWTIncConstructFromPacked] 110 iterations done. 945413326 characters processed.\n",
      "[BWTIncConstructFromPacked] 120 iterations done. 993619838 characters processed.\n",
      "[BWTIncConstructFromPacked] 130 iterations done. 1036461870 characters processed.\n",
      "[BWTIncConstructFromPacked] 140 iterations done. 1074535966 characters processed.\n",
      "[BWTIncConstructFromPacked] 150 iterations done. 1108372270 characters processed.\n",
      "[BWTIncConstructFromPacked] 160 iterations done. 1138442030 characters processed.\n",
      "[BWTIncConstructFromPacked] 170 iterations done. 1165164094 characters processed.\n",
      "[BWTIncConstructFromPacked] 180 iterations done. 1188910686 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 189 iterations.\n",
      "[bwa_index] 324.60 seconds elapse.\n",
      "[bwa_index] Update BWT... 2.66 sec\n",
      "[bwa_index] Pack forward-only FASTA... 3.15 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 142.79 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index ./metagenomics/assembly/scaffolds.fasta\n",
      "[main] Real time: 480.825 sec; CPU: 477.678 sec\n"
     ]
    }
   ],
   "source": [
    "!bwa index ./metagenomics/assembly/scaffolds.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>check the output of bwa</b><br>\n",
    "there should now be multiple files in the assembly folder <br>\n",
    "think about what command you could use to look at the files in a directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "allscaffolds.fasta.gz  scaffolds.fasta.ann  scaffolds.fasta.pac\r\n",
      "scaffolds.fasta        scaffolds.fasta.bwt  scaffolds.fasta.sa\r\n",
      "scaffolds.fasta.amb    scaffolds.fasta.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls ./metagenomics/assembly/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>run bwa mem for backmapping</b>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For this part we will use a bash for loop to run bwa mem and samtools view on all the reads in metagenomics/reads and  create BAM output files in a newly made directory\n",
    "\n",
    "Then we use a bash for loop to run samtools sort on all the files created by samtools view\n",
    "\n",
    "Step by step instruction:<br>\n",
    "1. make a new directory for the samtools view results, give this a usefull name<br>\n",
    "2. finish the for loop and run both `bwa mem` and `samtools view` <br>\n",
    "3. make a new directory for the samtools sort results, give this a usefull name<br>\n",
    "4. finish the for loop and run samtools sort<br>\n",
    "\n",
    "Mapping each individual sample will take about 5 to 6 minutes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "bwa mem help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Usage: bwa mem [options] <idxbase> <in1.fq> [in2.fq]\r\n",
      "\r\n",
      "Algorithm options:\r\n",
      "\r\n",
      "       -t INT        number of threads [1]\r\n",
      "       -k INT        minimum seed length [19]\r\n",
      "       -w INT        band width for banded alignment [100]\r\n",
      "       -d INT        off-diagonal X-dropoff [100]\r\n",
      "       -r FLOAT      look for internal seeds inside a seed longer than {-k} * FLOAT [1.5]\r\n",
      "       -y INT        seed occurrence for the 3rd round seeding [20]\r\n",
      "       -c INT        skip seeds with more than INT occurrences [500]\r\n",
      "       -D FLOAT      drop chains shorter than FLOAT fraction of the longest overlapping chain [0.50]\r\n",
      "       -W INT        discard a chain if seeded bases shorter than INT [0]\r\n",
      "       -m INT        perform at most INT rounds of mate rescues for each read [50]\r\n",
      "       -S            skip mate rescue\r\n",
      "       -P            skip pairing; mate rescue performed unless -S also in use\r\n",
      "\r\n",
      "Scoring options:\r\n",
      "\r\n",
      "       -A INT        score for a sequence match, which scales options -TdBOELU unless overridden [1]\r\n",
      "       -B INT        penalty for a mismatch [4]\r\n",
      "       -O INT[,INT]  gap open penalties for deletions and insertions [6,6]\r\n",
      "       -E INT[,INT]  gap extension penalty; a gap of size k cost '{-O} + {-E}*k' [1,1]\r\n",
      "       -L INT[,INT]  penalty for 5'- and 3'-end clipping [5,5]\r\n",
      "       -U INT        penalty for an unpaired read pair [17]\r\n",
      "\r\n",
      "       -x STR        read type. Setting -x changes multiple parameters unless overridden [null]\r\n",
      "                     pacbio: -k17 -W40 -r10 -A1 -B1 -O1 -E1 -L0  (PacBio reads to ref)\r\n",
      "                     ont2d: -k14 -W20 -r10 -A1 -B1 -O1 -E1 -L0  (Oxford Nanopore 2D-reads to ref)\r\n",
      "                     intractg: -B9 -O16 -L5  (intra-species contigs to ref)\r\n",
      "\r\n",
      "Input/output options:\r\n",
      "\r\n",
      "       -p            smart pairing (ignoring in2.fq)\r\n",
      "       -R STR        read group header line such as '@RG\\tID:foo\\tSM:bar' [null]\r\n",
      "       -H STR/FILE   insert STR to header if it starts with @; or insert lines in FILE [null]\r\n",
      "       -o FILE       sam file to output results to [stdout]\r\n",
      "       -j            treat ALT contigs as part of the primary assembly (i.e. ignore <idxbase>.alt file)\r\n",
      "       -5            for split alignment, take the alignment with the smallest coordinate as primary\r\n",
      "       -q            don't modify mapQ of supplementary alignments\r\n",
      "       -K INT        process INT input bases in each batch regardless of nThreads (for reproducibility) []\r\n",
      "\r\n",
      "       -v INT        verbosity level: 1=error, 2=warning, 3=message, 4+=debugging [3]\r\n",
      "       -T INT        minimum score to output [30]\r\n",
      "       -h INT[,INT]  if there are <INT hits with score >80% of the max score, output all in XA [5,200]\r\n",
      "       -a            output all alignments for SE or unpaired PE\r\n",
      "       -C            append FASTA/FASTQ comment to SAM output\r\n",
      "       -V            output the reference FASTA header in the XR tag\r\n",
      "       -Y            use soft clipping for supplementary alignments\r\n",
      "       -M            mark shorter split hits as secondary\r\n",
      "\r\n",
      "       -I FLOAT[,FLOAT[,INT[,INT]]]\r\n",
      "                     specify the mean, standard deviation (10% of the mean if absent), max\r\n",
      "                     (4 sigma from the mean if absent) and min of the insert size distribution.\r\n",
      "                     FR orientation only. [inferred]\r\n",
      "\r\n",
      "Note: Please read the man page for detailed description of the command line and options.\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!bwa mem"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "samtools view help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools view"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm ./metagenomics/mapping/ -rf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir ./metagenomics/mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L1.R1.fastq.gz  L2.R2.fastq.gz  P1.R1.fastq.gz  P2.R2.fastq.gz\r\n",
      "L1.R2.fastq.gz  L3.R1.fastq.gz  P1.R2.fastq.gz  P3.R1.fastq.gz\r\n",
      "L2.R1.fastq.gz  L3.R2.fastq.gz  P2.R1.fastq.gz  P3.R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls ./metagenomics/reads/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L1\n",
      "L2\n",
      "L3\n",
      "P1\n",
      "P2\n",
      "P3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 68246 sequences (10000095 bp)...\n",
      "[M::process] read 68222 sequences (10000032 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 20948, 2, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 418, 514)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (49, 824)\n",
      "[M::mem_pestat] mean and std.dev: (434.33, 134.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 979)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68246 reads in 10.325 CPU sec, 10.192 real sec\n",
      "[M::process] read 68066 sequences (10000164 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 20931, 2, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (189, 314, 526)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1200)\n",
      "[M::mem_pestat] mean and std.dev: (283.80, 155.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1537)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 420, 524)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (29, 854)\n",
      "[M::mem_pestat] mean and std.dev: (438.59, 138.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1019)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68222 reads in 10.303 CPU sec, 10.115 real sec\n",
      "[M::process] read 68152 sequences (10000153 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 20894, 5, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (177, 305, 458)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1020)\n",
      "[M::mem_pestat] mean and std.dev: (291.92, 174.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1301)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (360, 420, 527)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 861)\n",
      "[M::mem_pestat] mean and std.dev: (438.79, 139.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1028)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68066 reads in 10.069 CPU sec, 9.878 real sec\n",
      "[M::process] read 68162 sequences (10000063 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 20663, 2, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (164, 352, 588)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1436)\n",
      "[M::mem_pestat] mean and std.dev: (392.55, 299.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1860)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 419, 521)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (35, 845)\n",
      "[M::mem_pestat] mean and std.dev: (436.35, 137.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1007)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68152 reads in 10.102 CPU sec, 9.925 real sec\n",
      "[M::process] read 68118 sequences (10000153 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 20866, 4, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (163, 366, 542)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1300)\n",
      "[M::mem_pestat] mean and std.dev: (317.22, 175.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1679)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (358, 417, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (40, 835)\n",
      "[M::mem_pestat] mean and std.dev: (434.39, 136.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 994)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68162 reads in 10.461 CPU sec, 10.268 real sec\n",
      "[M::process] read 68214 sequences (10000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 20755, 10, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 419, 518)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (41, 836)\n",
      "[M::mem_pestat] mean and std.dev: (435.38, 136.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 995)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (82, 542, 1876)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5464)\n",
      "[M::mem_pestat] mean and std.dev: (1038.80, 1281.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7258)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68118 reads in 10.466 CPU sec, 10.285 real sec\n",
      "[M::process] read 68186 sequences (10000253 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 20738, 7, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (126, 311, 460)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1128)\n",
      "[M::mem_pestat] mean and std.dev: (283.14, 182.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1462)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (360, 417, 512)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (56, 816)\n",
      "[M::mem_pestat] mean and std.dev: (433.19, 132.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 968)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68214 reads in 10.036 CPU sec, 9.844 real sec\n",
      "[M::process] read 68170 sequences (10000109 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 20738, 4, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 418, 515)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (47, 827)\n",
      "[M::mem_pestat] mean and std.dev: (433.08, 135.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 983)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68186 reads in 10.371 CPU sec, 10.203 real sec\n",
      "[M::process] read 68206 sequences (10000165 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 20698, 6, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (245, 396, 453)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 869)\n",
      "[M::mem_pestat] mean and std.dev: (352.27, 156.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1077)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 416, 510)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (51, 816)\n",
      "[M::mem_pestat] mean and std.dev: (431.43, 132.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 969)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68170 reads in 10.533 CPU sec, 10.333 real sec\n",
      "[M::process] read 68454 sequences (10000079 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 20865, 6, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (238, 291, 481)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 967)\n",
      "[M::mem_pestat] mean and std.dev: (297.80, 161.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1210)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 416, 509)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (53, 813)\n",
      "[M::mem_pestat] mean and std.dev: (431.15, 132.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 965)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68206 reads in 10.335 CPU sec, 10.136 real sec\n",
      "[M::process] read 68620 sequences (10000210 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 20790, 1, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (217, 385, 489)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1033)\n",
      "[M::mem_pestat] mean and std.dev: (296.78, 135.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1305)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (356, 416, 503)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (62, 797)\n",
      "[M::mem_pestat] mean and std.dev: (428.53, 129.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 946)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68454 reads in 10.419 CPU sec, 10.223 real sec\n",
      "[M::process] read 68668 sequences (10000147 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 20943, 1, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 417, 508)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (61, 806)\n",
      "[M::mem_pestat] mean and std.dev: (431.18, 129.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 955)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68620 reads in 10.649 CPU sec, 10.467 real sec\n",
      "[M::process] read 68878 sequences (10000236 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 20964, 5, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (356, 415, 500)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (68, 788)\n",
      "[M::mem_pestat] mean and std.dev: (427.00, 128.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 942)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68668 reads in 10.443 CPU sec, 10.255 real sec\n",
      "[M::process] read 68798 sequences (10000090 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 21082, 3, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 416, 499)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (73, 783)\n",
      "[M::mem_pestat] mean and std.dev: (427.36, 127.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 935)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68878 reads in 10.167 CPU sec, 9.972 real sec\n",
      "[M::process] read 68352 sequences (10000040 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 21050, 8, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 417, 503)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (65, 795)\n",
      "[M::mem_pestat] mean and std.dev: (428.88, 129.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 946)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68798 reads in 10.518 CPU sec, 10.333 real sec\n",
      "[M::process] read 68312 sequences (10000014 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 21101, 4, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 417, 510)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (57, 812)\n",
      "[M::mem_pestat] mean and std.dev: (431.75, 131.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 963)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68352 reads in 10.130 CPU sec, 9.949 real sec\n",
      "[M::process] read 68314 sequences (10000146 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 20965, 4, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 416, 508)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (55, 810)\n",
      "[M::mem_pestat] mean and std.dev: (430.72, 131.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 961)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68312 reads in 10.024 CPU sec, 9.849 real sec\n",
      "[M::process] read 68274 sequences (10000245 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 21060, 4, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (165, 323, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1011)\n",
      "[M::mem_pestat] mean and std.dev: (275.91, 142.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1293)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (358, 417, 509)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (56, 811)\n",
      "[M::mem_pestat] mean and std.dev: (430.54, 131.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 962)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68314 reads in 10.398 CPU sec, 10.216 real sec\n",
      "[M::process] read 68268 sequences (10000276 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 20944, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 417, 504)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (69, 794)\n",
      "[M::mem_pestat] mean and std.dev: (429.09, 129.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 947)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68274 reads in 10.468 CPU sec, 10.282 real sec\n",
      "[M::process] read 68240 sequences (10000206 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 20751, 7, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 391, 488)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (95, 750)\n",
      "[M::mem_pestat] mean and std.dev: (406.33, 97.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 881)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (358, 418, 509)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (56, 811)\n",
      "[M::mem_pestat] mean and std.dev: (431.62, 131.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 962)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68268 reads in 10.512 CPU sec, 10.317 real sec\n",
      "[M::process] read 68262 sequences (10000046 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 20930, 5, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 199, 445)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1193)\n",
      "[M::mem_pestat] mean and std.dev: (248.08, 177.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1567)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (358, 417, 507)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (60, 805)\n",
      "[M::mem_pestat] mean and std.dev: (429.16, 130.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 954)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68240 reads in 10.055 CPU sec, 9.854 real sec\n",
      "[M::process] read 68292 sequences (10000044 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 20940, 5, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (356, 416, 501)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (66, 791)\n",
      "[M::mem_pestat] mean and std.dev: (427.09, 128.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 941)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68262 reads in 10.594 CPU sec, 10.403 real sec\n",
      "[M::process] read 68324 sequences (10000143 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 20945, 4, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (358, 416, 505)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (64, 799)\n",
      "[M::mem_pestat] mean and std.dev: (428.87, 130.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 951)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68292 reads in 10.245 CPU sec, 10.034 real sec\n",
      "[M::process] read 68520 sequences (10000292 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 20905, 2, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (356, 416, 503)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (62, 797)\n",
      "[M::mem_pestat] mean and std.dev: (428.40, 130.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 952)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68324 reads in 10.097 CPU sec, 9.918 real sec\n",
      "[M::process] read 68522 sequences (10000201 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 20915, 7, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (173, 436, 695)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1739)\n",
      "[M::mem_pestat] mean and std.dev: (433.89, 276.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2261)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (355, 414, 498)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (69, 784)\n",
      "[M::mem_pestat] mean and std.dev: (426.25, 128.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 940)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68520 reads in 10.458 CPU sec, 10.251 real sec\n",
      "[M::process] read 68682 sequences (10000143 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 21126, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (356, 415, 497)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (74, 779)\n",
      "[M::mem_pestat] mean and std.dev: (426.12, 125.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 930)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68522 reads in 10.328 CPU sec, 10.128 real sec\n",
      "[M::process] read 68792 sequences (10000265 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 21028, 2, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (177, 296, 381)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 789)\n",
      "[M::mem_pestat] mean and std.dev: (264.50, 119.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 993)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (354, 414, 494)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (74, 774)\n",
      "[M::mem_pestat] mean and std.dev: (423.97, 126.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 931)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68682 reads in 10.488 CPU sec, 10.309 real sec\n",
      "[M::process] read 69002 sequences (10000193 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 21240, 3, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (183, 226, 412)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 870)\n",
      "[M::mem_pestat] mean and std.dev: (259.60, 139.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1099)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (355, 415, 495)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (75, 775)\n",
      "[M::mem_pestat] mean and std.dev: (424.19, 125.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 926)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68792 reads in 9.774 CPU sec, 9.589 real sec\n",
      "[M::process] read 16684 sequences (2435559 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 21113, 2, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 416, 498)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (75, 780)\n",
      "[M::mem_pestat] mean and std.dev: (426.61, 126.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 932)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69002 reads in 9.946 CPU sec, 9.853 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 4976, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (359, 416, 504)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (69, 794)\n",
      "[M::mem_pestat] mean and std.dev: (430.24, 128.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 943)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 16684 reads in 2.529 CPU sec, 2.465 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 1 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/L1.R1.fastq.gz ./metagenomics/reads/L1.R2.fastq.gz\n",
      "[main] Real time: 296.358 sec; CPU: 301.487 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 68522 sequences (10000078 bp)...\n",
      "[M::process] read 68460 sequences (10000108 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (52, 19979, 4, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 205, 408)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 980)\n",
      "[M::mem_pestat] mean and std.dev: (251.69, 153.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1266)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 375, 523)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 927)\n",
      "[M::mem_pestat] mean and std.dev: (417.45, 151.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1129)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68522 reads in 10.723 CPU sec, 10.593 real sec\n",
      "[M::process] read 68384 sequences (10000109 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (45, 19776, 0, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (135, 251, 323)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 699)\n",
      "[M::mem_pestat] mean and std.dev: (235.64, 125.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 887)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (322, 375, 524)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 928)\n",
      "[M::mem_pestat] mean and std.dev: (418.04, 152.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1130)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68460 reads in 10.683 CPU sec, 10.496 real sec\n",
      "[M::process] read 68396 sequences (10000128 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (48, 19767, 9, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (163, 240, 389)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 841)\n",
      "[M::mem_pestat] mean and std.dev: (252.07, 144.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1067)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 375, 523)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 929)\n",
      "[M::mem_pestat] mean and std.dev: (416.45, 150.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1132)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68384 reads in 10.930 CPU sec, 10.700 real sec\n",
      "[M::process] read 68502 sequences (10000082 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (51, 19629, 5, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (125, 175, 348)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 794)\n",
      "[M::mem_pestat] mean and std.dev: (225.58, 126.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1017)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (322, 375, 525)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 931)\n",
      "[M::mem_pestat] mean and std.dev: (417.29, 151.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1134)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68396 reads in 10.887 CPU sec, 10.697 real sec\n",
      "[M::process] read 68554 sequences (10000084 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (55, 19567, 11, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (155, 270, 412)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 926)\n",
      "[M::mem_pestat] mean and std.dev: (277.51, 146.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1183)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (322, 375, 526)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 934)\n",
      "[M::mem_pestat] mean and std.dev: (417.55, 151.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1138)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (55, 319, 1739)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5107)\n",
      "[M::mem_pestat] mean and std.dev: (1051.55, 1477.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6961)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68502 reads in 10.833 CPU sec, 10.635 real sec\n",
      "[M::process] read 68600 sequences (10000078 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (44, 19570, 6, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 271, 394)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 774)\n",
      "[M::mem_pestat] mean and std.dev: (278.27, 122.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 964)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (318, 373, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 915)\n",
      "[M::mem_pestat] mean and std.dev: (411.15, 149.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1114)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68554 reads in 10.869 CPU sec, 10.699 real sec\n",
      "[M::process] read 68832 sequences (10000195 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (51, 19602, 6, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 236, 407)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 997)\n",
      "[M::mem_pestat] mean and std.dev: (255.38, 156.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1292)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 375, 521)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 923)\n",
      "[M::mem_pestat] mean and std.dev: (414.19, 148.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1124)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68600 reads in 11.014 CPU sec, 10.824 real sec\n",
      "[M::process] read 69010 sequences (10000014 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (37, 19775, 5, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (91, 200, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 706)\n",
      "[M::mem_pestat] mean and std.dev: (219.46, 137.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 911)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 375, 518)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 912)\n",
      "[M::mem_pestat] mean and std.dev: (414.04, 146.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1109)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68832 reads in 11.063 CPU sec, 10.888 real sec\n",
      "[M::process] read 68924 sequences (10000187 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (35, 19982, 3, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (140, 197, 301)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 623)\n",
      "[M::mem_pestat] mean and std.dev: (222.89, 125.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 784)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 373, 510)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 892)\n",
      "[M::mem_pestat] mean and std.dev: (408.19, 144.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1083)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 69010 reads in 10.963 CPU sec, 10.769 real sec\n",
      "[M::process] read 68640 sequences (10000235 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (49, 20007, 2, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (149, 262, 396)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 890)\n",
      "[M::mem_pestat] mean and std.dev: (260.98, 146.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1137)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (318, 373, 511)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 897)\n",
      "[M::mem_pestat] mean and std.dev: (408.78, 144.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1090)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68924 reads in 10.648 CPU sec, 10.477 real sec\n",
      "[M::process] read 68584 sequences (10000293 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (52, 19888, 6, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (121, 217, 349)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 805)\n",
      "[M::mem_pestat] mean and std.dev: (238.08, 135.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1033)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 374, 518)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 916)\n",
      "[M::mem_pestat] mean and std.dev: (412.03, 148.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1115)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68640 reads in 10.740 CPU sec, 10.562 real sec\n",
      "[M::process] read 68570 sequences (10000135 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (45, 19672, 9, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (113, 204, 302)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 680)\n",
      "[M::mem_pestat] mean and std.dev: (218.56, 127.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 869)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 374, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 913)\n",
      "[M::mem_pestat] mean and std.dev: (412.03, 147.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1111)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68584 reads in 10.603 CPU sec, 10.428 real sec\n",
      "[M::process] read 68530 sequences (10000109 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (47, 19760, 4, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 228, 340)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 800)\n",
      "[M::mem_pestat] mean and std.dev: (227.38, 142.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1030)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 374, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 909)\n",
      "[M::mem_pestat] mean and std.dev: (412.25, 147.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1105)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68570 reads in 10.674 CPU sec, 10.490 real sec\n",
      "[M::process] read 68570 sequences (10000053 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (46, 19594, 5, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (83, 215, 315)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 779)\n",
      "[M::mem_pestat] mean and std.dev: (218.72, 143.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1011)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (317, 373, 513)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 905)\n",
      "[M::mem_pestat] mean and std.dev: (408.49, 145.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1101)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68530 reads in 10.913 CPU sec, 10.718 real sec\n",
      "[M::process] read 68640 sequences (10000245 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (47, 19663, 5, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 190, 354)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 828)\n",
      "[M::mem_pestat] mean and std.dev: (231.55, 155.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1065)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 373, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 913)\n",
      "[M::mem_pestat] mean and std.dev: (411.66, 146.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1111)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68570 reads in 10.949 CPU sec, 10.752 real sec\n",
      "[M::process] read 68852 sequences (10000019 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (37, 19788, 4, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (167, 254, 385)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 821)\n",
      "[M::mem_pestat] mean and std.dev: (263.00, 121.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1039)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 374, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 911)\n",
      "[M::mem_pestat] mean and std.dev: (411.78, 146.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1108)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68640 reads in 10.855 CPU sec, 10.676 real sec\n",
      "[M::process] read 68878 sequences (10000224 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (54, 19704, 12, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 196, 355)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 825)\n",
      "[M::mem_pestat] mean and std.dev: (242.31, 148.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1060)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (318, 372, 511)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 897)\n",
      "[M::mem_pestat] mean and std.dev: (407.29, 144.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1090)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (61, 437, 758)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2152)\n",
      "[M::mem_pestat] mean and std.dev: (394.67, 319.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2849)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68852 reads in 11.097 CPU sec, 10.881 real sec\n",
      "[M::process] read 69220 sequences (10000162 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (35, 19833, 8, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (93, 210, 305)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 729)\n",
      "[M::mem_pestat] mean and std.dev: (236.69, 171.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 941)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (317, 371, 511)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 899)\n",
      "[M::mem_pestat] mean and std.dev: (406.48, 144.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1093)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68878 reads in 11.008 CPU sec, 10.828 real sec\n",
      "[M::process] read 68756 sequences (10000213 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (36, 19748, 4, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (164, 241, 337)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 683)\n",
      "[M::mem_pestat] mean and std.dev: (245.23, 125.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 856)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (317, 372, 511)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 899)\n",
      "[M::mem_pestat] mean and std.dev: (407.11, 143.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1093)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 69220 reads in 10.900 CPU sec, 10.713 real sec\n",
      "[M::process] read 68558 sequences (10000247 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (43, 19728, 2, 4)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (133, 248, 376)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 862)\n",
      "[M::mem_pestat] mean and std.dev: (252.31, 148.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1105)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (318, 373, 515)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 909)\n",
      "[M::mem_pestat] mean and std.dev: (411.10, 146.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1106)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68756 reads in 10.956 CPU sec, 10.773 real sec\n",
      "[M::process] read 68614 sequences (10000278 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (36, 19626, 8, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 230, 372)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 892)\n",
      "[M::mem_pestat] mean and std.dev: (244.39, 151.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1152)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 374, 518)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 912)\n",
      "[M::mem_pestat] mean and std.dev: (412.78, 146.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1109)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68558 reads in 10.899 CPU sec, 10.701 real sec\n",
      "[M::process] read 68564 sequences (10000299 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (38, 19595, 4, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (84, 266, 373)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 951)\n",
      "[M::mem_pestat] mean and std.dev: (246.68, 158.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1240)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 373, 518)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 916)\n",
      "[M::mem_pestat] mean and std.dev: (413.59, 149.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1115)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68614 reads in 11.133 CPU sec, 10.904 real sec\n",
      "[M::process] read 68618 sequences (10000081 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (49, 19625, 9, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (105, 205, 373)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 909)\n",
      "[M::mem_pestat] mean and std.dev: (236.53, 148.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1177)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 375, 522)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 926)\n",
      "[M::mem_pestat] mean and std.dev: (414.67, 149.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1128)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68564 reads in 10.722 CPU sec, 10.538 real sec\n",
      "[M::process] read 68560 sequences (10000124 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (50, 19739, 8, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (93, 250, 363)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 903)\n",
      "[M::mem_pestat] mean and std.dev: (237.18, 152.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1173)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 375, 522)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 924)\n",
      "[M::mem_pestat] mean and std.dev: (415.29, 150.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1125)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68618 reads in 10.735 CPU sec, 10.539 real sec\n",
      "[M::process] read 68592 sequences (10000072 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (50, 19851, 3, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (103, 235, 356)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 862)\n",
      "[M::mem_pestat] mean and std.dev: (241.80, 145.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1115)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 375, 519)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 917)\n",
      "[M::mem_pestat] mean and std.dev: (414.72, 149.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1116)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68560 reads in 10.669 CPU sec, 10.492 real sec\n",
      "[M::process] read 68670 sequences (10000267 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (54, 19798, 3, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 201, 318)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 718)\n",
      "[M::mem_pestat] mean and std.dev: (221.17, 139.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 918)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 375, 520)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 918)\n",
      "[M::mem_pestat] mean and std.dev: (415.30, 149.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1117)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68592 reads in 10.704 CPU sec, 10.531 real sec\n",
      "[M::process] read 68738 sequences (10000073 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (52, 19982, 7, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 209, 325)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 739)\n",
      "[M::mem_pestat] mean and std.dev: (218.85, 115.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 946)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (321, 375, 521)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 921)\n",
      "[M::mem_pestat] mean and std.dev: (415.29, 149.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1121)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68670 reads in 10.675 CPU sec, 10.456 real sec\n",
      "[M::process] read 68910 sequences (10000220 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (40, 20126, 3, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (95, 222, 352)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 866)\n",
      "[M::mem_pestat] mean and std.dev: (214.18, 128.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1123)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 375, 521)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 923)\n",
      "[M::mem_pestat] mean and std.dev: (414.82, 148.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1124)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68738 reads in 10.627 CPU sec, 10.426 real sec\n",
      "[M::process] read 8752 sequences (1269380 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (47, 19919, 6, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (145, 235, 358)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 784)\n",
      "[M::mem_pestat] mean and std.dev: (257.30, 145.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 997)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 373, 514)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 904)\n",
      "[M::mem_pestat] mean and std.dev: (410.55, 146.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1099)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68910 reads in 10.626 CPU sec, 10.552 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 2474, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (313, 370, 514)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 916)\n",
      "[M::mem_pestat] mean and std.dev: (406.64, 145.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1117)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 8752 reads in 1.490 CPU sec, 1.427 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 1 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/L2.R1.fastq.gz ./metagenomics/reads/L2.R2.fastq.gz\n",
      "[main] Real time: 310.547 sec; CPU: 315.827 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 68204 sequences (10000102 bp)...\n",
      "[M::process] read 68080 sequences (10000029 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 21361, 5, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (353, 412, 448)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (163, 638)\n",
      "[M::mem_pestat] mean and std.dev: (404.67, 92.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (33, 776)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68204 reads in 9.907 CPU sec, 9.765 real sec\n",
      "[M::process] read 68116 sequences (10000183 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 20984, 5, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (138, 249, 459)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1101)\n",
      "[M::mem_pestat] mean and std.dev: (246.27, 149.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1422)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (354, 412, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (168, 633)\n",
      "[M::mem_pestat] mean and std.dev: (403.71, 89.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (44, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68080 reads in 10.036 CPU sec, 9.859 real sec\n",
      "[M::process] read 68098 sequences (10000264 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 21149, 5, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (274, 341, 483)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 901)\n",
      "[M::mem_pestat] mean and std.dev: (342.18, 165.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1110)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (353, 411, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (165, 635)\n",
      "[M::mem_pestat] mean and std.dev: (402.83, 90.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (42, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68116 reads in 9.946 CPU sec, 9.755 real sec\n",
      "[M::process] read 68114 sequences (10000227 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 21082, 4, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (181, 229, 438)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 952)\n",
      "[M::mem_pestat] mean and std.dev: (269.94, 160.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1209)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (354, 410, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (168, 633)\n",
      "[M::mem_pestat] mean and std.dev: (403.46, 90.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (43, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68098 reads in 10.125 CPU sec, 9.933 real sec\n",
      "[M::process] read 68068 sequences (10000262 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 20913, 4, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (285, 429, 497)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 921)\n",
      "[M::mem_pestat] mean and std.dev: (356.62, 157.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1133)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (355, 412, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (171, 631)\n",
      "[M::mem_pestat] mean and std.dev: (402.81, 89.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (47, 759)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68114 reads in 10.117 CPU sec, 9.925 real sec\n",
      "[M::process] read 68130 sequences (10000284 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 20979, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 411, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (162, 637)\n",
      "[M::mem_pestat] mean and std.dev: (402.53, 91.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (36, 769)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68068 reads in 10.291 CPU sec, 10.119 real sec\n",
      "[M::process] read 68128 sequences (10000002 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 21130, 5, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (350, 410, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (158, 638)\n",
      "[M::mem_pestat] mean and std.dev: (401.67, 93.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (29, 775)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68130 reads in 10.239 CPU sec, 10.045 real sec\n",
      "[M::process] read 68140 sequences (10000135 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 20686, 3, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (354, 412, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (168, 633)\n",
      "[M::mem_pestat] mean and std.dev: (404.36, 89.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (47, 762)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68128 reads in 10.231 CPU sec, 10.042 real sec\n",
      "[M::process] read 68162 sequences (10000257 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 20863, 7, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 403, 526)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (58, 838)\n",
      "[M::mem_pestat] mean and std.dev: (417.20, 138.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 994)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (353, 410, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (167, 632)\n",
      "[M::mem_pestat] mean and std.dev: (402.93, 90.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (42, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68140 reads in 10.080 CPU sec, 9.909 real sec\n",
      "[M::process] read 68336 sequences (10000199 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 21147, 4, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 410, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (164, 634)\n",
      "[M::mem_pestat] mean and std.dev: (402.43, 90.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (39, 765)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68162 reads in 10.194 CPU sec, 9.994 real sec\n",
      "[M::process] read 68430 sequences (10000264 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 20924, 4, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 409, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (164, 634)\n",
      "[M::mem_pestat] mean and std.dev: (401.71, 90.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (40, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68336 reads in 10.365 CPU sec, 10.129 real sec\n",
      "[M::process] read 68546 sequences (10000060 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 20980, 6, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (360, 430, 1452)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3636)\n",
      "[M::mem_pestat] mean and std.dev: (670.17, 519.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4728)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 413, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 627)\n",
      "[M::mem_pestat] mean and std.dev: (405.43, 87.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (57, 753)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68430 reads in 10.261 CPU sec, 10.060 real sec\n",
      "[M::process] read 68696 sequences (10000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 21162, 2, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 410, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (164, 634)\n",
      "[M::mem_pestat] mean and std.dev: (402.77, 90.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (42, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68546 reads in 10.296 CPU sec, 10.092 real sec\n",
      "[M::process] read 68588 sequences (10000083 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 21169, 1, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (351, 409, 445)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (163, 633)\n",
      "[M::mem_pestat] mean and std.dev: (401.92, 90.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (40, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68696 reads in 10.331 CPU sec, 10.097 real sec\n",
      "[M::process] read 68304 sequences (10000108 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 21304, 5, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (153, 284, 436)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1002)\n",
      "[M::mem_pestat] mean and std.dev: (293.40, 145.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1285)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (351, 410, 445)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (163, 633)\n",
      "[M::mem_pestat] mean and std.dev: (401.85, 89.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (43, 761)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68588 reads in 10.145 CPU sec, 9.960 real sec\n",
      "[M::process] read 68268 sequences (10000080 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (15, 21115, 2, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (184, 322, 526)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1210)\n",
      "[M::mem_pestat] mean and std.dev: (326.14, 174.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1552)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (353, 411, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (165, 635)\n",
      "[M::mem_pestat] mean and std.dev: (403.49, 90.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (41, 766)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68304 reads in 9.995 CPU sec, 9.810 real sec\n",
      "[M::process] read 68288 sequences (10000175 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 21042, 3, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 410, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (164, 634)\n",
      "[M::mem_pestat] mean and std.dev: (401.97, 90.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (40, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68268 reads in 10.162 CPU sec, 9.976 real sec\n",
      "[M::process] read 68158 sequences (10000165 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 21096, 5, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 411, 445)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (166, 631)\n",
      "[M::mem_pestat] mean and std.dev: (402.03, 89.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (44, 760)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68288 reads in 10.172 CPU sec, 9.972 real sec\n",
      "[M::process] read 68218 sequences (10000278 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 21001, 3, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (381, 455, 723)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1407)\n",
      "[M::mem_pestat] mean and std.dev: (410.70, 208.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1749)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (354, 411, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (170, 630)\n",
      "[M::mem_pestat] mean and std.dev: (403.51, 88.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (49, 758)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68158 reads in 10.137 CPU sec, 9.938 real sec\n",
      "[M::process] read 68162 sequences (10000131 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 21050, 2, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (353, 410, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (167, 632)\n",
      "[M::mem_pestat] mean and std.dev: (403.04, 90.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (42, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68218 reads in 10.181 CPU sec, 9.991 real sec\n",
      "[M::process] read 68208 sequences (10000241 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 21008, 3, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (238, 333, 397)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 715)\n",
      "[M::mem_pestat] mean and std.dev: (270.00, 136.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 874)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (351, 410, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (161, 636)\n",
      "[M::mem_pestat] mean and std.dev: (401.71, 91.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (36, 768)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68162 reads in 10.183 CPU sec, 10.008 real sec\n",
      "[M::process] read 68168 sequences (10000093 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 21234, 6, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (302, 353, 396)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (114, 584)\n",
      "[M::mem_pestat] mean and std.dev: (357.00, 70.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (20, 678)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 410, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (164, 634)\n",
      "[M::mem_pestat] mean and std.dev: (402.46, 90.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (41, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68208 reads in 10.005 CPU sec, 9.809 real sec\n",
      "[M::process] read 68216 sequences (10000111 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 21083, 7, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (165, 367, 444)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1002)\n",
      "[M::mem_pestat] mean and std.dev: (308.67, 124.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1281)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 409, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (164, 634)\n",
      "[M::mem_pestat] mean and std.dev: (402.43, 90.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (39, 766)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68168 reads in 10.420 CPU sec, 10.226 real sec\n",
      "[M::process] read 68306 sequences (10000027 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 20970, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (348, 409, 445)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (154, 639)\n",
      "[M::mem_pestat] mean and std.dev: (400.28, 93.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (25, 775)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68216 reads in 10.227 CPU sec, 10.048 real sec\n",
      "[M::process] read 68386 sequences (10000255 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 21092, 6, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (260, 303, 436)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 788)\n",
      "[M::mem_pestat] mean and std.dev: (310.88, 151.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 964)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 410, 445)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (166, 631)\n",
      "[M::mem_pestat] mean and std.dev: (401.31, 88.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (47, 756)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68306 reads in 10.445 CPU sec, 10.268 real sec\n",
      "[M::process] read 68554 sequences (10000012 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 21204, 2, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 273, 412)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 996)\n",
      "[M::mem_pestat] mean and std.dev: (257.70, 147.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1288)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (350, 409, 445)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (160, 635)\n",
      "[M::mem_pestat] mean and std.dev: (401.28, 90.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (39, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68386 reads in 10.181 CPU sec, 9.999 real sec\n",
      "[M::process] read 68660 sequences (10000017 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 21217, 8, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 410, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (164, 634)\n",
      "[M::mem_pestat] mean and std.dev: (402.84, 90.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (42, 764)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68554 reads in 10.088 CPU sec, 9.882 real sec\n",
      "[M::process] read 68862 sequences (10000011 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 21350, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (351, 409, 445)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (163, 633)\n",
      "[M::mem_pestat] mean and std.dev: (401.57, 89.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (43, 760)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68660 reads in 10.057 CPU sec, 9.853 real sec\n",
      "[M::process] read 19406 sequences (2834759 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 21224, 6, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (351, 409, 444)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (165, 630)\n",
      "[M::mem_pestat] mean and std.dev: (401.34, 88.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (48, 755)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68862 reads in 10.334 CPU sec, 10.218 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 5883, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (353, 409, 445)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (169, 629)\n",
      "[M::mem_pestat] mean and std.dev: (400.62, 87.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (52, 750)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 19406 reads in 3.067 CPU sec, 3.001 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 1 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/L3.R1.fastq.gz ./metagenomics/reads/L3.R2.fastq.gz\n",
      "[main] Real time: 293.239 sec; CPU: 298.469 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 68614 sequences (10000268 bp)...\n",
      "[M::process] read 68650 sequences (10000188 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 11734, 3, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (373, 430, 550)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 904)\n",
      "[M::mem_pestat] mean and std.dev: (457.58, 142.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1081)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68614 reads in 12.075 CPU sec, 11.887 real sec\n",
      "[M::process] read 68592 sequences (10000201 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 11705, 4, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (374, 430, 555)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (12, 917)\n",
      "[M::mem_pestat] mean and std.dev: (458.95, 142.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1098)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68650 reads in 12.114 CPU sec, 11.927 real sec\n",
      "[M::process] read 68522 sequences (10000226 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 11732, 9, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (374, 430, 554)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 914)\n",
      "[M::mem_pestat] mean and std.dev: (459.66, 144.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1094)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68592 reads in 12.221 CPU sec, 12.022 real sec\n",
      "[M::process] read 68528 sequences (10000163 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 11566, 7, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (373, 429, 557)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (5, 925)\n",
      "[M::mem_pestat] mean and std.dev: (460.03, 145.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1109)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68522 reads in 12.226 CPU sec, 12.040 real sec\n",
      "[M::process] read 68558 sequences (10000121 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 11593, 9, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (374, 431, 558)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (6, 926)\n",
      "[M::mem_pestat] mean and std.dev: (460.96, 142.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1110)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68528 reads in 12.231 CPU sec, 12.041 real sec\n",
      "[M::process] read 68580 sequences (10000043 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 11648, 15, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (374, 430, 555)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (12, 917)\n",
      "[M::mem_pestat] mean and std.dev: (459.60, 142.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1098)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 525, 2926)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8372)\n",
      "[M::mem_pestat] mean and std.dev: (1608.60, 1993.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11095)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68558 reads in 12.180 CPU sec, 11.999 real sec\n",
      "[M::process] read 68578 sequences (10000020 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 11558, 6, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 429, 551)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 909)\n",
      "[M::mem_pestat] mean and std.dev: (455.49, 141.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1088)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68580 reads in 12.383 CPU sec, 12.194 real sec\n",
      "[M::process] read 68686 sequences (10000271 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 11715, 9, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 429, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 902)\n",
      "[M::mem_pestat] mean and std.dev: (454.37, 140.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1079)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68578 reads in 12.190 CPU sec, 11.998 real sec\n",
      "[M::process] read 68716 sequences (10000063 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 11510, 7, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (375, 431, 554)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 912)\n",
      "[M::mem_pestat] mean and std.dev: (458.94, 142.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1091)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68686 reads in 12.342 CPU sec, 12.163 real sec\n",
      "[M::process] read 68682 sequences (10000059 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 11581, 6, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 427, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 899)\n",
      "[M::mem_pestat] mean and std.dev: (453.56, 141.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1075)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68716 reads in 12.386 CPU sec, 12.192 real sec\n",
      "[M::process] read 68646 sequences (10000262 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 11802, 5, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (373, 428, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 895)\n",
      "[M::mem_pestat] mean and std.dev: (455.10, 139.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1069)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68682 reads in 12.227 CPU sec, 12.049 real sec\n",
      "[M::process] read 68688 sequences (10000090 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 11637, 8, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 429, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 902)\n",
      "[M::mem_pestat] mean and std.dev: (454.28, 142.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1079)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68646 reads in 12.251 CPU sec, 12.071 real sec\n",
      "[M::process] read 68744 sequences (10000005 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 11724, 3, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 428, 550)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (16, 906)\n",
      "[M::mem_pestat] mean and std.dev: (455.10, 141.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1084)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68688 reads in 12.262 CPU sec, 12.083 real sec\n",
      "[M::process] read 69040 sequences (10000227 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 11720, 5, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 428, 545)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 893)\n",
      "[M::mem_pestat] mean and std.dev: (452.58, 138.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1067)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68744 reads in 12.437 CPU sec, 12.260 real sec\n",
      "[M::process] read 69100 sequences (10000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 11687, 9, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (373, 429, 546)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 892)\n",
      "[M::mem_pestat] mean and std.dev: (454.62, 138.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1065)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69040 reads in 12.357 CPU sec, 12.157 real sec\n",
      "[M::process] read 69200 sequences (10000012 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 11678, 7, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 427, 542)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (29, 884)\n",
      "[M::mem_pestat] mean and std.dev: (451.52, 136.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1055)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69100 reads in 12.420 CPU sec, 12.215 real sec\n",
      "[M::process] read 69274 sequences (10000021 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 11638, 9, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 428, 542)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (32, 882)\n",
      "[M::mem_pestat] mean and std.dev: (453.21, 135.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1052)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69200 reads in 12.384 CPU sec, 12.163 real sec\n",
      "[M::process] read 69500 sequences (10000200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 11786, 4, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 427, 543)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (21, 891)\n",
      "[M::mem_pestat] mean and std.dev: (450.92, 136.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1065)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69274 reads in 12.380 CPU sec, 12.197 real sec\n",
      "[M::process] read 69696 sequences (10000272 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 11833, 9, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 426, 540)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (33, 878)\n",
      "[M::mem_pestat] mean and std.dev: (450.10, 134.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1047)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69500 reads in 12.187 CPU sec, 12.015 real sec\n",
      "[M::process] read 69380 sequences (10000206 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 11982, 8, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 426, 544)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 894)\n",
      "[M::mem_pestat] mean and std.dev: (449.61, 137.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1069)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69696 reads in 12.276 CPU sec, 12.082 real sec\n",
      "[M::process] read 68844 sequences (10000282 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 11889, 9, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 427, 541)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (31, 881)\n",
      "[M::mem_pestat] mean and std.dev: (450.86, 136.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1051)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69380 reads in 12.259 CPU sec, 12.057 real sec\n",
      "[M::process] read 68926 sequences (10000261 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 11792, 5, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 428, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 897)\n",
      "[M::mem_pestat] mean and std.dev: (454.12, 139.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1072)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68844 reads in 12.058 CPU sec, 11.871 real sec\n",
      "[M::process] read 68844 sequences (10000086 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 11871, 1, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 429, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 897)\n",
      "[M::mem_pestat] mean and std.dev: (454.12, 139.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1072)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68926 reads in 12.533 CPU sec, 12.335 real sec\n",
      "[M::process] read 68772 sequences (10000132 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 11658, 3, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 427, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 897)\n",
      "[M::mem_pestat] mean and std.dev: (454.24, 138.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1072)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68844 reads in 16.657 CPU sec, 16.586 real sec\n",
      "[M::process] read 68786 sequences (10000123 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 11814, 9, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 427, 541)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 883)\n",
      "[M::mem_pestat] mean and std.dev: (450.29, 138.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1054)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68772 reads in 12.298 CPU sec, 12.077 real sec\n",
      "[M::process] read 68826 sequences (10000036 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 11727, 5, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 428, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 900)\n",
      "[M::mem_pestat] mean and std.dev: (453.67, 141.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1076)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (242, 517, 671)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1529)\n",
      "[M::mem_pestat] mean and std.dev: (315.25, 229.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1958)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 68786 reads in 12.967 CPU sec, 12.790 real sec\n",
      "[M::process] read 68808 sequences (10000057 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 11852, 5, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 428, 546)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 894)\n",
      "[M::mem_pestat] mean and std.dev: (453.66, 139.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1068)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68826 reads in 12.617 CPU sec, 12.406 real sec\n",
      "[M::process] read 68728 sequences (10000298 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 11833, 9, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 427, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 902)\n",
      "[M::mem_pestat] mean and std.dev: (453.78, 139.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1079)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68808 reads in 12.369 CPU sec, 12.170 real sec\n",
      "[M::process] read 3492 sequences (506946 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 11734, 10, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 429, 541)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (31, 881)\n",
      "[M::mem_pestat] mean and std.dev: (452.15, 137.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1051)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (36, 245, 2149)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6375)\n",
      "[M::mem_pestat] mean and std.dev: (1111.90, 1425.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8488)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68728 reads in 12.846 CPU sec, 12.769 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 577, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 426, 546)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (15, 900)\n",
      "[M::mem_pestat] mean and std.dev: (452.86, 140.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1077)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 3492 reads in 0.701 CPU sec, 0.665 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 1 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/P1.R1.fastq.gz ./metagenomics/reads/P1.R2.fastq.gz\n",
      "[main] Real time: 358.230 sec; CPU: 363.096 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 68384 sequences (10000102 bp)...\n",
      "[M::process] read 68416 sequences (10000097 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15014, 7, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (382, 440, 552)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 892)\n",
      "[M::mem_pestat] mean and std.dev: (461.99, 137.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1062)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68384 reads in 11.152 CPU sec, 11.018 real sec\n",
      "[M::process] read 68372 sequences (10000216 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15182, 12, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (382, 442, 559)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 913)\n",
      "[M::mem_pestat] mean and std.dev: (465.32, 140.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1090)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (158, 637, 2934)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8486)\n",
      "[M::mem_pestat] mean and std.dev: (1580.25, 2221.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11262)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68416 reads in 10.883 CPU sec, 10.687 real sec\n",
      "[M::process] read 68334 sequences (10000159 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 14897, 12, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (381, 443, 558)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 912)\n",
      "[M::mem_pestat] mean and std.dev: (464.28, 141.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1089)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (142, 344, 675)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1741)\n",
      "[M::mem_pestat] mean and std.dev: (333.82, 244.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2274)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68372 reads in 11.607 CPU sec, 11.431 real sec\n",
      "[M::process] read 68372 sequences (10000244 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15010, 9, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (384, 443, 561)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (30, 915)\n",
      "[M::mem_pestat] mean and std.dev: (466.37, 138.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1092)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68334 reads in 11.399 CPU sec, 11.203 real sec\n",
      "[M::process] read 68384 sequences (10000253 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 14966, 9, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (382, 442, 556)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 904)\n",
      "[M::mem_pestat] mean and std.dev: (464.04, 137.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1078)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68372 reads in 11.106 CPU sec, 10.917 real sec\n",
      "[M::process] read 68428 sequences (10000087 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 14853, 8, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (380, 442, 554)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (32, 902)\n",
      "[M::mem_pestat] mean and std.dev: (462.81, 138.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1076)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68384 reads in 11.643 CPU sec, 11.446 real sec\n",
      "[M::process] read 68352 sequences (10000289 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 14989, 7, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (380, 441, 554)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (32, 902)\n",
      "[M::mem_pestat] mean and std.dev: (462.32, 138.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1076)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68428 reads in 11.620 CPU sec, 11.435 real sec\n",
      "[M::process] read 68456 sequences (10000194 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 14768, 6, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 440, 553)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (31, 901)\n",
      "[M::mem_pestat] mean and std.dev: (460.14, 138.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1075)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68352 reads in 11.291 CPU sec, 11.093 real sec\n",
      "[M::process] read 68564 sequences (10000100 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 14966, 9, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (380, 440, 553)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 899)\n",
      "[M::mem_pestat] mean and std.dev: (461.80, 136.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1072)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68456 reads in 11.672 CPU sec, 11.489 real sec\n",
      "[M::process] read 68384 sequences (10000291 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 14697, 9, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (382, 441, 553)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (40, 895)\n",
      "[M::mem_pestat] mean and std.dev: (462.05, 134.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1066)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68564 reads in 11.627 CPU sec, 11.405 real sec\n",
      "[M::process] read 68472 sequences (10000190 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 15005, 5, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 440, 550)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (37, 892)\n",
      "[M::mem_pestat] mean and std.dev: (458.90, 136.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1063)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68384 reads in 11.135 CPU sec, 10.950 real sec\n",
      "[M::process] read 68446 sequences (10000282 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 14918, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 440, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 883)\n",
      "[M::mem_pestat] mean and std.dev: (459.38, 136.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1051)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68472 reads in 11.509 CPU sec, 11.315 real sec\n",
      "[M::process] read 68476 sequences (10000277 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 14967, 11, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 440, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 883)\n",
      "[M::mem_pestat] mean and std.dev: (458.23, 134.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1051)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (75, 131, 889)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2517)\n",
      "[M::mem_pestat] mean and std.dev: (261.00, 308.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3331)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68446 reads in 11.650 CPU sec, 11.467 real sec\n",
      "[M::process] read 68586 sequences (10000260 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 14958, 6, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (383, 441, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (53, 878)\n",
      "[M::mem_pestat] mean and std.dev: (459.88, 133.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1043)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68476 reads in 11.321 CPU sec, 11.139 real sec\n",
      "[M::process] read 68864 sequences (10000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 14896, 5, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (380, 440, 546)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (48, 878)\n",
      "[M::mem_pestat] mean and std.dev: (458.04, 134.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1044)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68586 reads in 11.402 CPU sec, 11.180 real sec\n",
      "[M::process] read 68858 sequences (10000063 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15082, 13, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (378, 439, 544)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 876)\n",
      "[M::mem_pestat] mean and std.dev: (456.83, 132.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1042)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (298, 432, 3777)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10735)\n",
      "[M::mem_pestat] mean and std.dev: (1887.62, 2097.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14214)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68864 reads in 11.668 CPU sec, 11.469 real sec\n",
      "[M::process] read 69060 sequences (10000116 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15211, 13, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 439, 546)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 880)\n",
      "[M::mem_pestat] mean and std.dev: (457.30, 132.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1047)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (223, 641, 3409)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9781)\n",
      "[M::mem_pestat] mean and std.dev: (1800.15, 2026.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12967)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68858 reads in 11.385 CPU sec, 11.182 real sec\n",
      "[M::process] read 69104 sequences (10000190 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15170, 6, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (385, 441, 549)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (57, 877)\n",
      "[M::mem_pestat] mean and std.dev: (461.72, 133.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1041)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69060 reads in 11.441 CPU sec, 11.240 real sec\n",
      "[M::process] read 69206 sequences (10000268 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15175, 13, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (376, 437, 543)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 877)\n",
      "[M::mem_pestat] mean and std.dev: (454.55, 132.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1044)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 240, 4511)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13297)\n",
      "[M::mem_pestat] mean and std.dev: (1431.38, 2034.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 17690)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 69104 reads in 11.784 CPU sec, 11.581 real sec\n",
      "[M::process] read 69506 sequences (10000276 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 15056, 11, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 439, 542)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (53, 868)\n",
      "[M::mem_pestat] mean and std.dev: (454.80, 130.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1031)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (97, 789, 3316)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9754)\n",
      "[M::mem_pestat] mean and std.dev: (1684.73, 2037.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12973)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 69206 reads in 11.386 CPU sec, 11.206 real sec\n",
      "[M::process] read 69552 sequences (10000267 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15028, 6, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (378, 437, 540)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (54, 864)\n",
      "[M::mem_pestat] mean and std.dev: (452.99, 130.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1026)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69506 reads in 11.502 CPU sec, 11.316 real sec\n",
      "[M::process] read 68654 sequences (10000172 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15239, 10, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (378, 438, 542)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (50, 870)\n",
      "[M::mem_pestat] mean and std.dev: (455.06, 132.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1034)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (79, 763, 3266)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9640)\n",
      "[M::mem_pestat] mean and std.dev: (1337.10, 1502.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12827)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 69552 reads in 11.779 CPU sec, 11.575 real sec\n",
      "[M::process] read 68586 sequences (10000230 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15210, 11, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (377, 439, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (35, 890)\n",
      "[M::mem_pestat] mean and std.dev: (457.44, 134.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1061)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (231, 882, 3442)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9864)\n",
      "[M::mem_pestat] mean and std.dev: (1658.27, 1712.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13075)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68654 reads in 11.178 CPU sec, 10.967 real sec\n",
      "[M::process] read 68640 sequences (10000175 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 15071, 5, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 439, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 883)\n",
      "[M::mem_pestat] mean and std.dev: (458.55, 134.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1051)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68586 reads in 11.270 CPU sec, 11.083 real sec\n",
      "[M::process] read 68502 sequences (10000085 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 14936, 13, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (380, 440, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (44, 884)\n",
      "[M::mem_pestat] mean and std.dev: (459.08, 136.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1052)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (231, 2723, 5450)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15888)\n",
      "[M::mem_pestat] mean and std.dev: (2596.15, 2386.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 21107)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68640 reads in 11.636 CPU sec, 11.433 real sec\n",
      "[M::process] read 68598 sequences (10000075 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 14986, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (378, 438, 543)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (48, 873)\n",
      "[M::mem_pestat] mean and std.dev: (456.08, 133.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1038)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68502 reads in 11.354 CPU sec, 11.166 real sec\n",
      "[M::process] read 68634 sequences (10000061 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 15071, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (380, 440, 547)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 881)\n",
      "[M::mem_pestat] mean and std.dev: (457.84, 133.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1048)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68598 reads in 11.248 CPU sec, 11.057 real sec\n",
      "[M::process] read 68580 sequences (10000105 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 15049, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 439, 548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (41, 886)\n",
      "[M::mem_pestat] mean and std.dev: (457.92, 134.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1055)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68634 reads in 11.763 CPU sec, 11.567 real sec\n",
      "[M::process] read 9230 sequences (1343470 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 14950, 9, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (378, 438, 544)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 876)\n",
      "[M::mem_pestat] mean and std.dev: (455.28, 133.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1042)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68580 reads in 11.254 CPU sec, 11.172 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 2039, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (379, 438, 550)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (37, 892)\n",
      "[M::mem_pestat] mean and std.dev: (459.33, 135.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1063)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 9230 reads in 1.568 CPU sec, 1.502 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 1 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/P2.R1.fastq.gz ./metagenomics/reads/P2.R2.fastq.gz\n",
      "[main] Real time: 328.100 sec; CPU: 333.500 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 68608 sequences (10000038 bp)...\n",
      "[M::process] read 68658 sequences (10000056 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 13160, 7, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 436, 538)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (40, 870)\n",
      "[M::mem_pestat] mean and std.dev: (449.25, 138.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1036)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68608 reads in 12.019 CPU sec, 11.885 real sec\n",
      "[M::process] read 68628 sequences (10000131 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 12979, 7, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 436, 540)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (36, 876)\n",
      "[M::mem_pestat] mean and std.dev: (449.36, 139.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1044)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68658 reads in 12.285 CPU sec, 12.090 real sec\n",
      "[M::process] read 68472 sequences (10000267 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 12884, 8, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (373, 436, 542)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (35, 880)\n",
      "[M::mem_pestat] mean and std.dev: (450.79, 140.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1049)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68628 reads in 11.790 CPU sec, 11.596 real sec\n",
      "[M::process] read 68522 sequences (10000163 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 12810, 9, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 436, 536)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (38, 868)\n",
      "[M::mem_pestat] mean and std.dev: (447.98, 138.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1034)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68472 reads in 12.503 CPU sec, 12.317 real sec\n",
      "[M::process] read 68678 sequences (10000005 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 12729, 7, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 436, 538)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (40, 870)\n",
      "[M::mem_pestat] mean and std.dev: (448.61, 137.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1036)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68522 reads in 12.121 CPU sec, 11.924 real sec\n",
      "[M::process] read 68612 sequences (10000226 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 12830, 2, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (373, 438, 539)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (41, 871)\n",
      "[M::mem_pestat] mean and std.dev: (450.60, 138.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1037)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68678 reads in 12.102 CPU sec, 11.933 real sec\n",
      "[M::process] read 68688 sequences (10000264 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 12891, 7, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 436, 538)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (37, 872)\n",
      "[M::mem_pestat] mean and std.dev: (447.84, 138.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1039)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68612 reads in 12.500 CPU sec, 12.303 real sec\n",
      "[M::process] read 68642 sequences (10000113 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 12911, 11, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 435, 535)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 863)\n",
      "[M::mem_pestat] mean and std.dev: (446.89, 137.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1027)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (751, 2188, 5773)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15817)\n",
      "[M::mem_pestat] mean and std.dev: (2691.55, 2254.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 20839)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68688 reads in 12.183 CPU sec, 11.983 real sec\n",
      "[M::process] read 68704 sequences (10000283 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 12991, 12, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 435, 533)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (50, 855)\n",
      "[M::mem_pestat] mean and std.dev: (446.35, 136.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1016)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 2840, 5132)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14996)\n",
      "[M::mem_pestat] mean and std.dev: (2346.00, 2033.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 19928)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68642 reads in 12.281 CPU sec, 12.110 real sec\n",
      "[M::process] read 68708 sequences (10000269 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 12929, 7, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 434, 532)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 858)\n",
      "[M::mem_pestat] mean and std.dev: (445.19, 138.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1021)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68704 reads in 12.477 CPU sec, 12.284 real sec\n",
      "[M::process] read 68862 sequences (10000169 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 12797, 6, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 435, 534)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 860)\n",
      "[M::mem_pestat] mean and std.dev: (446.44, 135.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1023)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68708 reads in 12.015 CPU sec, 11.803 real sec\n",
      "[M::process] read 69004 sequences (10000010 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 12921, 12, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 434, 533)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (47, 857)\n",
      "[M::mem_pestat] mean and std.dev: (446.80, 136.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1019)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (22, 138, 3169)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9463)\n",
      "[M::mem_pestat] mean and std.dev: (1450.58, 2251.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12610)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68862 reads in 12.429 CPU sec, 12.227 real sec\n",
      "[M::process] read 69200 sequences (10000207 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 12914, 9, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 434, 532)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 856)\n",
      "[M::mem_pestat] mean and std.dev: (444.96, 134.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1018)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69004 reads in 12.427 CPU sec, 12.226 real sec\n",
      "[M::process] read 69296 sequences (10000162 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 13102, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (374, 435, 533)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (56, 851)\n",
      "[M::mem_pestat] mean and std.dev: (447.77, 132.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1010)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69200 reads in 11.773 CPU sec, 11.572 real sec\n",
      "[M::process] read 69730 sequences (10000104 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 13129, 10, 1)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (90, 468, 1727)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5001)\n",
      "[M::mem_pestat] mean and std.dev: (884.40, 1079.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6638)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 432, 521)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (65, 825)\n",
      "[M::mem_pestat] mean and std.dev: (440.34, 129.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 977)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (46, 274, 2441)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7231)\n",
      "[M::mem_pestat] mean and std.dev: (1328.90, 2111.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9775)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 69296 reads in 11.708 CPU sec, 11.513 real sec\n",
      "[M::process] read 69550 sequences (10000115 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 13000, 11, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 432, 523)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (61, 831)\n",
      "[M::mem_pestat] mean and std.dev: (441.98, 131.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 985)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (88, 199, 507)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1345)\n",
      "[M::mem_pestat] mean and std.dev: (242.30, 253.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1764)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 69730 reads in 11.704 CPU sec, 11.511 real sec\n",
      "[M::process] read 68862 sequences (10000026 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 13254, 9, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 434, 525)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (60, 835)\n",
      "[M::mem_pestat] mean and std.dev: (442.89, 131.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 990)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 69550 reads in 11.661 CPU sec, 11.466 real sec\n",
      "[M::process] read 68858 sequences (10000251 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 13210, 5, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 434, 531)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (51, 851)\n",
      "[M::mem_pestat] mean and std.dev: (444.26, 134.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1011)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68862 reads in 11.448 CPU sec, 11.241 real sec\n",
      "[M::process] read 68778 sequences (10000210 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 13101, 6, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 434, 526)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (55, 840)\n",
      "[M::mem_pestat] mean and std.dev: (442.86, 132.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 997)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68858 reads in 11.532 CPU sec, 11.347 real sec\n",
      "[M::process] read 68810 sequences (10000089 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 13097, 5, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 433, 528)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (51, 846)\n",
      "[M::mem_pestat] mean and std.dev: (443.25, 133.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1005)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68778 reads in 11.511 CPU sec, 11.337 real sec\n",
      "[M::process] read 68782 sequences (10000195 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 13059, 9, 2)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (251, 330, 393)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 677)\n",
      "[M::mem_pestat] mean and std.dev: (326.80, 142.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 898)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 436, 535)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 861)\n",
      "[M::mem_pestat] mean and std.dev: (446.70, 135.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1024)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68810 reads in 11.592 CPU sec, 11.408 real sec\n",
      "[M::process] read 68798 sequences (10000163 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 13064, 14, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 434, 527)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (53, 843)\n",
      "[M::mem_pestat] mean and std.dev: (443.48, 132.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1001)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (95, 463, 3409)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10037)\n",
      "[M::mem_pestat] mean and std.dev: (1970.36, 2804.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13351)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 68782 reads in 11.645 CPU sec, 11.466 real sec\n",
      "[M::process] read 68794 sequences (10000009 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 12908, 6, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 433, 531)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 855)\n",
      "[M::mem_pestat] mean and std.dev: (443.65, 135.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1017)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68798 reads in 11.772 CPU sec, 11.592 real sec\n",
      "[M::process] read 68870 sequences (10000213 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 13076, 6, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 434, 529)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (55, 845)\n",
      "[M::mem_pestat] mean and std.dev: (443.82, 135.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1003)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68794 reads in 11.590 CPU sec, 11.402 real sec\n",
      "[M::process] read 68788 sequences (10000180 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 13015, 6, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 433, 529)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (52, 847)\n",
      "[M::mem_pestat] mean and std.dev: (443.60, 133.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1006)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68870 reads in 11.728 CPU sec, 11.536 real sec\n",
      "[M::process] read 68798 sequences (10000280 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 12963, 9, 0)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (324, 382, 636)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1260)\n",
      "[M::mem_pestat] mean and std.dev: (339.00, 164.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1572)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 432, 526)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (55, 840)\n",
      "[M::mem_pestat] mean and std.dev: (442.59, 132.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 997)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 68788 reads in 11.797 CPU sec, 11.586 real sec\n",
      "[M::process] read 68842 sequences (10000294 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 13049, 7, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (369, 433, 525)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (57, 837)\n",
      "[M::mem_pestat] mean and std.dev: (442.13, 132.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 993)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68798 reads in 11.620 CPU sec, 11.432 real sec\n",
      "[M::process] read 69134 sequences (10000107 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 12953, 8, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 434, 528)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (54, 844)\n",
      "[M::mem_pestat] mean and std.dev: (443.12, 133.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1002)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 68842 reads in 11.857 CPU sec, 11.662 real sec\n",
      "[M::process] read 3324 sequences (481171 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 12990, 13, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (370, 434, 527)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (56, 841)\n",
      "[M::mem_pestat] mean and std.dev: (441.82, 132.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 998)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 1043, 2427)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7147)\n",
      "[M::mem_pestat] mean and std.dev: (1283.08, 1267.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9507)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 69134 reads in 11.553 CPU sec, 11.493 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 642, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (374, 439, 551)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (20, 905)\n",
      "[M::mem_pestat] mean and std.dev: (453.15, 139.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1082)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 3324 reads in 0.592 CPU sec, 0.563 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 1 ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/P3.R1.fastq.gz ./metagenomics/reads/P3.R2.fastq.gz\n",
      "[main] Real time: 341.699 sec; CPU: 346.495 sec\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "samples=$(find ./metagenomics/reads -name '*.fastq.gz' -type f -printf '%P\\n' | cut -d '.' -f 1 | sort | uniq)\n",
    "for i in $samples\n",
    "    do echo $i\n",
    "    bwa mem -t 1  ./metagenomics/assembly/scaffolds.fasta ./metagenomics/reads/\"$i\".R1.fastq.gz ./metagenomics/reads/\"$i\".R2.fastq.gz | samtools view -h -b > ./metagenomics/mapping/mapped.\"$i\".bam \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 1337764\r\n",
      "-rw-rw-r-- 1 laura laura 230239629 May 30 12:08 mapped.L1.bam\r\n",
      "-rw-rw-r-- 1 laura laura 229849912 May 30 12:13 mapped.L2.bam\r\n",
      "-rw-rw-r-- 1 laura laura 229877774 May 30 12:18 mapped.L3.bam\r\n",
      "-rw-rw-r-- 1 laura laura 226385741 May 30 12:24 mapped.P1.bam\r\n",
      "-rw-rw-r-- 1 laura laura 226103199 May 30 12:29 mapped.P2.bam\r\n",
      "-rw-rw-r-- 1 laura laura 227392875 May 30 12:35 mapped.P3.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls ./metagenomics/mapping/ -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "samtools sort help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools sort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./metagenomics/mapping/mapped.L1.bam.sorted\n",
      "./metagenomics/mapping/mapped.L2.bam.sorted\n",
      "./metagenomics/mapping/mapped.L3.bam.sorted\n",
      "./metagenomics/mapping/mapped.P1.bam.sorted\n",
      "./metagenomics/mapping/mapped.P2.bam.sorted\n",
      "./metagenomics/mapping/mapped.P3.bam.sorted\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for i in ./metagenomics/mapping/*.bam ;\n",
    "    do echo $i.sorted\n",
    "    samtools sort -o \"$i.sorted\" \"$i\"\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove the samtools view directory:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/laura/gitprojects/metagenomicspractical\n",
      "2.0G\t./metagenomics/assembly\n",
      "946M\t./metagenomics/reads\n",
      "4.0K\t./metagenomics/sorted\n",
      "2.8G\t./metagenomics/mapping\n",
      "5.7G\t./metagenomics/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "pwd\n",
    "du -h ./metagenomics/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!rm -rf ....yourdirectoryname"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "samtools help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools view"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run samtools view to view your BAM files:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools view ....yourbamsortedbam | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2> Binning with MetaBat </h2>\n",
    "\n",
    "Now that we have the reads, the scaffolds and the mapping of the reads on the scaffolds we can continue with the binning. In metagenomics, binning is the process of grouping reads or contigs and assigning them to operational taxonomic units. Binning methods can be based on either compositional features or alignment (similarity), or both. Metabat uses both the contig depth and tetra-nucleotide frequencies to bin the contigs. Every bin will represent one operational taxonomic unit that can be found in the metagenome.\n",
    "\n",
    "<b>Assignment:</b><br>\n",
    "Run the script provided in /metagenomics/scripts/jgi_summarize_bam_contig_depths to calculate the average depth per contig. <br>\n",
    "Then run metabat to bin the contigs, dont forget to include the previously included contig depths in the metabat command."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/sh: 1: ./metagenomics/scripts/jgi_summarize_bam_contig_depths: not found\r\n"
     ]
    }
   ],
   "source": [
    "!./metagenomics/scripts/jgi_summarize_bam_contig_depths -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "calculate the contig depth per scaffold<br>\n",
    "hint: a glob looks like this directoryname/* and includes all files included in the directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!./metagenomics/scripts/jgi_summarize_bam_contig_depths --outputDepth "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "metabat help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!metabat -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a new directory:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run metabat:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!metabat -i -o -t"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2> Checking completeness and contamination using CheckM </h2>\n",
    "\n",
    "Now that we have the bins made with metabat we can check them for contamination and completeness (quality), for this we will use CheckM. CheckM provides a set of tools for assessing the quality of genomes recovered from isolates, single cells, or metagenomes. It provides robust estimates of genome completeness and contamination by using collocated sets of genes that are ubiquitous and single-copy within a phylogenetic lineage (also called marker/signature genes). http://ecogenomics.github.io/CheckM/\n",
    "\n",
    "As you will be able to see in the checkm help pages, checkm has a workflow (lineage_wf) that will run all nessecary steps to asses bin quality. \n",
    "\n",
    "Lineage_wf (lineage-specific workflow) steps: <br>\n",
    "- The tree command places genome bins into a reference genome tree. <br>\n",
    "- The lineage_set command creates a marker file indicating lineage-specific marker sets suitable for evaluating each genome. <br>\n",
    "- This marker file is passed to the analyze command in order to identify marker genes and estimate the completeness and contamination of each genome bin.  <br>\n",
    "- Finally, the qa command can be used to produce different tables summarizing the quality of each genome bin. <br>\n",
    "\n",
    "\n",
    "<b>Assignment:</b><br>\n",
    "Sadly for this exercise the virtual machines we are using are not powerfull enough, therefore we provide the results of the first steps of the CheckM workflow up until the qa command (lineage_wf). Scan the help pages of CheckM to find out the correct command to finish the CheckM analysis. \n",
    "\n",
    "(OPTIONAL: you can try to find out what the limiting factor is of lineage_wf using !/usr/bin/time --verbose) <br>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!checkm -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!checkm lineage_wf -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "checkm qa help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!checkm qa -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run checkm qa:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!checkm qa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "(think and discuss these questions) <br>\n",
    "\n",
    "What did you do? <br>\n",
    "Where is your output? <br>\n",
    "What does your output look like? <br>\n",
    "What can you say about the bins with this output? <br>\n",
    "What can you say about lineages of the bins?<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2>Genome annotation with Prokka</h2>\n",
    "\n",
    "Now that we some extra information about our bins, we can continue to analyze the high quality bins. The final CheckM results will give you a good overview of the bins with low contamination and high completeness and also shows the lowest taxonomic rank of the bin. Pick a bin that you think is interesting to further study.\n",
    "\n",
    "With this bin we are going to do some genome annotation. Whole genome annotation is the process of identifying features of interest in a set of genomic DNA sequences, and labelling them with useful information. Prokka is a software tool to annotate bacterial, archaeal and viral genomes quickly and produce standards-compliant output files.\n",
    "\n",
    "<b>Assignment:</b><br>\n",
    "Have a look at the prokka help pages below and come up with the right command to run prokka. (HINT: look at the Usage tag to run prokka on default mode, think of threads, use --centre X --compliant to stop prokkas complaints about ugly contig names, direct the output towars metagenomics/prokka)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!prokka"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a directory for the prokka output:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "prokka help page:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!prokka -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run prokka:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "!prokka --cpus --outdir "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "(think and discuss these questions) <br>\n",
    "\n",
    "What did you do? <br>\n",
    "Where is your output? <br>\n",
    "What does your output look like? <br>\n",
    "Do you know what all the output files are or mean? <br>\n",
    "\n",
    "(HINT: you can look at the prokka files in the same way we looked at the reads and scaffold earlier)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>investigating prokka output</b><br>\n",
    "to investigate the prokka output you can use two webservers that both are able to place the annotations from prokka in KEGG pathways.<br>\n",
    "\n",
    "(1) prokka gives uniprot IDs in the gff files first we will collect these IDs,\n",
    "(2) since we have Uniprot IDs from prokka we are going to convert these to KO IDs that can be used by KEGG using this website:<br>\n",
    "http://www.uniprot.org/uploadlists/\n",
    "\n",
    "(3) then you can put your IDs in both websites and investigate the pathways\n",
    "http://www.genome.jp/kegg/tool/map_module.html<br>\n",
    "http://pathways.embl.de/iPath2.cgi#<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "view prokka output:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!less ....yourprokkaoutput.gff | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "take the uniprot IDs out of the gff file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!grep -o 'UniProt.*' ....yourprokkaoutput | cut -d';' -f1 | cut -d':' -f2 > listofuniprotIDs"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
