# Metagenomicspractical
Hands-on practical for learning a metagenomics workflow aimed at Masters Students in Life Sciences. 
The practical discusses a metagenomics workflow from acquiring sequencing data up to annotating binned microbial genomes. Steps up to and including assembly have already been performed for feasibility. This practical includes the following steps (and depends on the following tools):
* backmapping (bwa+samtools)
* binning (MetaBAT)
* quality controll of bins (CheckM)
* annotation (Prokka)

# Learning goals of this practical
* After this practical, you can name and explain the steps of a simple metagenomics workflow. Starting at acquiring sequencing data, all the way to annotation of individual draft genomes.
  * You can highlight the differences between ‘regular’ genome sequencing data and assemblies versus metagenomic sequencing data and assemblies.
  * You can replicate the workflow taken during the practical.
  * You can explain the workflow from biological and technical perspectives when either one or both are appropriate. 
  * You can design similar workflows for different metagenomic questions.
  * You can explain what binning signals are, why they are used and how you used them during the practical.
* If not already, you will understand the basics of the bash computer language and be able to run bio-informatic programmes in loops.

# Credits and references
Original practical (version 2017) was made By Margo Schuller and Laura Dijkhuizen. Current version was improved thereupon by [Laura Dijkhuizen](https://www.uu.nl/medewerkers/LWDijkhuizen). The practical is based on a subset of published data, the original paper is published open-access in New Phytologist: 
>Dijkhuizen, L. W., Brouwer, P., Bolhuis, H., Reichart, G. J., Koppers, N., Huettel, B., ... & Wong, G. K. S. (2018). Is there foul play in the leaf pocket? The metagenome of floating fern Azolla reveals endophytes that do not fix N2 but may denitrify. New Phytologist, 217(1), 453-466.
